<!-- build time:Fri Oct 10 2025 09:42:59 GMT+0800 (中国标准时间) --><!DOCTYPE html><html lang="zh-CN"><head><meta charset="UTF-8"><meta name="viewport" content="width=device-width,initial-scale=1,maximum-scale=2"><meta name="theme-color" content="#FFF"><link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon.png"><link rel="icon" type="image/ico" sizes="32x32" href="/images/favicon.ico"><meta http-equiv="Cache-Control" content="no-transform"><meta http-equiv="Cache-Control" content="no-siteapp"><link rel="alternate" type="application/rss+xml" href="http://amentiraz.github.io/rss.xml"><link rel="alternate" type="application/atom+xml" href="http://amentiraz.github.io/atom.xml"><link rel="alternate" type="application/json" href="http://amentiraz.github.io/feed.json"><link rel="stylesheet" href="//fonts.googleapis.com/css?family=Mulish:300,300italic,400,400italic,700,700italic%7CFredericka%20the%20Great:300,300italic,400,400italic,700,700italic%7CNoto%20Serif%20JP:300,300italic,400,400italic,700,700italic%7CNoto%20Serif%20SC:300,300italic,400,400italic,700,700italic%7CInconsolata:300,300italic,400,400italic,700,700italic&display=swap&subset=latin,latin-ext"><link rel="stylesheet" href="/css/app.css?v=0.2.5"><meta name="keywords" content="生物,学习笔记类,算法"><link rel="canonical" href="http://amentiraz.github.io/article/scPerturb/"><title>scPerturb - 论文 | Amentiraz =</title><meta name="generator" content="Hexo 7.3.0"></head><body itemscope itemtype="http://schema.org/WebPage"><div id="loading"><div class="cat"><div class="body"></div><div class="head"><div class="face"></div></div><div class="foot"><div class="tummy-end"></div><div class="bottom"></div><div class="legs left"></div><div class="legs right"></div></div><div class="paw"><div class="hands left"></div><div class="hands right"></div></div></div></div><div id="container"><header id="header" itemscope itemtype="http://schema.org/WPHeader"><div class="inner"><div id="brand"><div class="pjax"><h1 itemprop="name headline">scPerturb</h1><div class="meta"><span class="item" title="创建时间：2025-05-11 10:16:23"><span class="icon"><i class="ic i-calendar"></i> </span><span class="text">发表于</span> <time itemprop="dateCreated datePublished" datetime="2025-05-11T10:16:23+08:00">2025-05-11</time> </span><span class="item" title="本文字数"><span class="icon"><i class="ic i-pen"></i> </span><span class="text">本文字数</span> <span>13k</span> <span class="text">字</span> </span><span class="item" title="阅读时长"><span class="icon"><i class="ic i-clock"></i> </span><span class="text">阅读时长</span> <span>11 分钟</span></span></div></div></div><nav id="nav"><div class="inner"><div class="toggle"><div class="lines" aria-label="切换导航栏"><span class="line"></span> <span class="line"></span> <span class="line"></span></div></div><ul class="menu"><li class="item title"><a href="/" rel="start">Amentiraz</a></li></ul><ul class="right"><li class="item theme"><i class="ic i-sun"></i></li><li class="item search"><i class="ic i-search"></i></li></ul></div></nav></div><div id="imgs" class="pjax"><ul><li class="item" data-background-image="https://amentirazblogpic.oss-cn-hangzhou.aliyuncs.com/blogpic/%E5%BE%AE%E4%BF%A1%E5%9B%BE%E7%89%87_20240925162742.png"></li><li class="item" data-background-image="https://amentirazblogpic.oss-cn-hangzhou.aliyuncs.com/blogpic/%E5%BE%AE%E4%BF%A1%E5%9B%BE%E7%89%87_20240925162951.png"></li><li class="item" data-background-image="https://amentirazblogpic.oss-cn-hangzhou.aliyuncs.com/blogpic/%E5%BE%AE%E4%BF%A1%E5%9B%BE%E7%89%87_20240925162820.png"></li><li class="item" data-background-image="https://amentirazblogpic.oss-cn-hangzhou.aliyuncs.com/blogpic/%E5%BE%AE%E4%BF%A1%E5%9B%BE%E7%89%87_20240925124500.png"></li><li class="item" data-background-image="https://amentirazblogpic.oss-cn-hangzhou.aliyuncs.com/blogpic/%E5%BE%AE%E4%BF%A1%E5%9B%BE%E7%89%87_20240925162917.png"></li><li class="item" data-background-image="https://amentirazblogpic.oss-cn-hangzhou.aliyuncs.com/blogpic/%E5%BE%AE%E4%BF%A1%E5%9B%BE%E7%89%87_20240925162923.png"></li></ul></div></header><div id="waves"><svg class="waves" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" viewBox="0 24 150 28" preserveAspectRatio="none" shape-rendering="auto"><defs><path id="gentle-wave" d="M-160 44c30 0 58-18 88-18s 58 18 88 18 58-18 88-18 58 18 88 18 v44h-352z"/></defs><g class="parallax"><use xlink:href="#gentle-wave" x="48" y="0"/><use xlink:href="#gentle-wave" x="48" y="3"/><use xlink:href="#gentle-wave" x="48" y="5"/><use xlink:href="#gentle-wave" x="48" y="7"/></g></svg></div><main><div class="inner"><div id="main" class="pjax"><div class="article wrap"><div class="breadcrumb" itemscope itemtype="https://schema.org/BreadcrumbList"><i class="ic i-home"></i> <span><a href="/">首页</a></span><i class="ic i-angle-right"></i> <span class="current" itemprop="itemListElement" itemscope itemtype="https://schema.org/ListItem"><a href="/categories/article/" itemprop="item" rel="index" title="分类于 论文"><span itemprop="name">论文</span></a><meta itemprop="position" content="1"></span></div><article itemscope itemtype="http://schema.org/Article" class="post block" lang="zh-CN"><link itemprop="mainEntityOfPage" href="http://amentiraz.github.io/article/scPerturb/"><span hidden itemprop="author" itemscope itemtype="http://schema.org/Person"><meta itemprop="image" content="/images/avatar.jpg"><meta itemprop="name" content="Lemon Sour"><meta itemprop="description" content=", "></span><span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization"><meta itemprop="name" content=""></span><div class="body md" itemprop="articleBody"><p>scPerturb: harmonized single-cell perturbation data.</p><span id="more"></span><p>第一次接触这个领域，仔细的研读一下。</p><h1 id="abstract"><a class="anchor" href="#abstract">#</a> Abstract</h1><p>这篇文章应用了 uniform quality control pipelines 和 harmonize feature annotations. 然后对于它的模型 scPerturb 的目的是 facilitates comparison and integration across datasets. 它们描述了一个 energy statistics (E-statistics) 去衡量 perturbation 和显著性检验。</p><p>This work provides an information resource for researchers working with single-cell perturbation data and recommendations for experimental design, including optimal cell counts and read depth.</p><h1 id="introduction"><a class="anchor" href="#introduction">#</a> Introduction</h1><p>Perturbation experiments probe the response of cells or cellular systems to changes in conditions.</p><p>Nowadays, with advanced functional genomics techniques, single-cell genetic perturbations acting on individual cellular components are available. Perturbations using different technologies target different layers of the hierarchy of protein production.</p><p><img data-src="https://amentirazblogpic.oss-cn-hangzhou.aliyuncs.com/blogpic/41592_2023_2144_Fig1_HTML.webp" alt=""><br>这里解释一下这张图，上面提到的 different technologies 指的是例如 CRISPR-cas9、CRISPRi 这些东西。比如 CRISPR-cas9 它在基因上直接做增删，CRISPRi 和 CRISPRa 分别抑制和激活转录 (transcription)，CRISPR-cas13 切割 RNA，一些小分子的药物影响蛋白质产物例如酶和受体。</p><p>然后是比较关键的一句话： Barcodes associated with unique CRISPR guide perturbations are read alongside single-cell RNA sequencing (scRNA-seq), CITE-seq (cellular indexing of transcriptomes and epitopes by sequencing) or scATAC-seq (single-cell sequencing assay for transposase-accessible chromatin) to identify each cell’s perturbation condition.</p><p>感觉很难理解上面这句话，我拿 GPT 跑了一个解释。</p><details class="danger"><summary>三种数据</summary><div><p>下面分别对这三种单细胞多组学技术做详细介绍，包括它们的基本原理、实验流程、优势与局限，以及典型应用场景。</p><hr><h2 id="1-single-cell-rna-sequencing-scrna-seq"><a class="anchor" href="#1-single-cell-rna-sequencing-scrna-seq">#</a> 1. Single-cell RNA sequencing (scRNA-seq)</h2><h3 id="基本原理"><a class="anchor" href="#基本原理">#</a> 基本原理</h3><p>scRNA-seq 用于在单细胞水平上定量每个细胞的转录组（mRNA）表达。它能够揭示细胞间在基因表达谱上的异质性，识别不同细胞类型、细胞状态和发育轨迹。</p><h3 id="实验流程"><a class="anchor" href="#实验流程">#</a> 实验流程</h3><ol><li><p><strong>单细胞分离</strong></p><ul><li>微流控芯片（如 10x Genomics Chromium）或滴式微流控（Drop-seq、inDrop）将单个细胞包封于微小液滴或孔室中。</li></ul></li><li><p><strong>细胞裂解与逆转录</strong></p><ul><li>每个液滴内加入含有细胞条形码（cell barcode）和分子标签（UMI, unique molecular identifier）的引物，裂解细胞并逆转录生成带有条形码的 cDNA。</li></ul></li><li><p><strong>文库构建与测序</strong></p><ul><li>将所有液滴中的 cDNA 汇合，进行扩增、建库并上机高通量测序（Illumina）。</li></ul></li><li><p><strong>数据处理与分析</strong></p><ul><li>根据条形码将每条读数（read）分配回对应细胞，UMI 去重后得到每个细胞中每个基因的表达量矩阵。</li><li>下游分析包括细胞聚类、差异表达、伪时序分析等。</li></ul></li></ol><h3 id="优势"><a class="anchor" href="#优势">#</a> 优势</h3><ul><li>能在单细胞水平揭示生物体系的异质性。</li><li>支持大规模细胞通量（可达数万到数十万细胞）。</li><li>数据成熟，分析工具丰富（Seurat、Scanpy 等）。</li></ul><h3 id="局限"><a class="anchor" href="#局限">#</a> 局限</h3><ul><li>只捕获多聚 A 尾 mRNA，无法直接测到非 polyadenylated RNA（如部分非编码 RNA）。</li><li>有 “高掉落率”（dropout）现象：低丰度转录本常常检测不到。</li><li>无空间信息，需要结合空间转录组或成像技术。</li></ul><h3 id="应用场景"><a class="anchor" href="#应用场景">#</a> 应用场景</h3><ul><li>细胞类型鉴定与注释</li><li>发育过程伪时序重构</li><li>疾病微环境中的免疫细胞分型</li><li>药物或基因扰动后细胞响应分析</li></ul><hr><h2 id="2-cite-seq"><a class="anchor" href="#2-cite-seq">#</a> 2. CITE-seq</h2><p>(Cellular Indexing of Transcriptomes and Epitopes by Sequencing)</p><h3 id="基本原理-2"><a class="anchor" href="#基本原理-2">#</a> 基本原理</h3><p>CITE-seq 在常规模版 scRNA-seq 的基础上，额外对细胞表面或胞内蛋白进行标记与测量。它使用带有寡核苷酸标签的抗体（antibody–oligo conjugates），让蛋白质水平的定量也能通过测序来获取。</p><h3 id="实验流程-2"><a class="anchor" href="#实验流程-2">#</a> 实验流程</h3><ol><li><p><strong>抗体标记</strong></p><ul><li>每种待测抗体共轭一个短的 DNA 报告子序列（antibody barcode）。</li></ul></li><li><p><strong>细胞染色</strong></p><ul><li>将多重标记的抗体混合物与细胞（或细胞悬液）孵育，使抗体特异性结合对应表面蛋白。</li></ul></li><li><p><strong>单细胞分离与裂解</strong></p><ul><li>如同 scRNA-seq，包封单细胞并裂解，同时释放 mRNA 和抗体上的 DNA barcode。</li></ul></li><li><p><strong>共逆转录与扩增</strong></p><ul><li>同时逆转录 mRNA（带细胞条形码 + UMI）与抗体 DNA barcode。</li></ul></li><li><p><strong>文库构建与测序</strong></p><ul><li>分离两种 cDNA：一部分用于测序转录组，另一部分用于测序抗体验证条形码（常用定制引物扩增）。</li></ul></li><li><p><strong>数据融合</strong></p><ul><li>将每个细胞的基因表达数据与对应的抗体信号（epitope 数量）整合，得到 “RNA + 蛋白” 联合矩阵。</li></ul></li></ol><h3 id="优势-2"><a class="anchor" href="#优势-2">#</a> 优势</h3><ul><li>同时获得转录组与蛋白组信息，提高细胞类型鉴别的精度。</li><li>蛋白质水平数据对低丰度或翻译后调控的靶点更敏感。</li><li>可以同时检测几十到上百种表面 / 胞内蛋白。</li></ul><h3 id="局限-2"><a class="anchor" href="#局限-2">#</a> 局限</h3><ul><li>抗体质量和特异性决定实验效果，需要严格的抗体验证。</li><li>抗体–oligo 的成本和制备复杂度高于普通 scRNA-seq 试剂。</li><li>通量受抗体组数和测序深度限制。</li></ul><h3 id="应用场景-2"><a class="anchor" href="#应用场景-2">#</a> 应用场景</h3><ul><li>免疫细胞谱系分型（T 细胞亚群、B 细胞亚群鉴定）</li><li>转录组与表型标志物联合分析</li><li>细胞分化过程中蛋白与 RNA 调控耦合研究</li><li>临床样本中多标志物联合表型剖析</li></ul><hr><h2 id="3-single-cell-atac-sequencing-scatac-seq"><a class="anchor" href="#3-single-cell-atac-sequencing-scatac-seq">#</a> 3. Single-cell ATAC sequencing (scATAC-seq)</h2><p>(Assay for Transposase-Accessible Chromatin)</p><h3 id="基本原理-3"><a class="anchor" href="#基本原理-3">#</a> 基本原理</h3><p>scATAC-seq 用于在单细胞水平绘制染色质开放性（chromatin accessibility）图谱，揭示基因调控元件（如启动子、增强子）在不同细胞类型或状态下的可及性差异。</p><h3 id="实验流程-3"><a class="anchor" href="#实验流程-3">#</a> 实验流程</h3><ol><li><p><strong>单细胞分离</strong></p><ul><li>类似 scRNA-seq，可用微流控芯片（10x Genomics Chromium）或微流体滴式系统。</li></ul></li><li><p><strong>Tn5 转座酶标记</strong></p><ul><li>在裂解后的细胞核中加入加载了测序接头的 Tn5 转座酶，它会 preferentially 插入到开放染色质区，使这些区域同时获得测序接头。</li></ul></li><li><p><strong>条形码融合</strong></p><ul><li>通过在微液滴或孔室中完成转座反应，使插入片段带上细胞条形码和 UMI。</li></ul></li><li><p><strong>扩增与测序</strong></p><ul><li>扩增插入了接头的 DNA 片段，构建文库并进行高通量测序。</li></ul></li><li><p><strong>数据处理与分析</strong></p><ul><li>根据条形码将读数分配到细胞，去除 PCR 重复后得到每个细胞在基因组各位点的可及性矩阵（细胞 × peaks）。</li><li>下游可做细胞聚类、TF motif 分析、基因调控网络推断等。</li></ul></li></ol><h3 id="优势-3"><a class="anchor" href="#优势-3">#</a> 优势</h3><ul><li>直接捕获调控元件活性，补充转录组数据无法提供的调控层信息。</li><li>能鉴定特异细胞类型的关键增强子或 TF 結合位点。</li><li>与 scRNA-seq 联合（例如 Multiome）可同时获取转录组和染色质可及性数据。</li></ul><h3 id="局限-3"><a class="anchor" href="#局限-3">#</a> 局限</h3><ul><li>通量和数据稀疏性问题更严重，通常每个细胞只有几千到一万条有效片段。</li><li>峰（peak）调用和比对较复杂，对计算资源要求高。</li><li>需要更深的测序深度以覆盖低频开放区域。</li></ul><h3 id="应用场景-3"><a class="anchor" href="#应用场景-3">#</a> 应用场景</h3><ul><li>构建细胞类型特异的调控元件图谱</li><li>推断转录因子活性与基因调控网络</li><li>发育过程中染色质重塑动态研究</li><li>结合 GWAS 数据定位变异影响的调控元件</li></ul><hr><p><strong>总结对比</strong></p><table><thead><tr><th>特性</th><th>scRNA-seq</th><th>CITE-seq</th><th>scATAC-seq</th></tr></thead><tbody><tr><td>主要测量层面</td><td>mRNA 转录本</td><td>mRNA + 表面 / 胞内蛋白</td><td>染色质开放性（调控元件可及性）</td></tr><tr><td>条形码与 UMI</td><td>都有</td><td>都有</td><td>都有</td></tr><tr><td>数据稀疏性</td><td>较高（dropout）</td><td>较高（RNA 部分） + 蛋白部分相对低 sparsity</td><td>非常高（开放片段稀疏）</td></tr><tr><td>分辨信息</td><td>转录本丰度</td><td>转录本 + 表型蛋白</td><td>调控元件活性</td></tr><tr><td>典型通量</td><td>10⁴–10⁵ 细胞</td><td>10⁴ 细胞（视抗体数量与测序深度而定）</td><td>10³–10⁴ 细胞</td></tr></tbody></table><p>这三种技术各有侧重，常常结合使用以获得细胞多层次的分子图谱，帮助我们更全面地理解细胞状态与功能。</p></div></details><p>说实话，总感觉这段话有点牵强，真的和 perturbations 相关性这么大吗。</p><p>下面的一段说了一下大规模数据的影响，目前精确去预测网络的 interactions 还是受限于数据规模小的影响。Large-scale single-cell perturbation–response screens enable exploration of complex cellular behavior inaccessible in bulk measurements.</p><p>研究目的：Reliable analysis of increasingly large perturbation datasets requires efficient statistical tools to harness large numbers of cells and perturbations. 研究背景：The inherently high dimensionality of perturbation–response data complicates calculation of distances between perturbations, as does cell-to-cell variation and data sparsity. 本文基于各种 Multiple distance measures 方法中的 energy distance。</p><p>“Cell lines”（细胞系）是指在体外条件下（通常是培养皿或培养瓶中），从多细胞生物体取出的细胞经适当处理后，能够持续生长、分裂并保持特定特征的一类细胞群体。</p><p>Large perturbation screens are specifically designed to study a particular system, such as a cell line, under a set of perturbations of interest. 研究背景：“the field has accumulated a heterogeneous assortment of single-cell perturbation–response data with a wide range of different cell types, such as immortalized cell lines and induced pluripotent stem cell-derived models, and different perturbation technologies, including knockouts, activation, interference, base editing and prime editing”。所以 Computational methods to efficiently harmonize these different perturbation datasets are needed.</p><p>所以这个文章干了两件事，第一，提供了 scPerturb 去标准化数据集，第二，提供了 E-distance 作为衡量指标。</p><p>还是有点懵。</p><h1 id="results"><a class="anchor" href="#results">#</a> Results</h1><p><img data-src="https://amentirazblogpic.oss-cn-hangzhou.aliyuncs.com/blogpic/41592_2023_2144_Fig2_HTML.webp" alt=""></p><p>这里记一个结论：The average sequencing depth, that is, the mean number of reads per cell, in each study affects the number of low-expression genes observed. Increasing sequencing depth increases the UMI counts measured even for low-expression genes, reducing the uncertainty associated with zero counts.</p><p>“测序深度（sequencing depth）” 是指在测序实验中，每个位点或每个分子被测序的平均次数，它反映的是数据的丰富程度或覆盖程度。</p><p>这一大段挺难理解的，我拿 GPT 跑了一个解释：<br>这段话的核心在讲：<strong>不同数据集的测序质量指标差异很大，而测序深度对单细胞扰动实验的可解释性和分析结果有显著影响。</strong></p><p>具体讲了以下几点：</p><hr><h3 id="1-不同数据集的样本质量差异显著"><a class="anchor" href="#1-不同数据集的样本质量差异显著">#</a> 🔍 1. <strong>不同数据集的样本质量差异显著</strong></h3><ul><li><p>数据集中细胞总数的多少主要受限于实验条件（例如样本处理能力、试剂成本、设备通量），尽管随着技术进步总量有增长趋势（见 Fig. 2b）。</p></li><li><p>常用的质量控制指标包括：</p><ul><li>每个细胞的总 <strong>UMI 数（unique molecular identifier）</strong></li><li>每个细胞检测到的 <strong>基因数量</strong></li></ul><blockquote><p>👉 这些数值反映了测序的 “有效信息量”，在后续数据分析中用于筛选低质量细胞。</p></blockquote></li></ul><hr><h3 id="2-平均测序深度影响低表达基因的检测"><a class="anchor" href="#2-平均测序深度影响低表达基因的检测">#</a> 📊 2. <strong>平均测序深度影响低表达基因的检测</strong></h3><ul><li><p><strong>平均测序深度 = 每个细胞平均测得的 reads 数量</strong></p></li><li><p>测序越深，就越可能捕获到<strong>低表达的基因</strong>，UMI 数也越高，进而：</p><ul><li>减少 “0 表达” 现象的误判（dropout）</li><li>提高低表达基因的可靠检测</li></ul></li><li><p>这意味着：深度高不仅能检测更多基因，还能<strong>降低因低丰度而产生的假阴性</strong></p></li></ul><hr><h3 id="3-测序深度-数据质量"><a class="anchor" href="#3-测序深度-数据质量">#</a> 🧪 3. <strong>测序深度 ≠ 数据质量</strong></h3><ul><li><p><strong>并非只要加深测序就能弥补实验缺陷</strong>：</p><ul><li>如果在实验过程中 mRNA 已经降解了，reads 再多也无法 “复原” 丢失的转录本。</li><li>数据恢复的上限受限于样本质量，常用 ** 测序饱和度（sequencing saturation）** 来估计。</li></ul></li></ul><hr><h3 id="️-4-这些质量差异会影响后续分析"><a class="anchor" href="#️-4-这些质量差异会影响后续分析">#</a> ⚠️ 4. <strong>这些质量差异会影响后续分析</strong></h3><ul><li><p>不同扰动条件下，如果测序深度或数据质量不一致，可能会影响：</p><ul><li>扰动间的差异是否能被识别（distinguishability）</li><li>聚类、差异表达、轨迹分析等下游计算的效果（downstream analysis performance）</li></ul></li></ul><hr><p>这里介绍了 E-distance 这个概念，对于两组细胞，它描述了这两组细胞的分布的 distance。More precisely, it compares the mean pairwise distance of cells between two different perturbations to the mean pairwise distance of cells within each of the two distribution.</p><p><img data-src="https://amentirazblogpic.oss-cn-hangzhou.aliyuncs.com/blogpic/41592_2023_2144_Fig3_HTML.webp" alt=""></p><p>Fig (c) :This is in part caused by two-target perturbations using CRISPRa in that dataset: targeting the same gene with two single guides increases the chances of causing a considerable change in the transcript profile.</p><p>The E-distance can also be used to quantify similarity between different perturbations.<br>Fig (e): 观察 unperturbed 和其它的分布可以看出对于不同的 E-test 值可以明显区分出来。对于 CEBPA 和 KLF1+CEBPA 重叠的区域 is captured by a low E-distance:cells affected by these two perturbations are closer to each other than they are to unperturbed cells or to those from other perturbations.</p><p><img data-src="https://amentirazblogpic.oss-cn-hangzhou.aliyuncs.com/blogpic/41592_2023_2144_Fig4_HTML.webp" alt=""></p><p>这段话主要在说明：<strong>如何利用 E-distance（能量距离）分析扰动数据集中不同基因扰动之间的相似性，并验证其在免疫信号通路中的生物学一致性。</strong></p><hr><h3 id="具体讲了什么"><a class="anchor" href="#具体讲了什么">#</a> 📌 具体讲了什么？</h3><ol><li><p><strong>分析背景与数据集来源</strong></p><ul><li>研究选用了一个关于免疫抑制检查点的扰动数据集（见 Fig. 4），该数据集使用 <strong>CRISPR-Cas9</strong> 对多个调控 <strong>PD-L1 表达</strong> 的基因进行扰动（参考文献 [9]）。</li></ul></li><li><p><strong>方法：E-distance 计算扰动间的相似性</strong></p><ul><li>计算了所有基因扰动对之间的 <strong>两两 E-distance</strong>，即它们诱导的细胞状态差异有多大。</li><li>然后对这个距离矩阵进行了<strong>层次聚类（hierarchical clustering）</strong>，发现出现了两个显著不同的扰动子群（clusters）。</li></ul></li><li><p><strong>观察与解释：形成一个具有相似表型的基因群</strong></p><ul><li>其中一个扰动群体（IFNGR1、IFNGR2、JAK2、STAT1）之间的 E-distance 较小，说明它们产生了<strong>相似的细胞转录组响应</strong>。</li><li>而它们与 “未扰动对照组” 之间的 E-distance 较大，说明这些扰动引起了显著变化。</li></ul></li><li><p><strong>生物学一致性验证</strong></p><ul><li><p>这组基因（IFNGR1, IFNGR2, JAK2, STAT1）正好都位于 <strong>IFNγ（干扰素 -γ）信号通路</strong>中，属于同一个上游 / 下游调控链条：</p><ul><li>它们都在 <strong>IFNγ → IRF1 → PD-L1 表达</strong> 的通路中起作用（参考文献 [50]）。</li></ul></li><li><p>因此扰动这些基因会引发相似的表型改变，是<strong>功能一致性的体现</strong>。</p></li></ul></li></ol><hr><p><img data-src="https://amentirazblogpic.oss-cn-hangzhou.aliyuncs.com/blogpic/41592_2023_2144_Fig5_HTML.webp" alt=""></p><p>这段话讲的是：<strong>作者通过子采样（subsampling）实验系统评估了 E-distance 和 E-test 对样本数量变化的敏感性和鲁棒性（稳定性），并给出合理使用它们的建议。</strong></p><hr><h3 id="具体讲了什么-2"><a class="anchor" href="#具体讲了什么-2">#</a> 📌 具体讲了什么？</h3><h4 id="1-研究目的"><a class="anchor" href="#1-研究目的">#</a> ✅ 1. <strong>研究目的</strong></h4><ul><li>使用 scPerturb 提供的统一注释的单细胞扰动数据集，评估 <strong>E-distance（用于量化扰动差异）和 E-test（用于显著性检验）</strong> 对实验设计和数据处理参数的鲁棒性（尤其是细胞数目的影响）。</li></ul><h4 id="2-实验方法人为减少细胞数进行测试"><a class="anchor" href="#2-实验方法人为减少细胞数进行测试">#</a> ✅ 2. <strong>实验方法：人为减少细胞数进行测试</strong></h4><ul><li>人为<strong>下采样每个扰动条件中的细胞数</strong>，模拟 “小样本” 场景，观察这时 E-distance 和 E-test 分数如何变化。</li></ul><h4 id="3-偏差修正bias-correction"><a class="anchor" href="#3-偏差修正bias-correction">#</a> ✅ 3. <strong>偏差修正（bias correction）</strong></h4><ul><li>提出了一种<strong> E-distance 的偏差修正方法</strong>，以提高在小样本条件下的准确性（详见补充材料第 1 节）。</li></ul><h4 id="4-发现-1即使修正后e-distance-在细胞数减少时仍上升"><a class="anchor" href="#4-发现-1即使修正后e-distance-在细胞数减少时仍上升">#</a> ✅ 4. <strong>发现 1：即使修正后，E-distance 在细胞数减少时仍上升</strong></h4><ul><li><p>即使使用了偏差修正，在<strong>每个扰动下的细胞数减少时，E-distance 仍会人为变大</strong>，这说明：</p><ul><li><strong>计算 E-distance 前需要对每个扰动的细胞数进行标准化</strong>（例如统一抽取相同数量的细胞），否则可能引入比较偏差。</li><li>原因在于：<strong>PCA 在小样本情况下不能很好表示高维数据结构</strong>，从而扭曲了距离估计。</li></ul></li></ul><h4 id="5-发现-2显著性检验结果随着样本数下降而自然减少"><a class="anchor" href="#5-发现-2显著性检验结果随着样本数下降而自然减少">#</a> ✅ 5. <strong>发现 2：显著性检验结果随着样本数下降而自然减少</strong></h4><ul><li>当细胞数下降时，能被识别为显著扰动的基因数量减少，这是符合预期的：样本越少，统计功效越低。</li><li>同时观察到，并非所有数据集在 “全部细胞数” 时都达到了扰动检测的饱和状态（即：进一步增加细胞数可能还会发现新扰动）。</li></ul><h4 id="6-影响因素扰动强度与数据异质性"><a class="anchor" href="#6-影响因素扰动强度与数据异质性">#</a> ✅ 6. <strong>影响因素：扰动强度与数据异质性</strong></h4><ul><li><p>检测 “饱和点”（saturation point）取决于：</p><ul><li><strong>扰动的效应强度（effect size）</strong>：强效扰动易被发现，弱效扰动需要更多细胞。</li><li><strong>数据集的异质性（heterogeneity）</strong>：如果细胞彼此非常相似，则较少样本就足以捕捉扰动的平均效应。</li></ul></li></ul><hr><p>这段话主要讲的是：<strong>评估单位细胞内的测序深度（即 UMI 数量）对扰动分析结果的影响，并提出实际实验设计的参考建议。</strong></p><hr><h3 id="具体内容逐句解析"><a class="anchor" href="#具体内容逐句解析">#</a> 📌 具体内容逐句解析：</h3><h4 id="1-umi-数越多e-distance-越大"><a class="anchor" href="#1-umi-数越多e-distance-越大">#</a> ✅ 1. <strong>UMI 数越多，E-distance 越大</strong></h4><blockquote><p>“We subset the number of UMIs per cell, finding that E-distance increases as the UMI count per cell increases (Fig. 5c).”</p></blockquote><ul><li><p>作者控制变量，只改变每个细胞的 UMI 数（模拟不同测序深度）：</p><ul><li>结果发现：<strong>随着每个细胞的 UMI 数增加，E-distance（能量距离）也增加</strong>。</li><li>原因可能是：高 UMI 会暴露出更多细微的扰动效应，使得扰动组与对照组在表达空间中的距离拉大。</li></ul></li></ul><h4 id="2-但-e-test-的显著性结果在-~500-umi-就饱和了"><a class="anchor" href="#2-但-e-test-的显著性结果在-~500-umi-就饱和了">#</a> ✅ 2. <strong>但 E-test 的显著性结果在～500 UMI 就饱和了</strong></h4><blockquote><p>“The number of significant perturbations under the E-test, however, saturates at around 500 UMIs per cell...”</p></blockquote><ul><li><p>虽然 E-distance 随深度增加而变化，但用于判断扰动是否显著的 <strong>E-test（统计检验）</strong> 在 UMI 达到 500 左右时，显著性结果就趋于稳定（饱和）：</p><ul><li><strong>也就是说：即使 UMI 较少，统计显著的扰动大多数依然能被检测到</strong>。</li></ul></li></ul><h4 id="3-说明-e-test-更稳定-更实用"><a class="anchor" href="#3-说明-e-test-更稳定-更实用">#</a> ✅ 3. <strong>说明 E-test 更稳定、更实用</strong></h4><blockquote><p>“The stability of E-test results... demonstrates the necessity of the E-test... as the appropriate statistical measure...”</p></blockquote><ul><li><p><strong>E-test 比 E-distance 更稳健、不那么依赖测序深度</strong>，因为：</p><ul><li>它使用了 <strong>随机化对照（permutation control）</strong>，可以抵消一部分数据量波动带来的影响。</li><li>所以在实际分析中，<strong>E-test 是更适合用来判断扰动是否有效的统计工具</strong>。</li></ul></li></ul><h4 id="4-实验设计建议干货"><a class="anchor" href="#4-实验设计建议干货">#</a> ✅ 4. <strong>实验设计建议（干货！）</strong></h4><blockquote><p>“we suggest at least 200–500 cells per perturbation and an average of 1,000 UMIs per cell...”</p></blockquote><ul><li><p>综合分析后，作者给出一个<strong>推荐标准</strong>，用于指导单细胞扰动实验设计：</p><ul><li><strong>每个扰动条件下建议至少测序 200–500 个细胞</strong></li><li><strong>每个细胞建议测出平均 1,000 个 UMI</strong></li><li>这两个条件基本能保证对多数 “显著扰动” 的稳定检测。</li></ul></li></ul><h4 id="5-但最佳测序深度还要看任务"><a class="anchor" href="#5-但最佳测序深度还要看任务">#</a> ✅ 5. <strong>但最佳测序深度还要看任务</strong></h4><blockquote><p>“The optimal lower bounds... depend on downstream specific modeling tasks...”</p></blockquote><ul><li><p>注意：这个 “推荐标准” 是用于 “检测扰动是否显著” 的一般性指导；</p><ul><li>若下游任务更复杂（如预测建模、轨迹分析等），所需的测序深度和细胞数可能更高。</li></ul></li></ul><hr><h1 id="methods"><a class="anchor" href="#methods">#</a> Methods</h1><p>终于到方法的部分了。</p><h2 id="scatac-seq"><a class="anchor" href="#scatac-seq">#</a> scATAC-seq</h2><p>scATAC-seq is a biomolecular technique to assess chromatin accessibility in single cells.</p><p>这段话主要讲的是：<strong>作者如何处理 scATAC-seq 数据以提取标准化的特征，用于分析单细胞染色质可及性信息。</strong></p><hr><h3 id="逐句解释如下"><a class="anchor" href="#逐句解释如下">#</a> 📌 逐句解释如下：</h3><h4 id="1-技术背景"><a class="anchor" href="#1-技术背景">#</a> ✅ 1. 技术背景</h4><blockquote><p><em>“scATAC-seq is a biomolecular technique to assess chromatin accessibility in single cells.”</em></p></blockquote><ul><li><strong>scATAC-seq 是一种单细胞分辨率的技术</strong>，用于检测染色质的开放程度，即哪些 DNA 区域对转录因子等蛋白 “可及”（accessible）。</li></ul><hr><h4 id="2-数据输入格式"><a class="anchor" href="#2-数据输入格式">#</a> ✅ 2. 数据输入格式</h4><blockquote><p><em>“The starting point of our data processing pipeline is BED-like tabular fragment files...”</em></p></blockquote><ul><li>数据处理流程的起点是类似 <strong>BED 格式的片段文件</strong>（tab-delimited），每一行代表一个被测序捕获到的 <strong>ATAC 片段（fragment）</strong>，包含它所在的基因组位置和对应的<strong>细胞条形码（cell barcode）</strong>。</li></ul><hr><h4 id="3-数据处理目标"><a class="anchor" href="#3-数据处理目标">#</a> ✅ 3. 数据处理目标</h4><blockquote><p><em>“The goal... is to extract standardized features from this information.”</em></p></blockquote><ul><li>目标是从这些原始片段信息中提取出一系列<strong>标准化的、可用于分析的特征</strong>。</li></ul><hr><h4 id="4-提取的五类特征如下"><a class="anchor" href="#4-提取的五类特征如下">#</a> ✅ 4. 提取的五类特征如下：</h4><ol><li><p><strong>LSI Embeddings（低维嵌入）</strong></p><blockquote><p><em>“embeddings derived from latent-semantic-indexing (LSI) with 30 dimensions...”</em></p></blockquote><ul><li>使用 <strong>潜在语义索引（LSI）</strong> 方法将高维稀疏的 ATAC-seq 数据降维成每个细胞一个 <strong>30 维向量</strong>，适合稀疏性强的单细胞 ATAC 数据。</li></ul></li><li><p><strong>Gene Scores（基因开放性得分）</strong></p><blockquote><p><em>“gene scores that measure the chromatin accessibility around each gene...”</em></p></blockquote><ul><li>每个细胞对每个基因，计算其 <strong>转录起始位点（TSS）邻近区域的开放程度加权和</strong>，即基因周围的染色质可及性，距离越远的片段权重越低。</li></ul></li><li><p><strong>Peak–Barcode Matrix（峰 × 细胞矩阵）</strong></p><blockquote><p><em>“a peak–barcode matrix that quantifies the chromatin accessibility at consensus peaks...”</em></p></blockquote><ul><li>构建一个 <strong>细胞 × 共识峰（peaks）</strong> 的矩阵，用于表示各细胞在特定基因组区域（peaks）上的染色质开放强度。</li></ul></li><li><p><strong>chromVAR Scores（转录因子活性分数）</strong></p><blockquote><p><em>“chromVAR scores, which quantify the activity of a set of transcription factors...”</em></p></blockquote><ul><li>利用 <strong>chromVAR 工具</strong> 计算每个细胞中一组转录因子对应的 DNA footprint 区域的活性变化，用来反映转录因子的功能活性水平。</li></ul></li><li><p><strong>Marker Peaks（扰动标志峰）</strong></p><blockquote><p><em>“marker-peaks per perturbation target, which quantifies the differential regulation...”</em></p></blockquote><ul><li>对每种扰动条件，识别出<strong>差异化活跃的高变峰（highly variable peaks）</strong>，作为该扰动的 “标志性染色质调控区域”。</li></ul></li></ol><hr><h2 id="scrna-seq"><a class="anchor" href="#scrna-seq">#</a> scRNA-seq</h2><p>这段话讲的是：<strong>作者如何处理和标准化各个原始单细胞扰动数据集，以便进行统一分析。</strong></p><p>下面是逐句解释：</p><hr><h3 id="1-起始数据来源与处理方式"><a class="anchor" href="#1-起始数据来源与处理方式">#</a> 📌 1. <strong>起始数据来源与处理方式</strong></h3><blockquote><p><em>“Analysis was initiated using unfiltered, unnormalized cell-by-gene matrices as provided by source papers.”</em></p></blockquote><ul><li>分析从<strong>原始文献提供的 cell-by-gene 矩阵</strong>（即细胞 × 基因的表达矩阵）开始；</li><li>使用的是<strong>未经过滤或归一化的原始版本</strong>，确保保留最大的信息量，便于统一处理。</li></ul><hr><h3 id="2-针对缺少处理后的数据集"><a class="anchor" href="#2-针对缺少处理后的数据集">#</a> 📌 2. <strong>针对缺少处理后的数据集</strong></h3><blockquote><p><em>“For one dataset, preprocessed cell-by-gene matrices were unavailable...”</em></p></blockquote><ul><li>有一个数据集并未公开处理好的表达矩阵；</li><li>作者就<strong>按照原论文中描述的流程</strong>，<strong>使用原作者提供的代码</strong>自行进行预处理。</li></ul><hr><h3 id="3-细胞条形码barcode处理"><a class="anchor" href="#3-细胞条形码barcode处理">#</a> 📌 3. <strong>细胞条形码（barcode）处理</strong></h3><blockquote><p><em>“For datasets with cell barcodes, barcode assignments for cells were taken from the original paper when available...”</em></p></blockquote><ul><li><p>对于包含细胞条形码（即识别每个细胞身份）的数据集：</p><ul><li>如果原论文提供了 barcode 到细胞的映射，作者就<strong>直接采用原始提供的数据</strong>；</li><li>如果没提供，就<strong>按原论文方法部分自行分配 barcode（详见方法部分）</strong>。</li></ul></li></ul><hr><h3 id="4-处理同一细胞含多个引导序列guide的情况"><a class="anchor" href="#4-处理同一细胞含多个引导序列guide的情况">#</a> 📌 4. <strong>处理同一细胞含多个引导序列（guide）的情况</strong></h3><blockquote><p><em>“If multiple guides were assigned to the same cell...”</em></p></blockquote><ul><li><p>如果一个细胞中检测到多个干预（例如多个 CRISPR guide RNA）：</p><ul><li>作者将这些 guide 按照<strong>计数从高到低的顺序排列</strong>，记录在最终的分析对象中；</li><li>这样可以保留 “主要扰动” 的信息，同时也不完全丢失次要扰动。</li></ul></li></ul><hr><h3 id="5-代码开放性"><a class="anchor" href="#5-代码开放性">#</a> 📌 5. <strong>代码开放性</strong></h3><blockquote><p><em>“The code used to process each individual dataset... is available in our code repository.”</em></p></blockquote><ul><li>作者将所有数据处理过程的代码（包括 barcode 分配）<strong>公开在他们的代码仓库中</strong>，确保透明和可复现。</li></ul><hr><h2 id="e-distance"><a class="anchor" href="#e-distance">#</a> E-distance</h2><p>Let <span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>x</mi><mn>1</mn></msub><mo separator="true">,</mo><mi mathvariant="normal">.</mi><mi mathvariant="normal">.</mi><mi mathvariant="normal">.</mi><mo separator="true">,</mo><msub><mi>x</mi><mi>N</mi></msub><mo>∈</mo><msup><mi mathvariant="double-struck">R</mi><mi>d</mi></msup></mrow><annotation encoding="application/x-tex">x_1,...,x_N \in \R^d</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:.7335400000000001em;vertical-align:-.19444em"></span><span class="mord"><span class="mord mathnormal">x</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:.30110799999999993em"><span style="top:-2.5500000000000003em;margin-left:0;margin-right:.05em"><span class="pstrut" style="height:2.7em"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight">1</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:.15em"><span></span></span></span></span></span></span><span class="mpunct">,</span><span class="mspace" style="margin-right:.16666666666666666em"></span><span class="mord">.</span><span class="mord">.</span><span class="mord">.</span><span class="mpunct">,</span><span class="mspace" style="margin-right:.16666666666666666em"></span><span class="mord"><span class="mord mathnormal">x</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:.32833099999999993em"><span style="top:-2.5500000000000003em;margin-left:0;margin-right:.05em"><span class="pstrut" style="height:2.7em"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight" style="margin-right:.10903em">N</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:.15em"><span></span></span></span></span></span></span><span class="mspace" style="margin-right:.2777777777777778em"></span><span class="mrel">∈</span><span class="mspace" style="margin-right:.2777777777777778em"></span></span><span class="base"><span class="strut" style="height:.849108em;vertical-align:0"></span><span class="mord"><span class="mord"><span class="mord mathbb">R</span></span><span class="msupsub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:.849108em"><span style="top:-3.063em;margin-right:.05em"><span class="pstrut" style="height:2.7em"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight">d</span></span></span></span></span></span></span></span></span></span></span> and <span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>y</mi><mn>1</mn></msub><mo separator="true">,</mo><mi mathvariant="normal">.</mi><mi mathvariant="normal">.</mi><mi mathvariant="normal">.</mi><mo separator="true">,</mo><msub><mi>y</mi><mi>M</mi></msub><mo>∈</mo><msup><mi mathvariant="double-struck">R</mi><mi>d</mi></msup></mrow><annotation encoding="application/x-tex">y_1,...,y_M \in \R^d</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:.7335400000000001em;vertical-align:-.19444em"></span><span class="mord"><span class="mord mathnormal" style="margin-right:.03588em">y</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:.30110799999999993em"><span style="top:-2.5500000000000003em;margin-left:-.03588em;margin-right:.05em"><span class="pstrut" style="height:2.7em"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight">1</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:.15em"><span></span></span></span></span></span></span><span class="mpunct">,</span><span class="mspace" style="margin-right:.16666666666666666em"></span><span class="mord">.</span><span class="mord">.</span><span class="mord">.</span><span class="mpunct">,</span><span class="mspace" style="margin-right:.16666666666666666em"></span><span class="mord"><span class="mord mathnormal" style="margin-right:.03588em">y</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:.32833099999999993em"><span style="top:-2.5500000000000003em;margin-left:-.03588em;margin-right:.05em"><span class="pstrut" style="height:2.7em"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight" style="margin-right:.10903em">M</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:.15em"><span></span></span></span></span></span></span><span class="mspace" style="margin-right:.2777777777777778em"></span><span class="mrel">∈</span><span class="mspace" style="margin-right:.2777777777777778em"></span></span><span class="base"><span class="strut" style="height:.849108em;vertical-align:0"></span><span class="mord"><span class="mord"><span class="mord mathbb">R</span></span><span class="msupsub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height:.849108em"><span style="top:-3.063em;margin-right:.05em"><span class="pstrut" style="height:2.7em"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight">d</span></span></span></span></span></span></span></span></span></span></span> be samples from two distributions <span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>X</mi><mo separator="true">,</mo><mi>Y</mi></mrow><annotation encoding="application/x-tex">X,Y</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:.8777699999999999em;vertical-align:-.19444em"></span><span class="mord mathnormal" style="margin-right:.07847em">X</span><span class="mpunct">,</span><span class="mspace" style="margin-right:.16666666666666666em"></span><span class="mord mathnormal" style="margin-right:.22222em">Y</span></span></span></span>, corresponding to two sets of <span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>N</mi></mrow><annotation encoding="application/x-tex">N</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:.68333em;vertical-align:0"></span><span class="mord mathnormal" style="margin-right:.10903em">N</span></span></span></span> and <span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>M</mi></mrow><annotation encoding="application/x-tex">M</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:.68333em;vertical-align:0"></span><span class="mord mathnormal" style="margin-right:.10903em">M</span></span></span></span> cells, respectively.</p><p>We define :</p>\begin{equation} \delta_{XY} = \frac{1}{NM}\sum^N_{i=1}\sum^M_{j=1}{||x_i-y_j||} \end{equation} \begin{equation} \sigma_X = \frac{1}{N^2}\sum^N_{i=1}\sum^M_{j=1}{||x_i-y_j||} \end{equation}<p>and <span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>σ</mi><mi>Y</mi></msub></mrow><annotation encoding="application/x-tex">\sigma_Y</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:.58056em;vertical-align:-.15em"></span><span class="mord"><span class="mord mathnormal" style="margin-right:.03588em">σ</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:.32833099999999993em"><span style="top:-2.5500000000000003em;margin-left:-.03588em;margin-right:.05em"><span class="pstrut" style="height:2.7em"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight" style="margin-right:.22222em">Y</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:.15em"><span></span></span></span></span></span></span></span></span></span> defined accordingly.</p><p>Intuitively, <span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>δ</mi><mrow><mi>X</mi><mi>Y</mi></mrow></msub></mrow><annotation encoding="application/x-tex">\delta_{XY}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:.84444em;vertical-align:-.15em"></span><span class="mord"><span class="mord mathnormal" style="margin-right:.03785em">δ</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:.32833099999999993em"><span style="top:-2.5500000000000003em;margin-left:-.03785em;margin-right:.05em"><span class="pstrut" style="height:2.7em"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight" style="margin-right:.07847em">X</span><span class="mord mathnormal mtight" style="margin-right:.22222em">Y</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:.15em"><span></span></span></span></span></span></span></span></span></span> is the mean distance between cells from the two distributions, while <span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>σ</mi><mi>X</mi></msub></mrow><annotation encoding="application/x-tex">\sigma_X</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:.58056em;vertical-align:-.15em"></span><span class="mord"><span class="mord mathnormal" style="margin-right:.03588em">σ</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:.32833099999999993em"><span style="top:-2.5500000000000003em;margin-left:-.03588em;margin-right:.05em"><span class="pstrut" style="height:2.7em"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight" style="margin-right:.07847em">X</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:.15em"><span></span></span></span></span></span></span></span></span></span> describes the mean distance between a cell from <span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>X</mi></mrow><annotation encoding="application/x-tex">X</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:.68333em;vertical-align:0"></span><span class="mord mathnormal" style="margin-right:.07847em">X</span></span></span></span> to another cell from <span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>X</mi></mrow><annotation encoding="application/x-tex">X</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:.68333em;vertical-align:0"></span><span class="mord mathnormal" style="margin-right:.07847em">X</span></span></span></span>. The energy distance between <span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>X</mi></mrow><annotation encoding="application/x-tex">X</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:.68333em;vertical-align:0"></span><span class="mord mathnormal" style="margin-right:.07847em">X</span></span></span></span> and <span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>Y</mi></mrow><annotation encoding="application/x-tex">Y</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:.68333em;vertical-align:0"></span><span class="mord mathnormal" style="margin-right:.22222em">Y</span></span></span></span> is defined as:</p>\begin{equation} E(X,Y):=2\delta_{XY}-\sigma_X-\sigma_Y \end{equation}<p>For the bias-corrected energy distance, we define</p>\begin{equation} \sigma_X = \frac{1}{N(N-1)}\sum^N_{i=1}\sum^M_{j=1}{||x_i-y_j||} \end{equation}<p>这比原来的 $\frac<ruby>1}{N<rp>(</rp><rt>2</rt><rp>)</rp></ruby>，因为：</p><p>原公式包含了自己和自己的距离<span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi mathvariant="normal">∣</mi><mi mathvariant="normal">∣</mi><msub><mi>x</mi><mi>i</mi></msub><mo>−</mo><msub><mi>x</mi><mi>i</mi></msub><mi mathvariant="normal">∣</mi><mi mathvariant="normal">∣</mi><mo>=</mo><mn>0</mn></mrow><annotation encoding="application/x-tex">||x_i-x_i||=0</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:1em;vertical-align:-.25em"></span><span class="mord">∣</span><span class="mord">∣</span><span class="mord"><span class="mord mathnormal">x</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:.31166399999999994em"><span style="top:-2.5500000000000003em;margin-left:0;margin-right:.05em"><span class="pstrut" style="height:2.7em"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight">i</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:.15em"><span></span></span></span></span></span></span><span class="mspace" style="margin-right:.2222222222222222em"></span><span class="mbin">−</span><span class="mspace" style="margin-right:.2222222222222222em"></span></span><span class="base"><span class="strut" style="height:1em;vertical-align:-.25em"></span><span class="mord"><span class="mord mathnormal">x</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:.31166399999999994em"><span style="top:-2.5500000000000003em;margin-left:0;margin-right:.05em"><span class="pstrut" style="height:2.7em"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mathnormal mtight">i</span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:.15em"><span></span></span></span></span></span></span><span class="mord">∣</span><span class="mord">∣</span><span class="mspace" style="margin-right:.2777777777777778em"></span><span class="mrel">=</span><span class="mspace" style="margin-right:.2777777777777778em"></span></span><span class="base"><span class="strut" style="height:.64444em;vertical-align:0"></span><span class="mord">0</span></span></span></span></p><p>而修正版本只考虑不同的样本对，避免了系统性低估组内变异性；</p><p>修正后更适用于小样本（如每个扰动只有几十或几百个细胞）的情况。</p><h2 id="e-test-calculation"><a class="anchor" href="#e-test-calculation">#</a> E-test calculation</h2><p>E 检验（E-test）作为一种 Monte Carlo 置换检验，以能量距离（E-distance）作为检验统计量进行实施。对于每个数据集中的每一个扰动条件，我们将该扰动组的细胞与未扰动对照组的细胞合并，然后随机打乱扰动标签，将细胞重新分为两组，并计算这两组之间的 E-distance。我们将这一过程重复 10,000 次。在这 10,000 次中，打乱标签后计算得到的 E-distance 大于原始未打乱标签所得到的 E-distance 的次数，除以 10,000，即得到了对应的 P 值，该值表示一个 单边检验（one-sided test） 的显著性水平。我们为本资源中几乎所有的数据集都报告了该检验的结果（见补充表 3）。</p><p>对于每个数据集中的多重假设检验，我们采用 Holm–Sidak 方法 进行多重比较校正。</p><div class="tags"><a href="/tags/%E7%94%9F%E7%89%A9/" rel="tag"><i class="ic i-tag"></i> 生物</a> <a href="/tags/%E5%AD%A6%E4%B9%A0%E7%AC%94%E8%AE%B0%E7%B1%BB/" rel="tag"><i class="ic i-tag"></i> 学习笔记类</a> <a href="/tags/%E7%AE%97%E6%B3%95/" rel="tag"><i class="ic i-tag"></i> 算法</a></div></div><footer><div class="meta"><span class="item"><span class="icon"><i class="ic i-calendar-check"></i> </span><span class="text">更新于</span> <time title="修改时间：2025-05-11 15:48:08" itemprop="dateModified" datetime="2025-05-11T15:48:08+08:00">2025-05-11</time> </span><span id="article/scPerturb/" class="item leancloud_visitors" data-flag-title="scPerturb" title="阅读次数"><span class="icon"><i class="ic i-eye"></i> </span><span class="text">阅读次数</span> <span class="leancloud-visitors-count"></span> <span class="text">次</span></span></div><div id="copyright"><ul><li class="author"><strong>本文作者： </strong>Lemon Sour <i class="ic i-at"><em>@</em></i></li><li class="link"><strong>本文链接：</strong> <a href="http://amentiraz.github.io/article/scPerturb/" title="scPerturb">http://amentiraz.github.io/article/scPerturb/</a></li><li class="license"><strong>版权声明： </strong>本站所有文章除特别声明外，均采用 <span class="exturl" data-url="aHR0cHM6Ly9jcmVhdGl2ZWNvbW1vbnMub3JnL2xpY2Vuc2VzL2J5LW5jLXNhLzQuMC96aC1DTg=="><i class="ic i-creative-commons"><em>(CC)</em></i>BY-NC-SA</span> 许可协议。转载请注明出处！</li></ul></div></footer></article></div><div class="post-nav"><div class="item left"><a href="/other/2025-4%E5%B7%A5%E4%BD%9C%E6%80%BB%E7%BB%93/" itemprop="url" rel="prev" data-background-image="https:&#x2F;&#x2F;amentirazblogpic.oss-cn-hangzhou.aliyuncs.com&#x2F;blogpic&#x2F;%E5%BE%AE%E4%BF%A1%E5%9B%BE%E7%89%87_20240925162815.png" title="2025-4工作总结"><span class="type">上一篇</span> <span class="category"><i class="ic i-flag"></i> 工作总结</span><h3>2025-4工作总结</h3></a></div><div class="item right"><a href="/life/%E5%8F%A5%E5%AD%90%E6%91%98%E5%BD%95/" itemprop="url" rel="next" data-background-image="https:&#x2F;&#x2F;amentirazblogpic.oss-cn-hangzhou.aliyuncs.com&#x2F;blogpic&#x2F;%E5%BE%AE%E4%BF%A1%E5%9B%BE%E7%89%87_20240925162929.png" title="句子摘录"><span class="type">下一篇</span> <span class="category"><i class="ic i-flag"></i> 摘录</span><h3>句子摘录</h3></a></div></div><div class="wrap" id="comments"></div></div><div id="sidebar"><div class="inner"><div class="panels"><div class="inner"><div class="contents panel pjax" data-title="文章目录"><ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#abstract"><span class="toc-number">1.</span> <span class="toc-text">Abstract</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#introduction"><span class="toc-number">2.</span> <span class="toc-text">Introduction</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#1-single-cell-rna-sequencing-scrna-seq"><span class="toc-number">2.1.</span> <span class="toc-text">1. Single-cell RNA sequencing (scRNA-seq)</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#%E5%9F%BA%E6%9C%AC%E5%8E%9F%E7%90%86"><span class="toc-number">2.1.1.</span> <span class="toc-text">基本原理</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#%E5%AE%9E%E9%AA%8C%E6%B5%81%E7%A8%8B"><span class="toc-number">2.1.2.</span> <span class="toc-text">实验流程</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#%E4%BC%98%E5%8A%BF"><span class="toc-number">2.1.3.</span> <span class="toc-text">优势</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#%E5%B1%80%E9%99%90"><span class="toc-number">2.1.4.</span> <span class="toc-text">局限</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#%E5%BA%94%E7%94%A8%E5%9C%BA%E6%99%AF"><span class="toc-number">2.1.5.</span> <span class="toc-text">应用场景</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#2-cite-seq"><span class="toc-number">2.2.</span> <span class="toc-text">2. CITE-seq</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#%E5%9F%BA%E6%9C%AC%E5%8E%9F%E7%90%86-2"><span class="toc-number">2.2.1.</span> <span class="toc-text">基本原理</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#%E5%AE%9E%E9%AA%8C%E6%B5%81%E7%A8%8B-2"><span class="toc-number">2.2.2.</span> <span class="toc-text">实验流程</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#%E4%BC%98%E5%8A%BF-2"><span class="toc-number">2.2.3.</span> <span class="toc-text">优势</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#%E5%B1%80%E9%99%90-2"><span class="toc-number">2.2.4.</span> <span class="toc-text">局限</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#%E5%BA%94%E7%94%A8%E5%9C%BA%E6%99%AF-2"><span class="toc-number">2.2.5.</span> <span class="toc-text">应用场景</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#3-single-cell-atac-sequencing-scatac-seq"><span class="toc-number">2.3.</span> <span class="toc-text">3. Single-cell ATAC sequencing (scATAC-seq)</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#%E5%9F%BA%E6%9C%AC%E5%8E%9F%E7%90%86-3"><span class="toc-number">2.3.1.</span> <span class="toc-text">基本原理</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#%E5%AE%9E%E9%AA%8C%E6%B5%81%E7%A8%8B-3"><span class="toc-number">2.3.2.</span> <span class="toc-text">实验流程</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#%E4%BC%98%E5%8A%BF-3"><span class="toc-number">2.3.3.</span> <span class="toc-text">优势</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#%E5%B1%80%E9%99%90-3"><span class="toc-number">2.3.4.</span> <span class="toc-text">局限</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#%E5%BA%94%E7%94%A8%E5%9C%BA%E6%99%AF-3"><span class="toc-number">2.3.5.</span> <span class="toc-text">应用场景</span></a></li></ol></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#results"><span class="toc-number">3.</span> <span class="toc-text">Results</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#1-%E4%B8%8D%E5%90%8C%E6%95%B0%E6%8D%AE%E9%9B%86%E7%9A%84%E6%A0%B7%E6%9C%AC%E8%B4%A8%E9%87%8F%E5%B7%AE%E5%BC%82%E6%98%BE%E8%91%97"><span class="toc-number">3.0.1.</span> <span class="toc-text">🔍 1. 不同数据集的样本质量差异显著</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#2-%E5%B9%B3%E5%9D%87%E6%B5%8B%E5%BA%8F%E6%B7%B1%E5%BA%A6%E5%BD%B1%E5%93%8D%E4%BD%8E%E8%A1%A8%E8%BE%BE%E5%9F%BA%E5%9B%A0%E7%9A%84%E6%A3%80%E6%B5%8B"><span class="toc-number">3.0.2.</span> <span class="toc-text">📊 2. 平均测序深度影响低表达基因的检测</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#3-%E6%B5%8B%E5%BA%8F%E6%B7%B1%E5%BA%A6-%E6%95%B0%E6%8D%AE%E8%B4%A8%E9%87%8F"><span class="toc-number">3.0.3.</span> <span class="toc-text">🧪 3. 测序深度 ≠ 数据质量</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#%EF%B8%8F-4-%E8%BF%99%E4%BA%9B%E8%B4%A8%E9%87%8F%E5%B7%AE%E5%BC%82%E4%BC%9A%E5%BD%B1%E5%93%8D%E5%90%8E%E7%BB%AD%E5%88%86%E6%9E%90"><span class="toc-number">3.0.4.</span> <span class="toc-text">⚠️ 4. 这些质量差异会影响后续分析</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#%E5%85%B7%E4%BD%93%E8%AE%B2%E4%BA%86%E4%BB%80%E4%B9%88"><span class="toc-number">3.0.5.</span> <span class="toc-text">📌 具体讲了什么？</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#%E5%85%B7%E4%BD%93%E8%AE%B2%E4%BA%86%E4%BB%80%E4%B9%88-2"><span class="toc-number">3.0.6.</span> <span class="toc-text">📌 具体讲了什么？</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#1-%E7%A0%94%E7%A9%B6%E7%9B%AE%E7%9A%84"><span class="toc-number">3.0.6.1.</span> <span class="toc-text">✅ 1. 研究目的</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#2-%E5%AE%9E%E9%AA%8C%E6%96%B9%E6%B3%95%E4%BA%BA%E4%B8%BA%E5%87%8F%E5%B0%91%E7%BB%86%E8%83%9E%E6%95%B0%E8%BF%9B%E8%A1%8C%E6%B5%8B%E8%AF%95"><span class="toc-number">3.0.6.2.</span> <span class="toc-text">✅ 2. 实验方法：人为减少细胞数进行测试</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#3-%E5%81%8F%E5%B7%AE%E4%BF%AE%E6%AD%A3bias-correction"><span class="toc-number">3.0.6.3.</span> <span class="toc-text">✅ 3. 偏差修正（bias correction）</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#4-%E5%8F%91%E7%8E%B0-1%E5%8D%B3%E4%BD%BF%E4%BF%AE%E6%AD%A3%E5%90%8Ee-distance-%E5%9C%A8%E7%BB%86%E8%83%9E%E6%95%B0%E5%87%8F%E5%B0%91%E6%97%B6%E4%BB%8D%E4%B8%8A%E5%8D%87"><span class="toc-number">3.0.6.4.</span> <span class="toc-text">✅ 4. 发现 1：即使修正后，E-distance 在细胞数减少时仍上升</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#5-%E5%8F%91%E7%8E%B0-2%E6%98%BE%E8%91%97%E6%80%A7%E6%A3%80%E9%AA%8C%E7%BB%93%E6%9E%9C%E9%9A%8F%E7%9D%80%E6%A0%B7%E6%9C%AC%E6%95%B0%E4%B8%8B%E9%99%8D%E8%80%8C%E8%87%AA%E7%84%B6%E5%87%8F%E5%B0%91"><span class="toc-number">3.0.6.5.</span> <span class="toc-text">✅ 5. 发现 2：显著性检验结果随着样本数下降而自然减少</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#6-%E5%BD%B1%E5%93%8D%E5%9B%A0%E7%B4%A0%E6%89%B0%E5%8A%A8%E5%BC%BA%E5%BA%A6%E4%B8%8E%E6%95%B0%E6%8D%AE%E5%BC%82%E8%B4%A8%E6%80%A7"><span class="toc-number">3.0.6.6.</span> <span class="toc-text">✅ 6. 影响因素：扰动强度与数据异质性</span></a></li></ol></li><li class="toc-item toc-level-3"><a class="toc-link" href="#%E5%85%B7%E4%BD%93%E5%86%85%E5%AE%B9%E9%80%90%E5%8F%A5%E8%A7%A3%E6%9E%90"><span class="toc-number">3.0.7.</span> <span class="toc-text">📌 具体内容逐句解析：</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#1-umi-%E6%95%B0%E8%B6%8A%E5%A4%9Ae-distance-%E8%B6%8A%E5%A4%A7"><span class="toc-number">3.0.7.1.</span> <span class="toc-text">✅ 1. UMI 数越多，E-distance 越大</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#2-%E4%BD%86-e-test-%E7%9A%84%E6%98%BE%E8%91%97%E6%80%A7%E7%BB%93%E6%9E%9C%E5%9C%A8-~500-umi-%E5%B0%B1%E9%A5%B1%E5%92%8C%E4%BA%86"><span class="toc-number">3.0.7.2.</span> <span class="toc-text">✅ 2. 但 E-test 的显著性结果在～500 UMI 就饱和了</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#3-%E8%AF%B4%E6%98%8E-e-test-%E6%9B%B4%E7%A8%B3%E5%AE%9A-%E6%9B%B4%E5%AE%9E%E7%94%A8"><span class="toc-number">3.0.7.3.</span> <span class="toc-text">✅ 3. 说明 E-test 更稳定、更实用</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#4-%E5%AE%9E%E9%AA%8C%E8%AE%BE%E8%AE%A1%E5%BB%BA%E8%AE%AE%E5%B9%B2%E8%B4%A7"><span class="toc-number">3.0.7.4.</span> <span class="toc-text">✅ 4. 实验设计建议（干货！）</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#5-%E4%BD%86%E6%9C%80%E4%BD%B3%E6%B5%8B%E5%BA%8F%E6%B7%B1%E5%BA%A6%E8%BF%98%E8%A6%81%E7%9C%8B%E4%BB%BB%E5%8A%A1"><span class="toc-number">3.0.7.5.</span> <span class="toc-text">✅ 5. 但最佳测序深度还要看任务</span></a></li></ol></li></ol></li></ol><li class="toc-item toc-level-1"><a class="toc-link" href="#methods"><span class="toc-number">4.</span> <span class="toc-text">Methods</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#scatac-seq"><span class="toc-number">4.1.</span> <span class="toc-text">scATAC-seq</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#%E9%80%90%E5%8F%A5%E8%A7%A3%E9%87%8A%E5%A6%82%E4%B8%8B"><span class="toc-number">4.1.1.</span> <span class="toc-text">📌 逐句解释如下：</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#1-%E6%8A%80%E6%9C%AF%E8%83%8C%E6%99%AF"><span class="toc-number">4.1.1.1.</span> <span class="toc-text">✅ 1. 技术背景</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#2-%E6%95%B0%E6%8D%AE%E8%BE%93%E5%85%A5%E6%A0%BC%E5%BC%8F"><span class="toc-number">4.1.1.2.</span> <span class="toc-text">✅ 2. 数据输入格式</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#3-%E6%95%B0%E6%8D%AE%E5%A4%84%E7%90%86%E7%9B%AE%E6%A0%87"><span class="toc-number">4.1.1.3.</span> <span class="toc-text">✅ 3. 数据处理目标</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#4-%E6%8F%90%E5%8F%96%E7%9A%84%E4%BA%94%E7%B1%BB%E7%89%B9%E5%BE%81%E5%A6%82%E4%B8%8B"><span class="toc-number">4.1.1.4.</span> <span class="toc-text">✅ 4. 提取的五类特征如下：</span></a></li></ol></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#scrna-seq"><span class="toc-number">4.2.</span> <span class="toc-text">scRNA-seq</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#1-%E8%B5%B7%E5%A7%8B%E6%95%B0%E6%8D%AE%E6%9D%A5%E6%BA%90%E4%B8%8E%E5%A4%84%E7%90%86%E6%96%B9%E5%BC%8F"><span class="toc-number">4.2.1.</span> <span class="toc-text">📌 1. 起始数据来源与处理方式</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#2-%E9%92%88%E5%AF%B9%E7%BC%BA%E5%B0%91%E5%A4%84%E7%90%86%E5%90%8E%E7%9A%84%E6%95%B0%E6%8D%AE%E9%9B%86"><span class="toc-number">4.2.2.</span> <span class="toc-text">📌 2. 针对缺少处理后的数据集</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#3-%E7%BB%86%E8%83%9E%E6%9D%A1%E5%BD%A2%E7%A0%81barcode%E5%A4%84%E7%90%86"><span class="toc-number">4.2.3.</span> <span class="toc-text">📌 3. 细胞条形码（barcode）处理</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#4-%E5%A4%84%E7%90%86%E5%90%8C%E4%B8%80%E7%BB%86%E8%83%9E%E5%90%AB%E5%A4%9A%E4%B8%AA%E5%BC%95%E5%AF%BC%E5%BA%8F%E5%88%97guide%E7%9A%84%E6%83%85%E5%86%B5"><span class="toc-number">4.2.4.</span> <span class="toc-text">📌 4. 处理同一细胞含多个引导序列（guide）的情况</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#5-%E4%BB%A3%E7%A0%81%E5%BC%80%E6%94%BE%E6%80%A7"><span class="toc-number">4.2.5.</span> <span class="toc-text">📌 5. 代码开放性</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#e-distance"><span class="toc-number">4.3.</span> <span class="toc-text">E-distance</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#e-test-calculation"><span class="toc-number">4.4.</span> <span class="toc-text">E-test calculation</span></a></li></ol></li></div><div class="related panel pjax" data-title="系列文章"><ul><li><a href="/article/scRNA%E8%AE%BA%E6%96%87%E7%AC%94%E8%AE%B0/" rel="bookmark" title="scRNA论文笔记">scRNA论文笔记</a></li><li><a href="/article/What-is-A-Cell-Type/" rel="bookmark" title="What_is_A_Cell_Type">What_is_A_Cell_Type</a></li><li><a href="/article/Cell-Review-What-is-a-cell-type-and-how-to-define-it/" rel="bookmark" title="Cell_Review_What_is_a_cell_type_and_how_to_define_it">Cell_Review_What_is_a_cell_type_and_how_to_define_it</a></li><li><a href="/article/Benchmarking-spatial-claustering-methods-with-spatially-resolved-transcriptomic-data/" rel="bookmark" title="Benchmarking-spatial-claustering-methods-with-spatially-resolved-transcriptomic-data">Benchmarking-spatial-claustering-methods-with-spatially-resolved-transcriptomic-data</a></li><li><a href="/article/%E8%AE%BA%E6%96%87ppt1/" rel="bookmark" title="论文ppt1">论文ppt1</a></li><li><a href="/article/Matching-Anything-by-Segmenting-Anything/" rel="bookmark" title="Matching_Anything_by_Segmenting_Anything">Matching_Anything_by_Segmenting_Anything</a></li><li><a href="/article/SpaGCN/" rel="bookmark" title="SpaGCN">SpaGCN</a></li><li><a href="/article/STAGATE/" rel="bookmark" title="STAGATE">STAGATE</a></li><li><a href="/article/MENDER/" rel="bookmark" title="MENDER">MENDER</a></li><li><a href="/article/scanpy%E6%95%B0%E6%8D%AE%E4%BD%BF%E7%94%A8%E8%AE%B0%E5%BD%95/" rel="bookmark" title="scanpy数据使用笔记">scanpy数据使用笔记</a></li><li><a href="/article/BayesSpace/" rel="bookmark" title="BayesSpace">BayesSpace</a></li><li><a href="/article/%E6%95%B0%E6%8D%AE%E9%9B%86%E8%AE%B0%E5%BD%95/" rel="bookmark" title="数据集记录">数据集记录</a></li><li><a href="/article/DeepST/" rel="bookmark" title="DeepST">DeepST</a></li><li><a href="/article/EnSDD/" rel="bookmark" title="EnSDD">EnSDD</a></li><li><a href="/article/domain%E5%86%85%E5%AE%B9%E7%9A%84%E6%80%BB%E7%BB%93/" rel="bookmark" title="对spatial domain内容的总结">对spatial domain内容的总结</a></li><li><a href="/article/%E6%AF%95%E4%B8%9A%E8%AE%BE%E8%AE%A1%E6%96%87%E6%A1%A3%E6%B1%87%E6%80%BB/" rel="bookmark" title="毕业设计文档汇总">毕业设计文档汇总</a></li><li class="active"><a href="/article/scPerturb/" rel="bookmark" title="scPerturb">scPerturb</a></li><li><a href="/GEARS/" rel="bookmark" title="GEARS">GEARS</a></li></ul></div><div class="overview panel" data-title="站点概览"><div class="author" itemprop="author" itemscope itemtype="http://schema.org/Person"><img class="image" itemprop="image" alt="Lemon Sour" data-src="/images/avatar.jpg"><p class="name" itemprop="name">Lemon Sour</p><div class="description" itemprop="description"></div></div><nav class="state"><div class="item posts"><a href="/archives/"><span class="count">85</span> <span class="name">文章</span></a></div><div class="item categories"><a href="/categories/"><span class="count">26</span> <span class="name">分类</span></a></div><div class="item tags"><a href="/tags/"><span class="count">57</span> <span class="name">标签</span></a></div></nav><div class="social"><span class="exturl item github" data-url="aHR0cHM6Ly9naXRodWIuY29tL0FtZW50aXJheg==" title="https:&#x2F;&#x2F;github.com&#x2F;Amentiraz"><i class="ic i-github"></i></span> <span class="exturl item zhihu" data-url="aHR0cHM6Ly93d3cuemhpaHUuY29tL3Blb3BsZS9kc2ZseS04" title="https:&#x2F;&#x2F;www.zhihu.com&#x2F;people&#x2F;dsfly-8"><i class="ic i-zhihu"></i></span> <span class="exturl item music" data-url="aHR0cHM6Ly9tdXNpYy4xNjMuY29tLyMvdXNlci9ob21lP2lkPTE1MTc2ODUzMzM=" title="https:&#x2F;&#x2F;music.163.com&#x2F;#&#x2F;user&#x2F;home?id&#x3D;1517685333"><i class="ic i-cloud-music"></i></span> <span class="exturl item email" data-url="bWFpbHRvOnZpb2xlbW9uQDE2My5jb20=" title="mailto:violemon@163.com"><i class="ic i-envelope"></i></span> <span class="exturl item bangumi" data-url="aHR0cHM6Ly9iYW5ndW1pLnR2L2FuaW1lL2xpc3QvNjY4MDE2" title="https:&#x2F;&#x2F;bangumi.tv&#x2F;anime&#x2F;list&#x2F;668016"><i class="ic i-bilibili"></i></span></div><ul class="menu"><li class="item"><a href="/" rel="section"><i class="ic i-home"></i>首页</a></li><li class="item dropdown"><a href="javascript:void(0);"><i class="ic i-magic"></i>关于</a><ul class="submenu"><li class="item"><a href="/author/" rel="section"><i class="ic i-user"></i>本人</a></li></ul></li><li class="item dropdown"><a href="javascript:void(0);"><i class="ic i-feather"></i>文章</a><ul class="submenu"><li class="item"><a href="/archives/" rel="section"><i class="ic i-list-alt"></i>归档</a></li><li class="item"><a href="/categories/" rel="section"><i class="ic i-th"></i>分类</a></li><li class="item"><a href="/tags/" rel="section"><i class="ic i-tags"></i>标签</a></li></ul></li><li class="item dropdown"><a href="javascript:void(0);"><i class="ic i-cloud"></i>其它</a><ul class="submenu"><li class="item"><a href="/music/" rel="section"><i class="ic i-music"></i>音乐区</a></li><li class="item"><a href="/friends/" rel="section"><i class="ic i-heart"></i>朋友</a></li></ul></li></ul></div></div></div><ul id="quick"><li class="prev pjax"><a href="/other/2025-4%E5%B7%A5%E4%BD%9C%E6%80%BB%E7%BB%93/" rel="prev" title="上一篇"><i class="ic i-chevron-left"></i></a></li><li class="up"><i class="ic i-arrow-up"></i></li><li class="down"><i class="ic i-arrow-down"></i></li><li class="next pjax"><a href="/life/%E5%8F%A5%E5%AD%90%E6%91%98%E5%BD%95/" rel="next" title="下一篇"><i class="ic i-chevron-right"></i></a></li><li class="percent"></li></ul></div></div><div class="dimmer"></div></div></main><footer id="footer"><div class="inner"><div class="widgets"><div class="rpost pjax"><h2>随机文章</h2><ul><li class="item"><div class="breadcrumb"><a href="/categories/life/" title="分类于 生活">生活</a> <i class="ic i-angle-right"></i> <a href="/categories/life/%E6%80%9D%E8%80%83/" title="分类于 思考">思考</a></div><span><a href="/life/%E5%85%B3%E4%BA%8E%E9%9D%A2%E5%AD%94%E7%9A%84%E5%90%AB%E4%B9%89/" title="关于面孔的含义">关于面孔的含义</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/article/" title="分类于 论文">论文</a></div><span><a href="/article/EnSDD/" title="EnSDD">EnSDD</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/code/" title="分类于 代码">代码</a> <i class="ic i-angle-right"></i> <a href="/categories/code/%E7%BC%96%E7%A8%8B%E8%AF%AD%E8%A8%80/" title="分类于 编程语言">编程语言</a></div><span><a href="/code/c%E5%A4%8D%E4%B9%A0/" title="c复习">c复习</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/life/" title="分类于 生活">生活</a> <i class="ic i-angle-right"></i> <a href="/categories/life/%E6%80%9D%E8%80%83/" title="分类于 思考">思考</a></div><span><a href="/life/%E7%B2%BE%E7%A5%9E%E5%88%86%E6%9E%90%E4%B8%AD%E7%9A%84%E5%80%92%E9%94%99/" title="精神分析中的倒错">精神分析中的倒错</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/article/" title="分类于 论文">论文</a></div><span><a href="/article/%E6%AF%95%E4%B8%9A%E8%AE%BE%E8%AE%A1%E6%96%87%E6%A1%A3%E6%B1%87%E6%80%BB/" title="毕业设计文档汇总">毕业设计文档汇总</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/music/" title="分类于 音乐">音乐</a> <i class="ic i-angle-right"></i> <a href="/categories/music/%E4%B9%90%E7%90%86/" title="分类于 乐理">乐理</a></div><span><a href="/music/%E5%B8%B8%E8%A7%81%E5%92%8C%E5%BC%A6/" title="常见和弦">常见和弦</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/code/" title="分类于 代码">代码</a> <i class="ic i-angle-right"></i> <a href="/categories/code/%E7%BC%96%E7%A8%8B%E8%AF%AD%E8%A8%80/" title="分类于 编程语言">编程语言</a></div><span><a href="/code/python%E5%AD%A6%E4%B9%A0%E7%AC%94%E8%AE%B01/" title="python学习笔记1">python学习笔记1</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/music/" title="分类于 音乐">音乐</a> <i class="ic i-angle-right"></i> <a href="/categories/music/%E4%B9%90%E7%90%86/" title="分类于 乐理">乐理</a></div><span><a href="/music/%E5%90%89%E4%BB%96%E4%B9%90%E7%90%86-Funk%E7%B1%BB/" title="吉他乐理-Funk类">吉他乐理-Funk类</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/code/" title="分类于 代码">代码</a> <i class="ic i-angle-right"></i> <a href="/categories/code/%E6%95%B0%E5%AD%A6%E5%BB%BA%E6%A8%A1/" title="分类于 数学建模">数学建模</a></div><span><a href="/code/%E5%88%86%E5%B8%83%E5%BC%8FMOM%E4%BD%9C%E4%B8%9A/" title="分布式MOM作业">分布式MOM作业</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/code/" title="分类于 代码">代码</a> <i class="ic i-angle-right"></i> <a href="/categories/code/%E6%95%B0%E5%AD%A6%E5%BB%BA%E6%A8%A1/" title="分类于 数学建模">数学建模</a></div><span><a href="/code/TOPSIS-MATLAB/" title="TOPSIS-MATLAB">TOPSIS-MATLAB</a></span></li></ul></div><div><h2>最新评论</h2><ul class="leancloud-recent-comment"></ul></div></div><div class="status"><div class="copyright">&copy; 2010 – <span itemprop="copyrightYear">2025</span> <span class="with-love"><i class="ic i-sakura rotate"></i> </span><span class="author" itemprop="copyrightHolder">Lemon Sour @ Amentiraz</span></div><div class="powered-by">基于 <span class="exturl" data-url="aHR0cHM6Ly9oZXhvLmlv">Hexo</span> & Theme.<span class="exturl" data-url="aHR0cHM6Ly9naXRodWIuY29tL2FtZWhpbWUvaGV4by10aGVtZS1zaG9rYQ==">Shoka</span></div></div></div></footer></div><script data-config type="text/javascript">var LOCAL={path:"article/scPerturb/",favicon:{show:"⁽⁽ଘ( ˊᵕˋ )ଓ⁾⁾",hide:"(つд⊂)"},search:{placeholder:"文章搜索",empty:"关于 「 ${query} 」，什么也没搜到",stats:"${time} ms 内找到 ${hits} 条结果"},valine:!0,copy_tex:!0,katex:!0,fancybox:!0,copyright:'复制成功，转载请遵守 <i class="ic i-creative-commons"></i>BY-NC-SA 协议。',ignores:[function(e){return e.includes("#")},function(e){return new RegExp(LOCAL.path+"$").test(e)}]}</script><script src="https://cdn.polyfill.io/v2/polyfill.js"></script><script src="//cdn.jsdelivr.net/combine/npm/pace-js@1.0.2/pace.min.js,npm/pjax@0.2.8/pjax.min.js,npm/whatwg-fetch@3.4.0/dist/fetch.umd.min.js,npm/animejs@3.2.0/lib/anime.min.js,npm/algoliasearch@4/dist/algoliasearch-lite.umd.js,npm/instantsearch.js@4/dist/instantsearch.production.min.js,npm/lozad@1/dist/lozad.min.js,npm/quicklink@2/dist/quicklink.umd.js"></script><script src="/js/app.js?v=0.2.5"></script></body></html><!-- rebuild by hrmmi -->