<!-- build time:Sun Oct 26 2025 10:52:08 GMT+0800 (中国标准时间) --><!DOCTYPE html><html lang="zh-CN"><head><meta charset="UTF-8"><meta name="viewport" content="width=device-width,initial-scale=1,maximum-scale=2"><meta name="theme-color" content="#FFF"><link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon.png"><link rel="icon" type="image/ico" sizes="32x32" href="/images/favicon.ico"><meta http-equiv="Cache-Control" content="no-transform"><meta http-equiv="Cache-Control" content="no-siteapp"><link rel="alternate" type="application/rss+xml" href="http://amentiraz.github.io/rss.xml"><link rel="alternate" type="application/atom+xml" href="http://amentiraz.github.io/atom.xml"><link rel="alternate" type="application/json" href="http://amentiraz.github.io/feed.json"><link rel="stylesheet" href="//fonts.googleapis.com/css?family=Mulish:300,300italic,400,400italic,700,700italic%7CFredericka%20the%20Great:300,300italic,400,400italic,700,700italic%7CNoto%20Serif%20JP:300,300italic,400,400italic,700,700italic%7CNoto%20Serif%20SC:300,300italic,400,400italic,700,700italic%7CInconsolata:300,300italic,400,400italic,700,700italic&display=swap&subset=latin,latin-ext"><link rel="stylesheet" href="/css/app.css?v=0.2.5"><meta name="keywords" content="生物,学习笔记类,算法"><link rel="canonical" href="http://amentiraz.github.io/article/Benchmarking-spatial-claustering-methods-with-spatially-resolved-transcriptomic-data/"><title>Benchmarking-spatial-claustering-methods-with-spatially-resolved-transcriptomic-data - 论文 | Amentiraz =</title><meta name="generator" content="Hexo 7.3.0"></head><body itemscope itemtype="http://schema.org/WebPage"><div id="loading"><div class="cat"><div class="body"></div><div class="head"><div class="face"></div></div><div class="foot"><div class="tummy-end"></div><div class="bottom"></div><div class="legs left"></div><div class="legs right"></div></div><div class="paw"><div class="hands left"></div><div class="hands right"></div></div></div></div><div id="container"><header id="header" itemscope itemtype="http://schema.org/WPHeader"><div class="inner"><div id="brand"><div class="pjax"><h1 itemprop="name headline">Benchmarking-spatial-claustering-methods-with-spatially-resolved-transcriptomic-data</h1><div class="meta"><span class="item" title="创建时间：2024-10-07 11:36:00"><span class="icon"><i class="ic i-calendar"></i> </span><span class="text">发表于</span> <time itemprop="dateCreated datePublished" datetime="2024-10-07T11:36:00+08:00">2024-10-07</time> </span><span class="item" title="本文字数"><span class="icon"><i class="ic i-pen"></i> </span><span class="text">本文字数</span> <span>20k</span> <span class="text">字</span> </span><span class="item" title="阅读时长"><span class="icon"><i class="ic i-clock"></i> </span><span class="text">阅读时长</span> <span>18 分钟</span></span></div></div></div><nav id="nav"><div class="inner"><div class="toggle"><div class="lines" aria-label="切换导航栏"><span class="line"></span> <span class="line"></span> <span class="line"></span></div></div><ul class="menu"><li class="item title"><a href="/" rel="start">Amentiraz</a></li></ul><ul class="right"><li class="item theme"><i class="ic i-sun"></i></li><li class="item search"><i class="ic i-search"></i></li></ul></div></nav></div><div id="imgs" class="pjax"><ul><li class="item" data-background-image="https://amentirazblogpic.oss-cn-hangzhou.aliyuncs.com/blogpic/%E5%BE%AE%E4%BF%A1%E5%9B%BE%E7%89%87_20240925124421.png"></li><li class="item" data-background-image="https://amentirazblogpic.oss-cn-hangzhou.aliyuncs.com/blogpic/%E5%BE%AE%E4%BF%A1%E5%9B%BE%E7%89%87_20240925124604.jpg"></li><li class="item" data-background-image="https://amentirazblogpic.oss-cn-hangzhou.aliyuncs.com/blogpic/%E5%BE%AE%E4%BF%A1%E5%9B%BE%E7%89%87_20240925124513.png"></li><li class="item" data-background-image="https://amentirazblogpic.oss-cn-hangzhou.aliyuncs.com/blogpic/%E5%BE%AE%E4%BF%A1%E5%9B%BE%E7%89%87_20240925124522.jpg"></li><li class="item" data-background-image="https://amentirazblogpic.oss-cn-hangzhou.aliyuncs.com/blogpic/%E5%BE%AE%E4%BF%A1%E5%9B%BE%E7%89%87_20240925162843.jpg"></li><li class="item" data-background-image="https://amentirazblogpic.oss-cn-hangzhou.aliyuncs.com/blogpic/%E5%BE%AE%E4%BF%A1%E5%9B%BE%E7%89%87_20240925124558.png"></li></ul></div></header><div id="waves"><svg class="waves" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" viewBox="0 24 150 28" preserveAspectRatio="none" shape-rendering="auto"><defs><path id="gentle-wave" d="M-160 44c30 0 58-18 88-18s 58 18 88 18 58-18 88-18 58 18 88 18 v44h-352z"/></defs><g class="parallax"><use xlink:href="#gentle-wave" x="48" y="0"/><use xlink:href="#gentle-wave" x="48" y="3"/><use xlink:href="#gentle-wave" x="48" y="5"/><use xlink:href="#gentle-wave" x="48" y="7"/></g></svg></div><main><div class="inner"><div id="main" class="pjax"><div class="article wrap"><div class="breadcrumb" itemscope itemtype="https://schema.org/BreadcrumbList"><i class="ic i-home"></i> <span><a href="/">首页</a></span><i class="ic i-angle-right"></i> <span class="current" itemprop="itemListElement" itemscope itemtype="https://schema.org/ListItem"><a href="/categories/article/" itemprop="item" rel="index" title="分类于 论文"><span itemprop="name">论文</span></a><meta itemprop="position" content="1"></span></div><article itemscope itemtype="http://schema.org/Article" class="post block" lang="zh-CN"><link itemprop="mainEntityOfPage" href="http://amentiraz.github.io/article/Benchmarking-spatial-claustering-methods-with-spatially-resolved-transcriptomic-data/"><span hidden itemprop="author" itemscope itemtype="http://schema.org/Person"><meta itemprop="image" content="/images/avatar.jpg"><meta itemprop="name" content="Lemon Sour"><meta itemprop="description" content=", "></span><span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization"><meta itemprop="name" content=""></span><div class="body md" itemprop="articleBody"><h1 id="单词表"><a class="anchor" href="#单词表">#</a> 单词表</h1><details class="primary"><summary>单词表</summary><div><p>scope: 范围<br>evaluate: 评价<br>scenario: 设想的情况<br>multiplex: 多路运输<br>pose: 构成<br>substantial: 可观的<br>annotation: 注释<br>conquer: 征服<br>extant: 尚存的<br>extensively: 全面的<br>laminar: 薄片 / 层型<br>NMI: normalized mutual information<br>incentivize: 激励<br>enccompass: 包含<br>silhouette: 轮廓<br>ASW: average silhouette width<br>PAS: perccentage of abnormal spots<br>adjacency: 邻接物<br>quantutative: 定量的<br>necessitate: 使・・・成为必要<br>cutting-edge: 简短<br>multiomic: 多组学<br>predominantly: 全面的<br>deconvolution: 去卷积<br>imputation: 归咎<br>modularity: 模块化<br>histological: 组织学<br>autoencoder: 自编码<br>facilitate: 促进<br>contingency: 应急事件<br>contingency tabble: 列联表<br>spectrometry： 质谱<br>manifold: 多种多样的<br>UMAP: uniform manifold approximation and projection<br>concatenate: 把・・・练习起来<br>batch effect: 批次效应<br>unadorned: 未修饰的<br>HOM: the homogeneity score<br>COM: the completeness score</p></div></details><h1 id="文章概括"><a class="anchor" href="#文章概括">#</a> 文章概括</h1><p>这篇文章就不摘要观点了，基本都是思路和算法，后面会详尽的列出算法和实现的 python 代码，大致是实现 SpaGCN (HE) 的代码<br>这篇文章主要是针对 13 种对于在 34 个 SRT (spatially resolved transcriptomics) 的数据的计算方法做了 benchmark study，并针对 accuracy,sptial continuity,marker genes detection,scalability,and robustness 做了评估。针对各个方法的特点提出了自己的选择建议。并在最后提出了自己针对大规模数据的 Method。</p><p><img data-src="https://amentirazblogpic.oss-cn-hangzhou.aliyuncs.com/blogtab/benchmarksp1.jpg" alt=""></p><p>实验发现没有一种方法是适用于所有的数据的</p><p><span class="exturl" data-url="aHR0cDovL3NkbWJlbmNoLmRyYWkuY24v">选择方法和数据的网站</span></p><p>接着作者检测了现有方法的局限性，利用的数据是 (additional 22 data containing small and non continuous tissue domains, and during multislice analysis on another large-scale dataset containing 31 tissue slices)<br>作者提出解决局限性的策略是 'divide and conquer' strategy</p><p>最后，作者检测了鲁棒性和其它因素 (gene expression matrix sparsity, spatial resolution, the number of genes and the level of noise)， 也检测了 pre- and postpocessing steps on diffenrent methods' performance.</p><blockquote><p>As these data(10x Visium, stereo-Seq, BaristaSeq, MERFISH, osmFISH, STARmap and STARmap*) were pproduced using distinct spatial technologies, they exhibit different data characteristics and collectively span a wide range of potential spatiial transcriptomics data types, which allows users of different spatial technologies to benefit from study</p></blockquote><ul><li>nonspatial methods<br>Louvain<br>Leiden</li><li>sptial methods<br>SpaGCN<br>BayesSpace<br>stLearn<br>SEDR<br>CCST<br>SCAN-IT<br>STAGATE<br>SpaceFlow<br>conST<br>BASS<br>GraphST</li></ul><p><img data-src="https://amentirazblogpic.oss-cn-hangzhou.aliyuncs.com/blogtab/benchmarksp2.jpg" alt=""><br>Benchmarking analysis on 10x Visium dataset<br>VS<br>Benchmarking analysis on MERFISH dataset<br>对于 Comprehensive evaluation:</p><ul><li>Accuracy<br>NMI<br>HOM<br>COM</li><li>Continuity:<br>CHAOS<br>ASW<br>PAS</li><li>The quality of domain-specific marker genes<br>Moran's I<br>Geary's C</li><li>Scalability<br>Time<br>Memory</li></ul><p><img data-src="https://amentirazblogpic.oss-cn-hangzhou.aliyuncs.com/blogtab/benchmarksp3.jpg" alt=""></p><blockquote><p>Through the analysis of methods performance across different data, we have observed interesting correlation patterns as determined by the Spearman correlation coefficient, underlining the relative ordering of their performance (Fig. 3a, the underlying metrics values can be found in Extended Data Fig. 2). First, we found that, within the same spatial technology, a strong Spearman correlation emerged when mul- tiple data samples originated from the same individual and shared identical tissue structure. A notable example of this is the high consist- ency in the relative performance of all methods applied on three BaristaSeq datasets (Fig. 3a, blue box).<br>Second, when tissue structure is controlled, methods performance may be influenced by the use of different technologies. Evidence for this is seen when comparing the performance of methods on the 10x Visium dataset with that of the BaristaSeq dataset. Sharing similar tissue structures (brain cortex region, Supplementary Table 1), the two datasets, generated by differ- ent spatial technologies, yielded different performance outcomes (Spearman’s correlation is 0.33, Fig. 3b).<br>Third, when spatial technology is controlled, the origin of the data samples (that is, different donors) can also influence method performance. An example is on the 10x Visium dataset, the significantly higher correlations (Supplementary Fig. 9, P = 4.65 × 10−4) when applying methods on data from the same donor than from the different donors (correlation values from the same donor are highlighted in red box in Fig. 3a)</p></blockquote><ul><li>the performance of all methods bu partitioning the datasets into two groups:<br>10x Visium<br>imaging-based datasets(MERFISH, osmFISH, BaristaSeq, STARmap, STARmap*)</li></ul><p><img data-src="https://amentirazblogpic.oss-cn-hangzhou.aliyuncs.com/blogtab/benchmarksp4.jpg" alt=""></p><h2 id="limitations"><a class="anchor" href="#limitations">#</a> Limitations</h2><blockquote><p>We found that all methods encounter challenges when faced with smaller, noncontinuous tissue domains<br>All methods encountered time and/or memory issues when applied to the dataset</p></blockquote><p><img data-src="https://amentirazblogpic.oss-cn-hangzhou.aliyuncs.com/blogtab/benchmarksp5.jpg" alt=""></p><h2 id="a-divide-and-conquer-strategy-enables-large-scale-scalability"><a class="anchor" href="#a-divide-and-conquer-strategy-enables-large-scale-scalability">#</a> A divide and conquer strategy enables large-scale scalability</h2><p>使用的 base solver: SpaceFlow<br>这段话描述了一种 <strong>分而治之</strong> 的方法，应用于大规模的空间聚类问题。该方法基于 SpaceFlow 模型，结合了多种聚类算法和嵌入技术，以解决不同切片的空间数据分析，并整合它们的结果。</p><h3 id="核心概念解释"><a class="anchor" href="#核心概念解释">#</a> 核心概念解释：</h3><ol><li><p><strong>Divide and conquer strategy</strong>：分而治之策略将大规模问题拆解成多个小问题分别处理。在这里，原始数据是 31 个切片的空间数据，这些切片被分别处理，然后将结果进行整合。</p></li><li><p><strong>SpaceFlow</strong>：这是一个用于单切片分析的工具，能够生成<strong>上下文感知的嵌入</strong>和<strong>空间聚类标签</strong>。每个切片独立运行 SpaceFlow，生成相应的嵌入和聚类结果。</p></li><li><p><strong>UMAP 和批次效应</strong>：为了观察是否存在批次效应（不同批次之间的系统差异），他们将所有 31 个切片的嵌入拼接在一起，使用 UMAP 进行可视化，结果显示了批次效应的存在（见 Fig. 5a）。</p></li><li><p><strong>Harmony 整合方法</strong>：为了解决批次效应，使用了 <strong>Harmony</strong> 这种整合方法。具体步骤是：首先用 PCA 将拼接后的嵌入降维至 30 维，然后应用 Harmony 对嵌入进行整合。整合后的嵌入被用来构建邻域图，随后用 UMAP 进行可视化，并用不同的聚类算法进行空间聚类。</p></li><li><p><strong>空间聚类方法</strong>：使用了三种不同的聚类算法来生成空间聚类标签，包括：</p><ul><li><strong>Leiden</strong>：一种社区检测算法。</li><li><strong>Louvain</strong>：另一种社区检测算法，类似 Leiden。</li><li><strong>mclust</strong>：基于高斯混合模型的聚类方法。</li></ul><p>不同的组合方式被用于命名不同的方法，比如：</p><ul><li><strong>SpaceFlow-DC-mclust</strong>：使用整合后的嵌入，结合 mclust 聚类方法。</li><li><strong>SpaceFlow-DC-Louvain</strong>：使用整合后的嵌入，结合 Louvain 聚类方法。<br>不同的算法对上下文感知表示的影响不大，可以根据需要选择不同的聚类算法且不会对最终性能产生显著影响</li></ul></li><li><p><strong>嵌入的可视化</strong>：有两种不同的嵌入可视化方式：</p><ul><li><strong>Embedding (SpaceFlow)</strong>：表示未经过 Harmony 整合的嵌入。</li><li><strong>Embedding (Proposed)</strong>：表示经过 Harmony 整合的嵌入。</li></ul></li><li><p><strong>NMI（Normalized Mutual Information）</strong>：为了评估聚类的准确性，使用了 NMI 指标。分别记录了<strong>每个切片的 NMI</strong> 和<strong>所有切片的整体 NMI</strong>。由于未对跨切片的 SpaceFlow 聚类标签进行对齐，导致每个切片的聚类准确性高，而整体的准确性较低。</p></li></ol><h2 id="pre-and-postpocessing"><a class="anchor" href="#pre-and-postpocessing">#</a> pre- and postpocessing</h2><p>这段话讨论了在空间聚类分析中，预处理和后处理步骤对结果的影响，以及不同方法在不同数据特征下的鲁棒性（稳健性）。</p><p><img data-src="https://amentirazblogpic.oss-cn-hangzhou.aliyuncs.com/blogtab/benchmarksp6.jpg" alt=""></p><h3 id="1-预处理和后处理对性能的影响"><a class="anchor" href="#1-预处理和后处理对性能的影响">#</a> 1. <strong>预处理和后处理对性能的影响</strong>：</h3><ul><li><p><strong>预处理步骤</strong>：在空间转录组数据的聚类分析中，预处理通常包括对基因进行选择，比如选择高度可变基因（HVG）或空间上可变基因（SVG），或者不进行任何选择。不同的预处理方式可能会影响聚类结果的质量。在实验中，他们测试了三种预处理方式：</p><ul><li><strong>HVG 选择</strong>：选择高度可变基因。</li><li><strong>SVG 选择</strong>：选择空间上有差异的基因。</li><li><strong>不选择基因</strong>：不进行任何基因选择。</li></ul><p>通过分析，研究发现大约一半的方法在<strong>没有基因选择</strong>的情况下表现更好，表明预处理可能并非在所有情况下都有利。</p></li><li><p><strong>后处理步骤</strong>：后处理主要用于对聚类结果进行精细化，比如通过<strong> K 近邻（KNN）方法</strong>对空间域标签进行平滑处理，使得聚类结果在空间上更加一致。在这些实验中，发现所有方法在进行这种空间平滑的后处理步骤后，聚类性能都有所提升。</p></li></ul><h3 id="2-数据特征对方法鲁棒性的影响"><a class="anchor" href="#2-数据特征对方法鲁棒性的影响">#</a> 2. <strong>数据特征对方法鲁棒性的影响</strong>：</h3><p>空间转录组数据在不同实验中可能表现出不同的特性，比如基因表达矩阵的稀疏性、空间分辨率的高低、以及基因数量的多少。因此，模拟具有不同参数的数据有助于更广泛的参考，并且测试方法对这些特征的鲁棒性显得至关重要。</p><p>他们通过模拟数据，测试了不同方法对以下特征的鲁棒性：</p><ul><li><p><strong>基因表达矩阵稀疏性</strong>：即矩阵中缺失或零值的比例。在稀疏性增加时，大部分方法的性能下降，除了 CCST、Louvain 和 Leiden 这三种方法。</p></li><li><p><strong>空间分辨率</strong>：即每个空间单位所包含的细胞或基因数目。大部分方法的性能在空间分辨率降低时有所提升，除了 Louvain、Leiden、SCAN-IT 和 SpaceFlow，这几种方法的性能在分辨率变化时表现稳定或下降。</p></li><li><p><strong>基因数量</strong>：随着基因数量的增加，大多数方法的表现会提高，表明更丰富的基因信息有助于更准确的聚类。</p></li><li><p><strong>噪声的影响</strong>：通过加入噪声测试方法的稳健性，结果显示，大多数空间聚类方法对噪声具有一致的鲁棒性，表明它们在噪声环境下的表现相对稳定。</p></li></ul><h2 id="discussion"><a class="anchor" href="#discussion">#</a> Discussion</h2><p>这段话总结了在使用空间转录组数据进行空间聚类分析中对 13 种方法的基准分析，以及对未来基准研究设计的启示。</p><h3 id="1-研究概述"><a class="anchor" href="#1-研究概述">#</a> 1. <strong>研究概述</strong>：</h3><ul><li><strong>目标</strong>：评估 13 种空间聚类方法，用于识别空间转录组数据中的空间域。</li><li><strong>评价维度</strong>：包括准确性、连续性、标记基因得分（marker score）和可扩展性，为研究人员选择最优的聚类工具提供框架。</li><li><strong>主要结论</strong>：没有单一方法在所有数据集上都表现最优，不同数据特征适合不同方法。此外，当前方法在识别小区域、多片段分析和处理大规模数据上存在局限性。</li></ul><h3 id="2-方法的局限性与改进"><a class="anchor" href="#2-方法的局限性与改进">#</a> 2. <strong>方法的局限性与改进</strong>：</h3><ul><li><strong>小区域识别困难</strong>：当前方法难以有效识别小的空间区域。</li><li><strong>多片段分析能力欠缺</strong>：现有方法多局限于单片段分析，无法处理多个切片的数据。</li><li><strong>可扩展性问题</strong>：随着空间转录组数据的规模增长，内存和计算时间的要求增加，现有方法在处理大规模数据时存在可扩展性不足的问题。</li></ul><p><strong>改进策略</strong>：提出了 “分而治之” 的策略，通过将数据分割并分别处理，再通过合并步骤提升大规模数据的处理能力。这种策略可以激励开发者结合现有工具，解决大规模空间数据集带来的挑战。</p><h3 id="3-额外数据的引入"><a class="anchor" href="#3-额外数据的引入">#</a> 3. <strong>额外数据的引入</strong>：</h3><ul><li><strong>HE 染色图像的影响</strong>：在评估 SpaGCN 方法时，研究发现加入 HE 染色图像并不总能提升性能，有时可能引入噪声。可能的解释是 HE 图像中包含的信息无法直接帮助识别组织结构，或是 HE 图像的建模方式仍有改进空间。</li></ul><h3 id="4-研究的局限性与未来方向"><a class="anchor" href="#4-研究的局限性与未来方向">#</a> 4. <strong>研究的局限性与未来方向</strong>：</h3><ul><li><strong>快速发展的技术领域</strong>：空间转录组学领域发展迅速，现有的研究可能未能涵盖最新的数据类型。</li><li><strong>其他空间数据类型的缺失</strong>：研究主要集中在空间转录组数据，忽略了空间蛋白质组学和空间多组学数据，它们可能提供互补的信息。未来的研究应涵盖更多种类的数据和计算方法。</li><li><strong>基准测试设计的启发</strong>：研究强调了未来基准测试应考虑的数据集多样性、可靠的 “ground truth” 注释，以及更广泛的空间数据类型，以确保研究结果的普适性和可比性。</li></ul><h3 id="5-评价指标"><a class="anchor" href="#5-评价指标">#</a> 5. <strong>评价指标</strong>：</h3><ul><li><strong>相似度指标</strong>：衡量方法预测结果与 “ground truth” 之间的相似度。例如，研究中使用 NMI、HOM 和 COM 来评估这种相似性。</li><li><strong>空间连续性</strong>：如 CHAOS、PAS 和 ASW 指标，用于衡量预测的空间域的连续性。然而，这类指标并非总能代表好的预测结果，因为高空间连续性不一定意味着高准确性。</li><li><strong>可扩展性</strong>：运行时间和内存占用是衡量方法可扩展性的重要指标，尤其是在大规模数据集上。</li><li><strong>数据集选择</strong>：优先选择带有可靠 “ground truth” 的数据集。如果没有可靠的真实数据集，模拟数据可作为替代品。</li></ul><h2 id="method"><a class="anchor" href="#method">#</a> Method</h2><p>we adopted two approaches based on the capabili- ties of the methods.</p><ol><li>For algorithms where we could directly input the expected number of spatial domains (for example, SpaGCN and BayesSpace), we set the parameter to the actual number of domains obtained from our ground truth.</li><li>For those algorithms that could only accept clustering resolution, we searched the resolution that best matches the expected number of spatial domains.</li></ol><h3 id="空间聚类方法"><a class="anchor" href="#空间聚类方法">#</a> 空间聚类方法：</h3><ol><li><p><strong>Louvain</strong>：一种非空间聚类算法，基于模块度优化，通过反复合并和分割社区来分配每个点，直到得到期望的聚类结果。</p></li><li><p><strong>Leiden</strong>：类似于 Louvain 的非空间聚类算法，但进一步改进，可以通过合并相似社区来增强聚类效果。</p></li><li><p><strong>SpaGCN</strong>：利用图卷积网络（GCN）将基因表达、空间位置和组织学信息整合起来，通过无监督的迭代聚类划分空间域。用于无图像数据的情况。</p></li><li><p><strong>SpaGCN(HE)</strong>：类似于 SpaGCN，但适用于有 H&amp;E 组织学图像的数据，通过将 RGB 值转换为 3D 坐标以计算空间点之间的欧几里得距离。</p></li><li><p><strong>BayesSpace</strong>：贝叶斯统计方法，利用基因表达矩阵的低维表示进行空间聚类，通过空间先验鼓励相邻点属于同一簇。适用于 “spot” 形式的空间数据。</p></li><li><p><strong>stLearn</strong>：基于空间形态学基因表达（SME）归一化数据进行无监督聚类，将相似点归为同一簇，并利用空间信息识别组织内的子类群。需要组织学图像作为输入。</p></li><li><p><strong>SEDR</strong>：使用深度自编码器构建基因表达数据的低维表示，然后与空间信息结合，通过变分图自编码器进行同时的空间嵌入和聚类。</p></li><li><p><strong>CCST</strong>：将细胞位置和基因表达信息编码为邻接矩阵和基因表达矩阵，通过深度图信息最大化网络获取包含空间和基因表达信息的节点嵌入，并通过主成分分析（PCA）和 k-means++ 进行聚类。</p></li><li><p><strong>SCAN-IT</strong>：使用图像分割方法处理空间聚类问题，将细胞视为图像中的像素，基因表达视为类似 RGB 通道的数据，构建几何感知的空间邻接图，利用深度图信息最大化生成低维嵌入进行聚类。</p></li><li><p><strong>STAGATE</strong>：将空间位置信息转化为空间邻居网络，并结合基因表达信息，通过图注意力自编码网络生成低维嵌入，用于空间聚类。</p></li><li><p><strong>SpaceFlow</strong>：利用深度图神经网络（GNN）将基因表达相似性和空间信息融合，通过空间正则化生成空间一致的低维嵌入，用于聚类。</p></li><li><p><strong>conST</strong>：基于对比学习方法，整合多模态的空间转录组学（SRT）数据（如基因表达、空间信息和形态学），通过数据增强和三层对比学习生成低维嵌入进行聚类。</p></li><li><p><strong>BASS</strong>：一种多尺度、多样本的空间转录组学数据分析方法，采用贝叶斯分层建模框架进行聚类。</p></li><li><p><strong>GraphST</strong>：通过图自监督对比学习模型进行空间聚类，结合 GNN 和自监督学习，学习空间转录组数据中的基因表达和空间位置信息嵌入，用于聚类。</p></li></ol><p>这些方法都致力于结合空间位置信息和基因表达数据，通过不同的模型和算法来提高空间聚类效果。</p><h3 id="simulated-data"><a class="anchor" href="#simulated-data">#</a> Simulated data</h3><p>分别考察了不同的 spatial resolutions\expression matrix sparsity\gene subsampling rate\various levels of noise, 使用了不同的方法</p><h3 id="benchmark-metrics"><a class="anchor" href="#benchmark-metrics">#</a> Benchmark metrics</h3><h4 id="nmi"><a class="anchor" href="#nmi">#</a> NMI</h4><p>衡量两个聚类之间的相似性，NMI 属于 0-1 之间，越接近 1 说明越相似。<br>衡量的是从一个聚类结果预测另一个聚类结果所需的额外信息量<br>适用于类别数量不一致的聚类结果评估</p><blockquote><p>Suppose P is the spatial domain clustering result, T is the ground truth clustering label, their entropies are H(P) and H(T) , respectively, and the mutual information is MI(P,T), then NMI is computed as:</p></blockquote>\begin{align*} NMI = \frac{MI(P,T)}{\sqrt{H(P)H(T)}} \end{align*}<h4 id="ari"><a class="anchor" href="#ari">#</a> ARI</h4><p>比较两个聚类结果种每一对数据点的划分是否一致<br>适合在类别数量一致的情况下评估两个聚类结果<br>ARI is computed as:</p>\begin{align*} & ARI = \frac{TP+TN-E}{TP+TN+FP+FN-E} \\ & where: \\ & E = \frac{(TP+FP)\times(TP)+(FN+TN)\times(FP+TN)}{TP+TN+FP+FN} \end{align*}<h4 id="homhomogeneity-score"><a class="anchor" href="#homhomogeneity-score">#</a> HOM（Homogeneity Score）：</h4><p>衡量聚类结果与已知真实标签相比的同质性。如果一个聚类的每个簇只包含属于同一类别的数据点，则聚类是同质的。HOM 得分范围为 0 到 1，1 表示完全同质。</p><h4 id="comcompleteness-score"><a class="anchor" href="#comcompleteness-score">#</a> COM（Completeness Score）：</h4><p>评估聚类结果相对于真实标签的完整性。如果属于同一类别的所有数据点都被正确分配到同一簇，则聚类是完整的。COM 得分范围为 0 到 1，1 表示完全完整。</p><h4 id="chaos"><a class="anchor" href="#chaos">#</a> CHAOS</h4><p><strong>CHAOS（Continuity Assessment for High-throughput Open Science）</strong>：用于评估质谱成像和空间转录组学领域中空间连续性的指标。CHAOS 得分越低，表示空间域识别的连续性结果越好。计算过程包括以下步骤：</p><ol><li>构建每个数据的 1 - 最近邻（1-NN）图，每个细胞与在物理空间中具有最小欧几里得距离的其他细胞连接。</li><li>定义连接的权重 \( w_{kij} \)：如果细胞 \( i \) 和细胞 \( j \) 在簇 \( k \) 中相连，则 \( w_{kij} = d_{ij} \)（它们之间的欧几里得距离），否则为 0。</li><li>CHAOS 得分计算公式为：\begin{align*} CHAOS = \sum_{k=1}^{K} \sum_{i=1}^{n_k} w_{kij} \end{align*} 其中 \( n_k \) 是簇 \( k \) 中的细胞数量，\( n \) 是数据中的总细胞数，\( K \) 是唯一空间域的数量。</li></ol><h4 id="paspercentage-of-area-shared"><a class="anchor" href="#paspercentage-of-area-shared">#</a> PAS（Percentage of Area Shared）：</h4><p>用于量化空间转录组学中空间域识别算法的空间同质性指标。较低的 PAS 分数表示检测到的空间域的连续性更好，期望在空间域内细胞的同质性更高。PAS 分数的计算方式是：具有不同空间域标签的细胞占其邻近十个细胞中至少六个的百分比。</p><h4 id="aswaverage-silhouette-width"><a class="anchor" href="#aswaverage-silhouette-width">#</a> ASW（Average Silhouette Width）：</h4><p>最初用于评估聚类标签与嵌入（或距离矩阵）之间一致性的指标。ASW 被扩展用于评估预测空间域相对于物理空间的空间一致性。ASW 的值范围为 - 1 到 1（通常重标定到 0-1），ASW 值越接近 1，性能越好。ASW 的计算需要先定义<strong>轮廓宽度（Silhouette Width，SW）</strong>，SW 的计算方式如下：</p><ol><li><p><strong>SW 的定义</strong>：</p><ul><li>设 \( a \) 为细胞与同一空间域内所有其他细胞的平均距离。</li><li>设 \( b \) 为细胞与下一个最近聚类中所有其他细胞的平均距离。</li><li>则细胞的 SW 计算公式为：</li></ul>\begin{align*} SW = \frac{b - a}{\max(a, b)} \end{align*}</li><li><p><strong>ASW 的计算</strong>：ASW 通过对所有细胞的 SW 值取平均来获得。</p></li></ol><h4 id="morans-i"><a class="anchor" href="#morans-i">#</a> Moran's I</h4><p>是一种用于量化空间自相关程度的指标，广泛应用于空间统计学，尤其是在空间组学领域评估检测到的 SVG（空间变异基因）是否表现出有序的空间表达模式。Moran's I 的值范围为 - 1 到 1：</p><ul><li><strong>接近 1</strong>：表示存在明显的空间基因表达模式，说明相似的基因表达值在空间上是聚集的。</li><li><strong>接近 0</strong>：表示随机的空间基因表达模式，说明基因表达在空间上没有显著的规律性。</li><li><strong>接近 - 1</strong>：表示基因表达模式呈现出棋盘状的分布，即相似和不同的表达值交替分布。</li></ul><p>在计算 Moran's I 时，假设有两个细胞 \(x_i\) 和 \(x_j\) 的基因表达值，\(\bar{x}\) 是该基因的平均表达值，\(N\) 是细胞的总数。Moran's I 的计算公式为：</p>\begin{align*} & Moran's \ I = \frac{N}{W} \cdot \frac{\sum_{i=1}^{N} \sum_{j=1}^{N} w_{ij} (x_i - \bar{x})(x_j - \bar{x})}{\sum_{i=1}^{N} (x_i - \bar{x})^2} \\ & where \\ & w_{ij} = \begin{cases} 1,& \text{ifi and j are spatial neighbors} \\ 0,& \text{else} \end{cases} \\ & where \\ & W = \sum_{i,j}{} w_{ij} \end{align*}<p>其中：</p><ul><li>\(W\) 是所有权重 \(w_{ij}\) 的总和，通常表示细胞之间的空间权重。</li><li>\(w_{ij}\) 是细胞 \(i\) 和细胞 \(j\) 之间的空间权重，反映它们在空间上的关系（例如，距离或邻接关系）。</li></ul><h4 id="gearys-c"><a class="anchor" href="#gearys-c">#</a> Geary's C</h4><p>是一种用于量化空间自相关程度的指标，类似于 <strong>Moran's I</strong>，但它的计算方式和数值范围有所不同。以下是 Geary's C 的主要特点和计算方法：</p><ul><li><strong>比较</strong>：Geary's C 和 Moran's I 都用于分析空间自相关，但 Geary's C 更加关注相邻细胞之间的差异性，而 Moran's I 更加关注整体的聚集性。</li></ul><p>计算方法</p><p>在计算 Geary's C 时，假设使用与 Moran's I 相同的符号表示，计算公式为：</p>\begin{align*} & Geary's \ C = \frac{N \cdot \sum_{i=1}^{N} \sum_{j=1}^{N} w_{ij} (x_i - x_j)^2}{2W \cdot \sum_{i=1}^{N} (x_i - \bar{x})^2} \end{align*}<p>其中：</p><ul><li>\(N\) 是细胞的总数。</li><li>\(w_{ij}\) 是细胞 \(i\) 和细胞 \(j\) 之间的空间权重（通常基于距离或邻接关系）。</li><li>\(x_i\) 和 \(x_j\) 是细胞 \(i\) 和细胞 \(j\) 的基因表达值。</li><li>\(\bar{x}\) 是该基因的平均表达值。</li></ul><h2 id="代码实现"><a class="anchor" href="#代码实现">#</a> 代码实现</h2><h3 id="主代码"><a class="anchor" href="#主代码">#</a> 主代码</h3><figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br><span class="line">44</span><br><span class="line">45</span><br><span class="line">46</span><br><span class="line">47</span><br><span class="line">48</span><br><span class="line">49</span><br><span class="line">50</span><br><span class="line">51</span><br><span class="line">52</span><br><span class="line">53</span><br><span class="line">54</span><br><span class="line">55</span><br><span class="line">56</span><br><span class="line">57</span><br><span class="line">58</span><br><span class="line">59</span><br><span class="line">60</span><br><span class="line">61</span><br><span class="line">62</span><br><span class="line">63</span><br><span class="line">64</span><br><span class="line">65</span><br><span class="line">66</span><br><span class="line">67</span><br><span class="line">68</span><br><span class="line">69</span><br><span class="line">70</span><br><span class="line">71</span><br><span class="line">72</span><br><span class="line">73</span><br><span class="line">74</span><br><span class="line">75</span><br><span class="line">76</span><br><span class="line">77</span><br><span class="line">78</span><br><span class="line">79</span><br><span class="line">80</span><br><span class="line">81</span><br><span class="line">82</span><br><span class="line">83</span><br><span class="line">84</span><br><span class="line">85</span><br><span class="line">86</span><br><span class="line">87</span><br><span class="line">88</span><br><span class="line">89</span><br><span class="line">90</span><br><span class="line">91</span><br><span class="line">92</span><br><span class="line">93</span><br><span class="line">94</span><br><span class="line">95</span><br><span class="line">96</span><br><span class="line">97</span><br><span class="line">98</span><br><span class="line">99</span><br><span class="line">100</span><br><span class="line">101</span><br><span class="line">102</span><br><span class="line">103</span><br><span class="line">104</span><br><span class="line">105</span><br><span class="line">106</span><br><span class="line">107</span><br><span class="line">108</span><br><span class="line">109</span><br><span class="line">110</span><br><span class="line">111</span><br><span class="line">112</span><br><span class="line">113</span><br><span class="line">114</span><br><span class="line">115</span><br><span class="line">116</span><br><span class="line">117</span><br><span class="line">118</span><br><span class="line">119</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">import</span> scanpy <span class="keyword">as</span> sc</span><br><span class="line"><span class="keyword">from</span> sklearn <span class="keyword">import</span> metrics</span><br><span class="line"></span><br><span class="line"><span class="keyword">import</span> os,csv,re</span><br><span class="line"><span class="keyword">import</span> pandas <span class="keyword">as</span> pd</span><br><span class="line"><span class="keyword">import</span> numpy <span class="keyword">as</span> np</span><br><span class="line"><span class="keyword">import</span> scanpy <span class="keyword">as</span> sc</span><br><span class="line"><span class="keyword">import</span> math</span><br><span class="line"><span class="keyword">import</span> SpaGCN <span class="keyword">as</span> spg</span><br><span class="line"><span class="keyword">from</span> scipy.sparse <span class="keyword">import</span> issparse</span><br><span class="line"><span class="keyword">import</span> random, torch</span><br><span class="line"><span class="keyword">import</span> warnings</span><br><span class="line">warnings.filterwarnings(<span class="string">&quot;ignore&quot;</span>)</span><br><span class="line"><span class="keyword">import</span> matplotlib.colors <span class="keyword">as</span> clr</span><br><span class="line"><span class="keyword">import</span> matplotlib.pyplot <span class="keyword">as</span> plt</span><br><span class="line"><span class="keyword">import</span> SpaGCN <span class="keyword">as</span> spg</span><br><span class="line"><span class="keyword">import</span> anndata <span class="keyword">as</span> ad</span><br><span class="line"><span class="keyword">import</span> cv2</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">datadir = <span class="string">&#x27;../Data&#x27;</span><span class="comment">#dataset path</span></span><br><span class="line">memory=[<span class="number">1</span>,<span class="number">2</span>,<span class="number">3</span>,<span class="number">4</span>,<span class="number">5</span>,<span class="number">6</span>,<span class="number">7</span>,<span class="number">8</span>,<span class="number">9</span>,<span class="number">10</span>]</span><br><span class="line">during_time=[<span class="number">1</span>,<span class="number">2</span>,<span class="number">3</span>,<span class="number">4</span>,<span class="number">5</span>,<span class="number">6</span>,<span class="number">7</span>,<span class="number">8</span>,<span class="number">9</span>,<span class="number">10</span>]</span><br><span class="line">ari_list=[<span class="number">1</span>,<span class="number">2</span>,<span class="number">3</span>,<span class="number">4</span>,<span class="number">5</span>,<span class="number">6</span>,<span class="number">7</span>,<span class="number">8</span>,<span class="number">9</span>,<span class="number">10</span>]</span><br><span class="line"></span><br><span class="line">new_data = sc.read_h5ad(<span class="string">f&#x27;<span class="subst">&#123;datadir&#125;</span>/151673.h5ad&#x27;</span>)</span><br><span class="line"></span><br><span class="line"><span class="keyword">for</span> j <span class="keyword">in</span> <span class="built_in">range</span>(<span class="number">10</span>):</span><br><span class="line"></span><br><span class="line">    <span class="keyword">import</span> tracemalloc</span><br><span class="line">    <span class="keyword">import</span> time</span><br><span class="line"> </span><br><span class="line">    tracemalloc.start()  </span><br><span class="line">    start_time=time.time()</span><br><span class="line"></span><br><span class="line">    adata = sc.read_h5ad(<span class="string">f&#x27;<span class="subst">&#123;datadir&#125;</span>/151673.h5ad&#x27;</span>)</span><br><span class="line">    </span><br><span class="line">    adata.var_names=adata.var.index.astype(<span class="string">&quot;str&quot;</span>)</span><br><span class="line">    adata.var_names_make_unique()</span><br><span class="line">    spg.prefilter_genes(adata,min_cells=<span class="number">3</span>) <span class="comment"># avoiding all genes are zeros</span></span><br><span class="line">    spg.prefilter_specialgenes(adata)</span><br><span class="line">    <span class="comment">#Normalize and take log for UMI</span></span><br><span class="line">    sc.pp.normalize_per_cell(adata)</span><br><span class="line">    sc.pp.log1p(adata)</span><br><span class="line"></span><br><span class="line">    adata.obs[<span class="string">&#x27;x_pixel&#x27;</span>]=adata.obsm[<span class="string">&#x27;spatial&#x27;</span>][:,<span class="number">0</span>]</span><br><span class="line">    adata.obs[<span class="string">&#x27;y_pixel&#x27;</span>]=adata.obsm[<span class="string">&#x27;spatial&#x27;</span>][:,<span class="number">1</span>]</span><br><span class="line"></span><br><span class="line">    x_pixel=adata.obs[<span class="string">&quot;x_pixel&quot;</span>].tolist()</span><br><span class="line">    y_pixel=adata.obs[<span class="string">&quot;y_pixel&quot;</span>].tolist()</span><br><span class="line"></span><br><span class="line">    <span class="comment">#Calculate adjacent matrix</span></span><br><span class="line">    s=<span class="number">1</span></span><br><span class="line">    b=<span class="number">49</span></span><br><span class="line">    </span><br><span class="line">    <span class="comment">#If histlogy image is not available, SpaGCN can calculate the adjacent matrix using the fnction below</span></span><br><span class="line">    adj=spg.calculate_adj_matrix(x=x_pixel,y=y_pixel, histology=<span class="literal">False</span>)</span><br><span class="line">    np.savetxt(<span class="string">f&#x27;./adj.csv&#x27;</span>, adj, delimiter=<span class="string">&#x27;,&#x27;</span>)</span><br><span class="line">    <span class="comment">#spatial domain detection using SpaGCN</span></span><br><span class="line"></span><br><span class="line">    <span class="comment">#expression data preprocessing</span></span><br><span class="line">    adj=np.loadtxt(<span class="string">f&#x27;./adj.csv&#x27;</span>, delimiter=<span class="string">&#x27;,&#x27;</span>)</span><br><span class="line">    </span><br><span class="line">    <span class="comment">#set hyper-parameters</span></span><br><span class="line">    p=<span class="number">0.5</span> </span><br><span class="line">    <span class="comment">#Find the l value given p</span></span><br><span class="line">    l=spg.search_l(p, adj, start=<span class="number">0.01</span>, end=<span class="number">1000</span>, tol=<span class="number">0.01</span>, max_run=<span class="number">100</span>)</span><br><span class="line"></span><br><span class="line">    n_clusters=<span class="number">7</span></span><br><span class="line">    <span class="comment">#Set seed</span></span><br><span class="line">    r_seed=t_seed=n_seed=<span class="number">100</span></span><br><span class="line">    <span class="comment">#Seaech for suitable resolution</span></span><br><span class="line">    res=spg.search_res(adata, adj, l, n_clusters, start=<span class="number">0.7</span>, step=<span class="number">0.1</span>, tol=<span class="number">5e-3</span>, lr=<span class="number">0.05</span>, max_epochs=<span class="number">20</span>, r_seed=r_seed, t_seed=t_seed, n_seed=n_seed)</span><br><span class="line"></span><br><span class="line">    <span class="comment">#run SpaGCN</span></span><br><span class="line">    clf=spg.SpaGCN()</span><br><span class="line">    clf.set_l(l)</span><br><span class="line">    <span class="comment">#Set seed</span></span><br><span class="line">    random.seed(r_seed)</span><br><span class="line">    torch.manual_seed(t_seed)</span><br><span class="line">    np.random.seed(n_seed)</span><br><span class="line">    <span class="comment">#Run</span></span><br><span class="line">    clf.train(adata,adj,init_spa=<span class="literal">True</span>,init=<span class="string">&quot;louvain&quot;</span>,res=res, tol=<span class="number">5e-3</span>, lr=<span class="number">0.05</span>, max_epochs=<span class="number">200</span>)</span><br><span class="line">    y_pred, prob=clf.predict()</span><br><span class="line"></span><br><span class="line">    adata.obs[<span class="string">&quot;pred&quot;</span>]= y_pred</span><br><span class="line">    adata.obs[<span class="string">&quot;pred&quot;</span>]=adata.obs[<span class="string">&quot;pred&quot;</span>].astype(<span class="string">&#x27;category&#x27;</span>)</span><br><span class="line"></span><br><span class="line">    <span class="keyword">from</span> sklearn <span class="keyword">import</span> metrics</span><br><span class="line">    obs_df = adata.obs.dropna()</span><br><span class="line">    ari = metrics.adjusted_rand_score(obs_df[<span class="string">&#x27;pred&#x27;</span>], obs_df[<span class="string">&#x27;Region&#x27;</span>])</span><br><span class="line">   </span><br><span class="line">    ari_list[j]=ari</span><br><span class="line"></span><br><span class="line">    end_time=time.time()</span><br><span class="line">    during=end_time-start_time</span><br><span class="line">    size, peak = tracemalloc.get_traced_memory()</span><br><span class="line"></span><br><span class="line">    tracemalloc.stop()</span><br><span class="line"></span><br><span class="line">    memory[j]=peak /<span class="number">1024</span>/<span class="number">1024</span></span><br><span class="line">    during_time[j]=during</span><br><span class="line">    </span><br><span class="line"></span><br><span class="line">    <span class="built_in">print</span>(<span class="string">&#x27;ARI score: &#123;:.3f&#125;&#x27;</span>.<span class="built_in">format</span>(ari_list[j]))</span><br><span class="line">    <span class="built_in">print</span>(<span class="string">&#x27;memory blocks peak:&#123;:&gt;10.4f&#125; MB&#x27;</span>.<span class="built_in">format</span>(memory[j]))</span><br><span class="line">    <span class="built_in">print</span>(<span class="string">&#x27;time: &#123;:.4f&#125; s&#x27;</span>.<span class="built_in">format</span>(during_time[j]))</span><br><span class="line"></span><br><span class="line"></span><br><span class="line">    new_data.obs[<span class="string">&#x27;pred_&#123;&#125;&#x27;</span>.<span class="built_in">format</span>(j+<span class="number">1</span>)]=adata.obs[<span class="string">&#x27;pred&#x27;</span>]</span><br><span class="line"></span><br><span class="line">result_path=<span class="string">&#x27;./&#x27;</span></span><br><span class="line">new_data.uns[<span class="string">&#x27;time&#x27;</span>]=during_time</span><br><span class="line">new_data.uns[<span class="string">&#x27;memory&#x27;</span>]=memory</span><br><span class="line">new_data.uns[<span class="string">&#x27;ari&#x27;</span>]=ari_list</span><br><span class="line">new_data.write(<span class="string">f&#x27;<span class="subst">&#123;result_path&#125;</span>/SpaGCN_151673.h5ad&#x27;</span>)</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br></pre></td></tr></table></figure><h3 id="调用的代码"><a class="anchor" href="#调用的代码">#</a> 调用的代码</h3><h4 id="prefilter_genes"><a class="anchor" href="#prefilter_genes">#</a> prefilter_genes</h4><figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">def</span> <span class="title function_">prefilter_genes</span>(<span class="params">adata,min_counts=<span class="literal">None</span>,max_counts=<span class="literal">None</span>,min_cells=<span class="number">10</span>,max_cells=<span class="literal">None</span></span>):</span><br><span class="line">    <span class="keyword">if</span> min_cells <span class="keyword">is</span> <span class="literal">None</span> <span class="keyword">and</span> min_counts <span class="keyword">is</span> <span class="literal">None</span> <span class="keyword">and</span> max_cells <span class="keyword">is</span> <span class="literal">None</span> <span class="keyword">and</span> max_counts <span class="keyword">is</span> <span class="literal">None</span>:</span><br><span class="line">        <span class="keyword">raise</span> ValueError(<span class="string">&#x27;Provide one of min_counts, min_genes, max_counts or max_genes.&#x27;</span>)</span><br><span class="line">    id_tmp=np.asarray([<span class="literal">True</span>]*adata.shape[<span class="number">1</span>],dtype=<span class="built_in">bool</span>)</span><br><span class="line">    <span class="comment"># 创建一个初始值为真的函数</span></span><br><span class="line">    id_tmp=np.logical_and(id_tmp,sc.pp.filter_genes(adata.X,min_cells=min_cells)[<span class="number">0</span>]) <span class="keyword">if</span> min_cells <span class="keyword">is</span> <span class="keyword">not</span> <span class="literal">None</span>  <span class="keyword">else</span> id_tmp</span><br><span class="line">    id_tmp=np.logical_and(id_tmp,sc.pp.filter_genes(adata.X,max_cells=max_cells)[<span class="number">0</span>]) <span class="keyword">if</span> max_cells <span class="keyword">is</span> <span class="keyword">not</span> <span class="literal">None</span>  <span class="keyword">else</span> id_tmp</span><br><span class="line">    id_tmp=np.logical_and(id_tmp,sc.pp.filter_genes(adata.X,min_counts=min_counts)[<span class="number">0</span>]) <span class="keyword">if</span> min_counts <span class="keyword">is</span> <span class="keyword">not</span> <span class="literal">None</span>  <span class="keyword">else</span> id_tmp</span><br><span class="line">    id_tmp=np.logical_and(id_tmp,sc.pp.filter_genes(adata.X,max_counts=max_counts)[<span class="number">0</span>]) <span class="keyword">if</span> max_counts <span class="keyword">is</span> <span class="keyword">not</span> <span class="literal">None</span>  <span class="keyword">else</span> id_tmp</span><br><span class="line">    adata._inplace_subset_var(id_tmp)</span><br></pre></td></tr></table></figure><p>对于 sc.pp.filter_genes 的解释：<br>数据预处理<br>这里介绍一下 scanpy 中常用的组件</p><p>pp: 数据预处理</p><p>tl: 添加额外信息</p><p>pl：可视化</p><figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">sc.pl.highest_expr_genes(adata, n_top=<span class="number">20</span>) <span class="comment"># 每一个基因在所有细胞中的平均表达量（这里计算了百分比含量）</span></span><br><span class="line">sc.pp.filter_cells(adata, min_genes=<span class="number">200</span>) <span class="comment"># 每一个细胞至少表达200个基因</span></span><br><span class="line">sc.pp.filter_genes(adata, min_cells=<span class="number">3</span>) <span class="comment"># 每一个基因至少在3个细胞中表达</span></span><br></pre></td></tr></table></figure><h4 id="prefilter_specialgenes"><a class="anchor" href="#prefilter_specialgenes">#</a> prefilter_specialgenes</h4><figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">def</span> <span class="title function_">prefilter_specialgenes</span>(<span class="params">adata,Gene1Pattern=<span class="string">&quot;ERCC&quot;</span>,Gene2Pattern=<span class="string">&quot;MT-&quot;</span></span>):</span><br><span class="line">    id_tmp1=np.asarray([<span class="keyword">not</span> <span class="built_in">str</span>(name).startswith(Gene1Pattern) <span class="keyword">for</span> name <span class="keyword">in</span> adata.var_names],dtype=<span class="built_in">bool</span>)</span><br><span class="line">    id_tmp2=np.asarray([<span class="keyword">not</span> <span class="built_in">str</span>(name).startswith(Gene2Pattern) <span class="keyword">for</span> name <span class="keyword">in</span> adata.var_names],dtype=<span class="built_in">bool</span>)</span><br><span class="line">    id_tmp=np.logical_and(id_tmp1,id_tmp2)</span><br><span class="line">    adata._inplace_subset_var(id_tmp)</span><br></pre></td></tr></table></figure><p>过滤掉以 ERCC 和 MT - 开头的基因</p><ul><li>&quot;ERCC&quot; 是指外源对照基因（External RNA Controls Consortium），通常用于实验控制。</li><li>&quot;MT-&quot; 是指线粒体基因（Mitochondrial genes），通常用于质量控制。</li></ul><h4 id="search_l"><a class="anchor" href="#search_l">#</a> search_l</h4><figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">def</span> <span class="title function_">search_l</span>(<span class="params">p, adj, start=<span class="number">0.01</span>, end=<span class="number">1000</span>, tol=<span class="number">0.01</span>, max_run=<span class="number">100</span></span>):</span><br><span class="line">    run=<span class="number">0</span></span><br><span class="line">    p_low=calculate_p(adj, start)</span><br><span class="line">    p_high=calculate_p(adj, end)</span><br><span class="line">    <span class="comment"># 计算高斯核权重 用于将距离值转换为相应的权重，较小的距离得到较大的权重，较大的距离得到较小的权重。</span></span><br><span class="line">    <span class="keyword">if</span> p_low&gt;p+tol:</span><br><span class="line">        <span class="built_in">print</span>(<span class="string">&quot;l not found, try smaller start point.&quot;</span>)</span><br><span class="line">        <span class="keyword">return</span> <span class="literal">None</span></span><br><span class="line">    <span class="keyword">elif</span> p_high&lt;p-tol:</span><br><span class="line">        <span class="built_in">print</span>(<span class="string">&quot;l not found, try bigger end point.&quot;</span>)</span><br><span class="line">        <span class="keyword">return</span> <span class="literal">None</span></span><br><span class="line">    <span class="keyword">elif</span>  np.<span class="built_in">abs</span>(p_low-p) &lt;=tol:</span><br><span class="line">        <span class="built_in">print</span>(<span class="string">&quot;recommended l = &quot;</span>, <span class="built_in">str</span>(start))</span><br><span class="line">        <span class="keyword">return</span> start</span><br><span class="line">    <span class="keyword">elif</span>  np.<span class="built_in">abs</span>(p_high-p) &lt;=tol:</span><br><span class="line">        <span class="built_in">print</span>(<span class="string">&quot;recommended l = &quot;</span>, <span class="built_in">str</span>(end))</span><br><span class="line">        <span class="keyword">return</span> end</span><br><span class="line">    <span class="keyword">while</span> (p_low+tol)&lt;p&lt;(p_high-tol):</span><br><span class="line">        run+=<span class="number">1</span></span><br><span class="line">        <span class="built_in">print</span>(<span class="string">&quot;Run &quot;</span>+<span class="built_in">str</span>(run)+<span class="string">&quot;: l [&quot;</span>+<span class="built_in">str</span>(start)+<span class="string">&quot;, &quot;</span>+<span class="built_in">str</span>(end)+<span class="string">&quot;], p [&quot;</span>+<span class="built_in">str</span>(p_low)+<span class="string">&quot;, &quot;</span>+<span class="built_in">str</span>(p_high)+<span class="string">&quot;]&quot;</span>)</span><br><span class="line">        <span class="keyword">if</span> run &gt;max_run:</span><br><span class="line">            <span class="built_in">print</span>(<span class="string">&quot;Exact l not found, closest values are:\n&quot;</span>+<span class="string">&quot;l=&quot;</span>+<span class="built_in">str</span>(start)+<span class="string">&quot;: &quot;</span>+<span class="string">&quot;p=&quot;</span>+<span class="built_in">str</span>(p_low)+<span class="string">&quot;\nl=&quot;</span>+<span class="built_in">str</span>(end)+<span class="string">&quot;: &quot;</span>+<span class="string">&quot;p=&quot;</span>+<span class="built_in">str</span>(p_high))</span><br><span class="line">            <span class="keyword">return</span> <span class="literal">None</span></span><br><span class="line">        mid=(start+end)/<span class="number">2</span></span><br><span class="line">        p_mid=calculate_p(adj, mid)</span><br><span class="line">        <span class="keyword">if</span> np.<span class="built_in">abs</span>(p_mid-p)&lt;=tol:</span><br><span class="line">            <span class="built_in">print</span>(<span class="string">&quot;recommended l = &quot;</span>, <span class="built_in">str</span>(mid))</span><br><span class="line">            <span class="keyword">return</span> mid</span><br><span class="line">        <span class="keyword">if</span> p_mid&lt;=p:</span><br><span class="line">            start=mid</span><br><span class="line">            p_low=p_mid</span><br><span class="line">        <span class="keyword">else</span>:</span><br><span class="line">            end=mid</span><br><span class="line">            p_high=p_mid</span><br></pre></td></tr></table></figure><h4 id="search_res"><a class="anchor" href="#search_res">#</a> search_res</h4><figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br><span class="line">42</span><br><span class="line">43</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">def</span> <span class="title function_">search_res</span>(<span class="params">adata, adj, l, target_num, start=<span class="number">0.4</span>, step=<span class="number">0.1</span>, tol=<span class="number">5e-3</span>, lr=<span class="number">0.05</span>, max_epochs=<span class="number">10</span>, r_seed=<span class="number">100</span>, t_seed=<span class="number">100</span>, n_seed=<span class="number">100</span>, max_run=<span class="number">10</span></span>):</span><br><span class="line">    random.seed(r_seed)</span><br><span class="line">    torch.manual_seed(t_seed)</span><br><span class="line">    np.random.seed(n_seed)</span><br><span class="line">    res=start</span><br><span class="line">    <span class="built_in">print</span>(<span class="string">&quot;Start at res = &quot;</span>, res, <span class="string">&quot;step = &quot;</span>, step)</span><br><span class="line">    clf=SpaGCN()</span><br><span class="line">    clf.set_l(l)</span><br><span class="line">    clf.train(adata,adj,init_spa=<span class="literal">True</span>,init=<span class="string">&quot;louvain&quot;</span>,res=res, tol=tol, lr=lr, max_epochs=max_epochs)</span><br><span class="line">    y_pred, _=clf.predict()</span><br><span class="line">    old_num=<span class="built_in">len</span>(<span class="built_in">set</span>(y_pred))</span><br><span class="line">    <span class="built_in">print</span>(<span class="string">&quot;Res = &quot;</span>, res, <span class="string">&quot;Num of clusters = &quot;</span>, old_num)</span><br><span class="line">    run=<span class="number">0</span></span><br><span class="line">    <span class="keyword">while</span> old_num!=target_num:</span><br><span class="line">        random.seed(r_seed)</span><br><span class="line">        torch.manual_seed(t_seed)</span><br><span class="line">        np.random.seed(n_seed)</span><br><span class="line">        old_sign=<span class="number">1</span> <span class="keyword">if</span> (old_num&lt;target_num) <span class="keyword">else</span> -<span class="number">1</span></span><br><span class="line">        clf=SpaGCN()</span><br><span class="line">        clf.set_l(l)</span><br><span class="line">        clf.train(adata,adj,init_spa=<span class="literal">True</span>,init=<span class="string">&quot;louvain&quot;</span>,res=res+step*old_sign, tol=tol, lr=lr, max_epochs=max_epochs)</span><br><span class="line">        y_pred, _=clf.predict()</span><br><span class="line">        new_num=<span class="built_in">len</span>(<span class="built_in">set</span>(y_pred))</span><br><span class="line">        <span class="built_in">print</span>(<span class="string">&quot;Res = &quot;</span>, res+step*old_sign, <span class="string">&quot;Num of clusters = &quot;</span>, new_num)</span><br><span class="line">        <span class="keyword">if</span> new_num==target_num:</span><br><span class="line">            res=res+step*old_sign</span><br><span class="line">            <span class="built_in">print</span>(<span class="string">&quot;recommended res = &quot;</span>, <span class="built_in">str</span>(res))</span><br><span class="line">            <span class="keyword">return</span> res</span><br><span class="line">        new_sign=<span class="number">1</span> <span class="keyword">if</span> (new_num&lt;target_num) <span class="keyword">else</span> -<span class="number">1</span></span><br><span class="line">        <span class="keyword">if</span> new_sign==old_sign:</span><br><span class="line">            res=res+step*old_sign</span><br><span class="line">            <span class="built_in">print</span>(<span class="string">&quot;Res changed to&quot;</span>, res)</span><br><span class="line">            old_num=new_num</span><br><span class="line">        <span class="keyword">else</span>:</span><br><span class="line">            step=step/<span class="number">2</span></span><br><span class="line">            <span class="built_in">print</span>(<span class="string">&quot;Step changed to&quot;</span>, step)</span><br><span class="line">        <span class="keyword">if</span> run &gt;max_run:</span><br><span class="line">            <span class="built_in">print</span>(<span class="string">&quot;Exact resolution not found&quot;</span>)</span><br><span class="line">            <span class="built_in">print</span>(<span class="string">&quot;Recommended res = &quot;</span>, <span class="built_in">str</span>(res))</span><br><span class="line">            <span class="keyword">return</span> res</span><br><span class="line">        run+=<span class="number">1</span></span><br><span class="line">    <span class="built_in">print</span>(<span class="string">&quot;recommended res = &quot;</span>, <span class="built_in">str</span>(res))</span><br><span class="line">    <span class="keyword">return</span> res</span><br></pre></td></tr></table></figure><h4 id="spagcn里面的train函数"><a class="anchor" href="#spagcn里面的train函数">#</a> SpaGCN 里面的 train 函数</h4><figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">def</span> <span class="title function_">train</span>(<span class="params">self,adata,adj, </span></span><br><span class="line"><span class="params">        num_pcs=<span class="number">50</span>, </span></span><br><span class="line"><span class="params">        lr=<span class="number">0.005</span>,</span></span><br><span class="line"><span class="params">        max_epochs=<span class="number">2000</span>,</span></span><br><span class="line"><span class="params">        weight_decay=<span class="number">0</span>,</span></span><br><span class="line"><span class="params">        opt=<span class="string">&quot;admin&quot;</span>,</span></span><br><span class="line"><span class="params">        init_spa=<span class="literal">True</span>,</span></span><br><span class="line"><span class="params">        init=<span class="string">&quot;louvain&quot;</span>, <span class="comment">#louvain or kmeans</span></span></span><br><span class="line"><span class="params">        n_neighbors=<span class="number">10</span>, <span class="comment">#for louvain</span></span></span><br><span class="line"><span class="params">        n_clusters=<span class="literal">None</span>, <span class="comment">#for kmeans</span></span></span><br><span class="line"><span class="params">        res=<span class="number">0.4</span>, <span class="comment">#for louvain</span></span></span><br><span class="line"><span class="params">        tol=<span class="number">1e-3</span></span>):</span><br><span class="line">    <span class="variable language_">self</span>.num_pcs=num_pcs</span><br><span class="line">    <span class="variable language_">self</span>.res=res</span><br><span class="line">    <span class="variable language_">self</span>.lr=lr</span><br><span class="line">    <span class="variable language_">self</span>.max_epochs=max_epochs</span><br><span class="line">    <span class="variable language_">self</span>.weight_decay=weight_decay</span><br><span class="line">    <span class="variable language_">self</span>.opt=opt</span><br><span class="line">    <span class="variable language_">self</span>.init_spa=init_spa</span><br><span class="line">    <span class="variable language_">self</span>.init=init</span><br><span class="line">    <span class="variable language_">self</span>.n_neighbors=n_neighbors</span><br><span class="line">    <span class="variable language_">self</span>.n_clusters=n_clusters</span><br><span class="line">    <span class="variable language_">self</span>.res=res</span><br><span class="line">    <span class="variable language_">self</span>.tol=tol</span><br><span class="line">    <span class="keyword">assert</span> adata.shape[<span class="number">0</span>]==adj.shape[<span class="number">0</span>]==adj.shape[<span class="number">1</span>]</span><br><span class="line">    pca = PCA(n_components=<span class="variable language_">self</span>.num_pcs)</span><br><span class="line">    <span class="keyword">if</span> issparse(adata.X):</span><br><span class="line">        pca.fit(adata.X.A)</span><br><span class="line">        embed=pca.transform(adata.X.A)</span><br><span class="line">    <span class="keyword">else</span>:</span><br><span class="line">        pca.fit(adata.X)</span><br><span class="line">        embed=pca.transform(adata.X)</span><br><span class="line">    <span class="comment">###------------------------------------------###</span></span><br><span class="line">    <span class="keyword">if</span> <span class="variable language_">self</span>.l <span class="keyword">is</span> <span class="literal">None</span>:</span><br><span class="line">        <span class="keyword">raise</span> ValueError(<span class="string">&#x27;l should not be set before fitting the model!&#x27;</span>)</span><br><span class="line">    adj_exp=np.exp(-<span class="number">1</span>*(adj**<span class="number">2</span>)/(<span class="number">2</span>*(<span class="variable language_">self</span>.l**<span class="number">2</span>)))</span><br><span class="line">    <span class="comment">#----------Train model----------</span></span><br><span class="line">    <span class="variable language_">self</span>.model=simple_GC_DEC(embed.shape[<span class="number">1</span>],embed.shape[<span class="number">1</span>])</span><br><span class="line">    <span class="variable language_">self</span>.model.fit(embed,adj_exp,lr=<span class="variable language_">self</span>.lr,max_epochs=<span class="variable language_">self</span>.max_epochs,weight_decay=<span class="variable language_">self</span>.weight_decay,opt=<span class="variable language_">self</span>.opt,init_spa=<span class="variable language_">self</span>.init_spa,init=<span class="variable language_">self</span>.init,n_neighbors=<span class="variable language_">self</span>.n_neighbors,n_clusters=<span class="variable language_">self</span>.n_clusters,res=<span class="variable language_">self</span>.res, tol=<span class="variable language_">self</span>.tol)</span><br><span class="line">    <span class="variable language_">self</span>.embed=embed</span><br><span class="line">    <span class="variable language_">self</span>.adj_exp=adj_exp</span><br></pre></td></tr></table></figure><h4 id="predict"><a class="anchor" href="#predict">#</a> predict</h4><figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">def</span> <span class="title function_">predict</span>(<span class="params">self</span>):</span><br><span class="line">    z,q=<span class="variable language_">self</span>.model.predict(<span class="variable language_">self</span>.embed,<span class="variable language_">self</span>.adj_exp)</span><br><span class="line">    y_pred = torch.argmax(q, dim=<span class="number">1</span>).data.cpu().numpy()</span><br><span class="line">    <span class="comment"># Max probability plot</span></span><br><span class="line">    prob=q.detach().numpy()</span><br><span class="line">    <span class="keyword">return</span> y_pred, prob</span><br></pre></td></tr></table></figure><div class="tags"><a href="/tags/%E7%94%9F%E7%89%A9/" rel="tag"><i class="ic i-tag"></i> 生物</a> <a href="/tags/%E5%AD%A6%E4%B9%A0%E7%AC%94%E8%AE%B0%E7%B1%BB/" rel="tag"><i class="ic i-tag"></i> 学习笔记类</a> <a href="/tags/%E7%AE%97%E6%B3%95/" rel="tag"><i class="ic i-tag"></i> 算法</a></div></div><footer><div class="meta"><span class="item"><span class="icon"><i class="ic i-calendar-check"></i> </span><span class="text">更新于</span> <time title="修改时间：2025-04-30 10:51:14" itemprop="dateModified" datetime="2025-04-30T10:51:14+08:00">2025-04-30</time> </span><span id="article/Benchmarking-spatial-claustering-methods-with-spatially-resolved-transcriptomic-data/" class="item leancloud_visitors" data-flag-title="Benchmarking-spatial-claustering-methods-with-spatially-resolved-transcriptomic-data" title="阅读次数"><span class="icon"><i class="ic i-eye"></i> </span><span class="text">阅读次数</span> <span class="leancloud-visitors-count"></span> <span class="text">次</span></span></div><div id="copyright"><ul><li class="author"><strong>本文作者： </strong>Lemon Sour <i class="ic i-at"><em>@</em></i></li><li class="link"><strong>本文链接：</strong> <a href="http://amentiraz.github.io/article/Benchmarking-spatial-claustering-methods-with-spatially-resolved-transcriptomic-data/" title="Benchmarking-spatial-claustering-methods-with-spatially-resolved-transcriptomic-data">http://amentiraz.github.io/article/Benchmarking-spatial-claustering-methods-with-spatially-resolved-transcriptomic-data/</a></li><li class="license"><strong>版权声明： </strong>本站所有文章除特别声明外，均采用 <span class="exturl" data-url="aHR0cHM6Ly9jcmVhdGl2ZWNvbW1vbnMub3JnL2xpY2Vuc2VzL2J5LW5jLXNhLzQuMC96aC1DTg=="><i class="ic i-creative-commons"><em>(CC)</em></i>BY-NC-SA</span> 许可协议。转载请注明出处！</li></ul></div></footer></article></div><div class="post-nav"><div class="item left"><a href="/music/%E5%90%89%E4%BB%96%E4%B9%90%E7%90%86-RnB%E7%B1%BB/" itemprop="url" rel="prev" data-background-image="https:&#x2F;&#x2F;amentirazblogpic.oss-cn-hangzhou.aliyuncs.com&#x2F;blogpic&#x2F;%E5%BE%AE%E4%BF%A1%E5%9B%BE%E7%89%87_20240925162718.png" title="吉他乐理-RnB类"><span class="type">上一篇</span> <span class="category"><i class="ic i-flag"></i> 乐理</span><h3>吉他乐理-RnB类</h3></a></div><div class="item right"><a href="/life/%E8%A7%81%E5%AD%97%E5%A6%82%E9%9D%A2/" itemprop="url" rel="next" data-background-image="https:&#x2F;&#x2F;amentirazblogpic.oss-cn-hangzhou.aliyuncs.com&#x2F;blogpic&#x2F;%E5%BE%AE%E4%BF%A1%E5%9B%BE%E7%89%87_20240925162836.png" title="见字如面"><span class="type">下一篇</span> <span class="category"><i class="ic i-flag"></i> 似水流年</span><h3>见字如面</h3></a></div></div><div class="wrap" id="comments"></div></div><div id="sidebar"><div class="inner"><div class="panels"><div class="inner"><div class="contents panel pjax" data-title="文章目录"><ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#%E5%8D%95%E8%AF%8D%E8%A1%A8"><span class="toc-number">1.</span> <span class="toc-text">单词表</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#%E6%96%87%E7%AB%A0%E6%A6%82%E6%8B%AC"><span class="toc-number">2.</span> <span class="toc-text">文章概括</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#limitations"><span class="toc-number">2.1.</span> <span class="toc-text">Limitations</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#a-divide-and-conquer-strategy-enables-large-scale-scalability"><span class="toc-number">2.2.</span> <span class="toc-text">A divide and conquer strategy enables large-scale scalability</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#%E6%A0%B8%E5%BF%83%E6%A6%82%E5%BF%B5%E8%A7%A3%E9%87%8A"><span class="toc-number">2.2.1.</span> <span class="toc-text">核心概念解释：</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#pre-and-postpocessing"><span class="toc-number">2.3.</span> <span class="toc-text">pre- and postpocessing</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#1-%E9%A2%84%E5%A4%84%E7%90%86%E5%92%8C%E5%90%8E%E5%A4%84%E7%90%86%E5%AF%B9%E6%80%A7%E8%83%BD%E7%9A%84%E5%BD%B1%E5%93%8D"><span class="toc-number">2.3.1.</span> <span class="toc-text">1. 预处理和后处理对性能的影响：</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#2-%E6%95%B0%E6%8D%AE%E7%89%B9%E5%BE%81%E5%AF%B9%E6%96%B9%E6%B3%95%E9%B2%81%E6%A3%92%E6%80%A7%E7%9A%84%E5%BD%B1%E5%93%8D"><span class="toc-number">2.3.2.</span> <span class="toc-text">2. 数据特征对方法鲁棒性的影响：</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#discussion"><span class="toc-number">2.4.</span> <span class="toc-text">Discussion</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#1-%E7%A0%94%E7%A9%B6%E6%A6%82%E8%BF%B0"><span class="toc-number">2.4.1.</span> <span class="toc-text">1. 研究概述：</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#2-%E6%96%B9%E6%B3%95%E7%9A%84%E5%B1%80%E9%99%90%E6%80%A7%E4%B8%8E%E6%94%B9%E8%BF%9B"><span class="toc-number">2.4.2.</span> <span class="toc-text">2. 方法的局限性与改进：</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#3-%E9%A2%9D%E5%A4%96%E6%95%B0%E6%8D%AE%E7%9A%84%E5%BC%95%E5%85%A5"><span class="toc-number">2.4.3.</span> <span class="toc-text">3. 额外数据的引入：</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#4-%E7%A0%94%E7%A9%B6%E7%9A%84%E5%B1%80%E9%99%90%E6%80%A7%E4%B8%8E%E6%9C%AA%E6%9D%A5%E6%96%B9%E5%90%91"><span class="toc-number">2.4.4.</span> <span class="toc-text">4. 研究的局限性与未来方向：</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#5-%E8%AF%84%E4%BB%B7%E6%8C%87%E6%A0%87"><span class="toc-number">2.4.5.</span> <span class="toc-text">5. 评价指标：</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#method"><span class="toc-number">2.5.</span> <span class="toc-text">Method</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#%E7%A9%BA%E9%97%B4%E8%81%9A%E7%B1%BB%E6%96%B9%E6%B3%95"><span class="toc-number">2.5.1.</span> <span class="toc-text">空间聚类方法：</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#simulated-data"><span class="toc-number">2.5.2.</span> <span class="toc-text">Simulated data</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#benchmark-metrics"><span class="toc-number">2.5.3.</span> <span class="toc-text">Benchmark metrics</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#nmi"><span class="toc-number">2.5.3.1.</span> <span class="toc-text">NMI</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#ari"><span class="toc-number">2.5.3.2.</span> <span class="toc-text">ARI</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#homhomogeneity-score"><span class="toc-number">2.5.3.3.</span> <span class="toc-text">HOM（Homogeneity Score）：</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#comcompleteness-score"><span class="toc-number">2.5.3.4.</span> <span class="toc-text">COM（Completeness Score）：</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#chaos"><span class="toc-number">2.5.3.5.</span> <span class="toc-text">CHAOS</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#paspercentage-of-area-shared"><span class="toc-number">2.5.3.6.</span> <span class="toc-text">PAS（Percentage of Area Shared）：</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#aswaverage-silhouette-width"><span class="toc-number">2.5.3.7.</span> <span class="toc-text">ASW（Average Silhouette Width）：</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#morans-i"><span class="toc-number">2.5.3.8.</span> <span class="toc-text">Moran&#39;s I</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#gearys-c"><span class="toc-number">2.5.3.9.</span> <span class="toc-text">Geary&#39;s C</span></a></li></ol></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E4%BB%A3%E7%A0%81%E5%AE%9E%E7%8E%B0"><span class="toc-number">2.6.</span> <span class="toc-text">代码实现</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#%E4%B8%BB%E4%BB%A3%E7%A0%81"><span class="toc-number">2.6.1.</span> <span class="toc-text">主代码</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#%E8%B0%83%E7%94%A8%E7%9A%84%E4%BB%A3%E7%A0%81"><span class="toc-number">2.6.2.</span> <span class="toc-text">调用的代码</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#prefilter_genes"><span class="toc-number">2.6.2.1.</span> <span class="toc-text">prefilter_genes</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#prefilter_specialgenes"><span class="toc-number">2.6.2.2.</span> <span class="toc-text">prefilter_specialgenes</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#search_l"><span class="toc-number">2.6.2.3.</span> <span class="toc-text">search_l</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#search_res"><span class="toc-number">2.6.2.4.</span> <span class="toc-text">search_res</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#spagcn%E9%87%8C%E9%9D%A2%E7%9A%84train%E5%87%BD%E6%95%B0"><span class="toc-number">2.6.2.5.</span> <span class="toc-text">SpaGCN 里面的 train 函数</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#predict"><span class="toc-number">2.6.2.6.</span> <span class="toc-text">predict</span></a></li></ol></li></ol></li></ol></li></ol></div><div class="related panel pjax" data-title="系列文章"><ul><li><a href="/article/scRNA%E8%AE%BA%E6%96%87%E7%AC%94%E8%AE%B0/" rel="bookmark" title="scRNA论文笔记">scRNA论文笔记</a></li><li><a href="/article/What-is-A-Cell-Type/" rel="bookmark" title="What_is_A_Cell_Type">What_is_A_Cell_Type</a></li><li><a href="/article/Cell-Review-What-is-a-cell-type-and-how-to-define-it/" rel="bookmark" title="Cell_Review_What_is_a_cell_type_and_how_to_define_it">Cell_Review_What_is_a_cell_type_and_how_to_define_it</a></li><li class="active"><a href="/article/Benchmarking-spatial-claustering-methods-with-spatially-resolved-transcriptomic-data/" rel="bookmark" title="Benchmarking-spatial-claustering-methods-with-spatially-resolved-transcriptomic-data">Benchmarking-spatial-claustering-methods-with-spatially-resolved-transcriptomic-data</a></li><li><a href="/article/%E8%AE%BA%E6%96%87ppt1/" rel="bookmark" title="论文ppt1">论文ppt1</a></li><li><a href="/article/Matching-Anything-by-Segmenting-Anything/" rel="bookmark" title="Matching_Anything_by_Segmenting_Anything">Matching_Anything_by_Segmenting_Anything</a></li><li><a href="/article/SpaGCN/" rel="bookmark" title="SpaGCN">SpaGCN</a></li><li><a href="/article/STAGATE/" rel="bookmark" title="STAGATE">STAGATE</a></li><li><a href="/article/MENDER/" rel="bookmark" title="MENDER">MENDER</a></li><li><a href="/article/scanpy%E6%95%B0%E6%8D%AE%E4%BD%BF%E7%94%A8%E8%AE%B0%E5%BD%95/" rel="bookmark" title="scanpy数据使用笔记">scanpy数据使用笔记</a></li><li><a href="/article/BayesSpace/" rel="bookmark" title="BayesSpace">BayesSpace</a></li><li><a href="/article/%E6%95%B0%E6%8D%AE%E9%9B%86%E8%AE%B0%E5%BD%95/" rel="bookmark" title="数据集记录">数据集记录</a></li><li><a href="/article/DeepST/" rel="bookmark" title="DeepST">DeepST</a></li><li><a href="/article/EnSDD/" rel="bookmark" title="EnSDD">EnSDD</a></li><li><a href="/article/domain%E5%86%85%E5%AE%B9%E7%9A%84%E6%80%BB%E7%BB%93/" rel="bookmark" title="对spatial domain内容的总结">对spatial domain内容的总结</a></li><li><a href="/article/%E6%AF%95%E4%B8%9A%E8%AE%BE%E8%AE%A1%E6%96%87%E6%A1%A3%E6%B1%87%E6%80%BB/" rel="bookmark" title="毕业设计文档汇总">毕业设计文档汇总</a></li><li><a href="/article/scPerturb/" rel="bookmark" title="scPerturb">scPerturb</a></li><li><a href="/GEARS/" rel="bookmark" title="GEARS">GEARS</a></li><li><a href="/CellOT/" rel="bookmark" title="CellOT">CellOT</a></li><li><a href="/Transport/" rel="bookmark" title="Optimal Transport">Optimal Transport</a></li><li><a href="/Simple-Perform-better/" rel="bookmark" title="Simple_Perform_better">Simple_Perform_better</a></li><li><a href="/Systema/" rel="bookmark" title="Systema">Systema</a></li><li><a href="/Virtual-Challenge/" rel="bookmark" title="Virtual_Challenge">Virtual_Challenge</a></li></ul></div><div class="overview panel" data-title="站点概览"><div class="author" itemprop="author" itemscope itemtype="http://schema.org/Person"><img class="image" itemprop="image" alt="Lemon Sour" data-src="/images/avatar.jpg"><p class="name" itemprop="name">Lemon Sour</p><div class="description" itemprop="description"></div></div><nav class="state"><div class="item posts"><a href="/archives/"><span class="count">93</span> <span class="name">文章</span></a></div><div class="item categories"><a href="/categories/"><span class="count">26</span> <span class="name">分类</span></a></div><div class="item tags"><a href="/tags/"><span class="count">57</span> <span class="name">标签</span></a></div></nav><div class="social"><span class="exturl item github" data-url="aHR0cHM6Ly9naXRodWIuY29tL0FtZW50aXJheg==" title="https:&#x2F;&#x2F;github.com&#x2F;Amentiraz"><i class="ic i-github"></i></span> <span class="exturl item zhihu" data-url="aHR0cHM6Ly93d3cuemhpaHUuY29tL3Blb3BsZS9kc2ZseS04" title="https:&#x2F;&#x2F;www.zhihu.com&#x2F;people&#x2F;dsfly-8"><i class="ic i-zhihu"></i></span> <span class="exturl item music" data-url="aHR0cHM6Ly9tdXNpYy4xNjMuY29tLyMvdXNlci9ob21lP2lkPTE1MTc2ODUzMzM=" title="https:&#x2F;&#x2F;music.163.com&#x2F;#&#x2F;user&#x2F;home?id&#x3D;1517685333"><i class="ic i-cloud-music"></i></span> <span class="exturl item email" data-url="bWFpbHRvOnZpb2xlbW9uQDE2My5jb20=" title="mailto:violemon@163.com"><i class="ic i-envelope"></i></span> <span class="exturl item bangumi" data-url="aHR0cHM6Ly9iYW5ndW1pLnR2L2FuaW1lL2xpc3QvNjY4MDE2" title="https:&#x2F;&#x2F;bangumi.tv&#x2F;anime&#x2F;list&#x2F;668016"><i class="ic i-bilibili"></i></span></div><ul class="menu"><li class="item"><a href="/" rel="section"><i class="ic i-home"></i>首页</a></li><li class="item dropdown"><a href="javascript:void(0);"><i class="ic i-magic"></i>关于</a><ul class="submenu"><li class="item"><a href="/author/" rel="section"><i class="ic i-user"></i>本人</a></li></ul></li><li class="item dropdown"><a href="javascript:void(0);"><i class="ic i-feather"></i>文章</a><ul class="submenu"><li class="item"><a href="/archives/" rel="section"><i class="ic i-list-alt"></i>归档</a></li><li class="item"><a href="/categories/" rel="section"><i class="ic i-th"></i>分类</a></li><li class="item"><a href="/tags/" rel="section"><i class="ic i-tags"></i>标签</a></li></ul></li><li class="item dropdown"><a href="javascript:void(0);"><i class="ic i-cloud"></i>其它</a><ul class="submenu"><li class="item"><a href="/music/" rel="section"><i class="ic i-music"></i>音乐区</a></li><li class="item"><a href="/friends/" rel="section"><i class="ic i-heart"></i>朋友</a></li></ul></li></ul></div></div></div><ul id="quick"><li class="prev pjax"><a href="/music/%E5%90%89%E4%BB%96%E4%B9%90%E7%90%86-RnB%E7%B1%BB/" rel="prev" title="上一篇"><i class="ic i-chevron-left"></i></a></li><li class="up"><i class="ic i-arrow-up"></i></li><li class="down"><i class="ic i-arrow-down"></i></li><li class="next pjax"><a href="/life/%E8%A7%81%E5%AD%97%E5%A6%82%E9%9D%A2/" rel="next" title="下一篇"><i class="ic i-chevron-right"></i></a></li><li class="percent"></li></ul></div></div><div class="dimmer"></div></div></main><footer id="footer"><div class="inner"><div class="widgets"><div class="rpost pjax"><h2>随机文章</h2><ul><li class="item"><div class="breadcrumb"><a href="/categories/life/" title="分类于 生活">生活</a> <i class="ic i-angle-right"></i> <a href="/categories/life/%E6%80%9D%E8%80%83/" title="分类于 思考">思考</a></div><span><a href="/life/%E5%85%B3%E4%BA%8E%E9%9D%A2%E5%AD%94%E7%9A%84%E5%90%AB%E4%B9%89/" title="关于面孔的含义">关于面孔的含义</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/note/" title="分类于 影视书籍">影视书籍</a> <i class="ic i-angle-right"></i> <a href="/categories/note/%E4%B9%A6/" title="分类于 书">书</a></div><span><a href="/note/%E6%83%85%E7%88%B1%E7%8E%B0%E8%B1%A1%E5%AD%A6%E6%9D%BF%E5%9D%971%E6%80%BB%E7%BB%93/" title="情爱现象学板块1总结">情爱现象学板块1总结</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/code/" title="分类于 代码">代码</a> <i class="ic i-angle-right"></i> <a href="/categories/code/python/" title="分类于 python">python</a></div><span><a href="/code/%E7%B3%BB%E7%BB%9F%E5%AD%A6%E4%B9%A0python4/" title="系统学习python（四）对于Scanpy的整理">系统学习python（四）对于Scanpy的整理</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/code/" title="分类于 代码">代码</a> <i class="ic i-angle-right"></i> <a href="/categories/code/%E7%BC%96%E7%A8%8B%E8%AF%AD%E8%A8%80/" title="分类于 编程语言">编程语言</a></div><span><a href="/code/c%E5%A4%8D%E4%B9%A0/" title="c复习">c复习</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/article/" title="分类于 论文">论文</a></div><span><a href="/Systema/" title="Systema">Systema</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/life/" title="分类于 生活">生活</a> <i class="ic i-angle-right"></i> <a href="/categories/life/%E6%BC%94%E5%A5%8F%E4%BC%9A/" title="分类于 演奏会">演奏会</a></div><span><a href="/life/MeetingJazz2025-06-01/" title="MeetingJazz2025-06-01">MeetingJazz2025-06-01</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/music/" title="分类于 音乐">音乐</a> <i class="ic i-angle-right"></i> <a href="/categories/music/%E4%B9%90%E7%90%86/" title="分类于 乐理">乐理</a></div><span><a href="/music/%E5%90%89%E4%BB%96%E4%B9%90%E7%90%86-POP/" title="吉他乐理-POP">吉他乐理-POP</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/music/" title="分类于 音乐">音乐</a> <i class="ic i-angle-right"></i> <a href="/categories/music/%E4%B9%90%E7%90%86/" title="分类于 乐理">乐理</a></div><span><a href="/music/%E5%90%89%E4%BB%96%E4%B9%90%E7%90%86-RnB%E7%B1%BB/" title="吉他乐理-RnB类">吉他乐理-RnB类</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/code/" title="分类于 代码">代码</a> <i class="ic i-angle-right"></i> <a href="/categories/code/%E7%BC%96%E7%A8%8B%E8%AF%AD%E8%A8%80/" title="分类于 编程语言">编程语言</a></div><span><a href="/code/%E6%96%87%E4%BB%B6%E5%AD%A6%E4%B9%A0%E7%AC%94%E8%AE%B0/" title="文件学习笔记">文件学习笔记</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/article/" title="分类于 论文">论文</a></div><span><a href="/article/Matching-Anything-by-Segmenting-Anything/" title="Matching_Anything_by_Segmenting_Anything">Matching_Anything_by_Segmenting_Anything</a></span></li></ul></div><div><h2>最新评论</h2><ul class="leancloud-recent-comment"></ul></div></div><div class="status"><div class="copyright">&copy; 2010 – <span itemprop="copyrightYear">2025</span> <span class="with-love"><i class="ic i-sakura rotate"></i> </span><span class="author" itemprop="copyrightHolder">Lemon Sour @ Amentiraz</span></div><div class="powered-by">基于 <span class="exturl" data-url="aHR0cHM6Ly9oZXhvLmlv">Hexo</span> & Theme.<span class="exturl" data-url="aHR0cHM6Ly9naXRodWIuY29tL2FtZWhpbWUvaGV4by10aGVtZS1zaG9rYQ==">Shoka</span></div></div></div></footer></div><script data-config type="text/javascript">var LOCAL={path:"article/Benchmarking-spatial-claustering-methods-with-spatially-resolved-transcriptomic-data/",favicon:{show:"⁽⁽ଘ( ˊᵕˋ )ଓ⁾⁾",hide:"(つд⊂)"},search:{placeholder:"文章搜索",empty:"关于 「 ${query} 」，什么也没搜到",stats:"${time} ms 内找到 ${hits} 条结果"},valine:!0,copy_tex:!0,katex:!0,fancybox:!0,copyright:'复制成功，转载请遵守 <i class="ic i-creative-commons"></i>BY-NC-SA 协议。',ignores:[function(t){return t.includes("#")},function(t){return new RegExp(LOCAL.path+"$").test(t)}]}</script><script src="https://cdn.polyfill.io/v2/polyfill.js"></script><script src="//cdn.jsdelivr.net/combine/npm/pace-js@1.0.2/pace.min.js,npm/pjax@0.2.8/pjax.min.js,npm/whatwg-fetch@3.4.0/dist/fetch.umd.min.js,npm/animejs@3.2.0/lib/anime.min.js,npm/algoliasearch@4/dist/algoliasearch-lite.umd.js,npm/instantsearch.js@4/dist/instantsearch.production.min.js,npm/lozad@1/dist/lozad.min.js,npm/quicklink@2/dist/quicklink.umd.js"></script><script src="/js/app.js?v=0.2.5"></script></body></html><!-- rebuild by hrmmi -->