<!-- build time:Fri Oct 10 2025 09:42:59 GMT+0800 (中国标准时间) --><!DOCTYPE html><html lang="zh-CN"><head><meta charset="UTF-8"><meta name="viewport" content="width=device-width,initial-scale=1,maximum-scale=2"><meta name="theme-color" content="#FFF"><link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon.png"><link rel="icon" type="image/ico" sizes="32x32" href="/images/favicon.ico"><meta http-equiv="Cache-Control" content="no-transform"><meta http-equiv="Cache-Control" content="no-siteapp"><link rel="alternate" type="application/rss+xml" href="http://amentiraz.github.io/rss.xml"><link rel="alternate" type="application/atom+xml" href="http://amentiraz.github.io/atom.xml"><link rel="alternate" type="application/json" href="http://amentiraz.github.io/feed.json"><link rel="stylesheet" href="//fonts.googleapis.com/css?family=Mulish:300,300italic,400,400italic,700,700italic%7CFredericka%20the%20Great:300,300italic,400,400italic,700,700italic%7CNoto%20Serif%20JP:300,300italic,400,400italic,700,700italic%7CNoto%20Serif%20SC:300,300italic,400,400italic,700,700italic%7CInconsolata:300,300italic,400,400italic,700,700italic&display=swap&subset=latin,latin-ext"><link rel="stylesheet" href="/css/app.css?v=0.2.5"><meta name="keywords" content="生物,学习笔记类,算法"><link rel="canonical" href="http://amentiraz.github.io/article/domain%E5%86%85%E5%AE%B9%E7%9A%84%E6%80%BB%E7%BB%93/"><title>对spatial domain内容的总结 - 论文 | Amentiraz =</title><meta name="generator" content="Hexo 7.3.0"></head><body itemscope itemtype="http://schema.org/WebPage"><div id="loading"><div class="cat"><div class="body"></div><div class="head"><div class="face"></div></div><div class="foot"><div class="tummy-end"></div><div class="bottom"></div><div class="legs left"></div><div class="legs right"></div></div><div class="paw"><div class="hands left"></div><div class="hands right"></div></div></div></div><div id="container"><header id="header" itemscope itemtype="http://schema.org/WPHeader"><div class="inner"><div id="brand"><div class="pjax"><h1 itemprop="name headline">对spatial domain内容的总结</h1><div class="meta"><span class="item" title="创建时间：2025-03-15 21:38:28"><span class="icon"><i class="ic i-calendar"></i> </span><span class="text">发表于</span> <time itemprop="dateCreated datePublished" datetime="2025-03-15T21:38:28+08:00">2025-03-15</time> </span><span class="item" title="本文字数"><span class="icon"><i class="ic i-pen"></i> </span><span class="text">本文字数</span> <span>17k</span> <span class="text">字</span> </span><span class="item" title="阅读时长"><span class="icon"><i class="ic i-clock"></i> </span><span class="text">阅读时长</span> <span>16 分钟</span></span></div></div></div><nav id="nav"><div class="inner"><div class="toggle"><div class="lines" aria-label="切换导航栏"><span class="line"></span> <span class="line"></span> <span class="line"></span></div></div><ul class="menu"><li class="item title"><a href="/" rel="start">Amentiraz</a></li></ul><ul class="right"><li class="item theme"><i class="ic i-sun"></i></li><li class="item search"><i class="ic i-search"></i></li></ul></div></nav></div><div id="imgs" class="pjax"><ul><li class="item" data-background-image="https://amentirazblogpic.oss-cn-hangzhou.aliyuncs.com/blogpic/%E5%BE%AE%E4%BF%A1%E5%9B%BE%E7%89%87_20240925162718.png"></li><li class="item" data-background-image="https://amentirazblogpic.oss-cn-hangzhou.aliyuncs.com/blogpic/%E5%BE%AE%E4%BF%A1%E5%9B%BE%E7%89%87_20240925124526.jpg"></li><li class="item" data-background-image="https://amentirazblogpic.oss-cn-hangzhou.aliyuncs.com/blogpic/wallhaven-r2g7rm.jpg"></li><li class="item" data-background-image="https://amentirazblogpic.oss-cn-hangzhou.aliyuncs.com/blogpic/%E5%BE%AE%E4%BF%A1%E5%9B%BE%E7%89%87_20240925124519.png"></li><li class="item" data-background-image="https://amentirazblogpic.oss-cn-hangzhou.aliyuncs.com/blogpic/%E5%BE%AE%E4%BF%A1%E5%9B%BE%E7%89%87_20240925124347.png"></li><li class="item" data-background-image="https://amentirazblogpic.oss-cn-hangzhou.aliyuncs.com/blogpic/%E5%BE%AE%E4%BF%A1%E5%9B%BE%E7%89%87_20240925124507.png"></li></ul></div></header><div id="waves"><svg class="waves" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" viewBox="0 24 150 28" preserveAspectRatio="none" shape-rendering="auto"><defs><path id="gentle-wave" d="M-160 44c30 0 58-18 88-18s 58 18 88 18 58-18 88-18 58 18 88 18 v44h-352z"/></defs><g class="parallax"><use xlink:href="#gentle-wave" x="48" y="0"/><use xlink:href="#gentle-wave" x="48" y="3"/><use xlink:href="#gentle-wave" x="48" y="5"/><use xlink:href="#gentle-wave" x="48" y="7"/></g></svg></div><main><div class="inner"><div id="main" class="pjax"><div class="article wrap"><div class="breadcrumb" itemscope itemtype="https://schema.org/BreadcrumbList"><i class="ic i-home"></i> <span><a href="/">首页</a></span><i class="ic i-angle-right"></i> <span class="current" itemprop="itemListElement" itemscope itemtype="https://schema.org/ListItem"><a href="/categories/article/" itemprop="item" rel="index" title="分类于 论文"><span itemprop="name">论文</span></a><meta itemprop="position" content="1"></span></div><article itemscope itemtype="http://schema.org/Article" class="post block" lang="zh-CN"><link itemprop="mainEntityOfPage" href="http://amentiraz.github.io/article/domain%E5%86%85%E5%AE%B9%E7%9A%84%E6%80%BB%E7%BB%93/"><span hidden itemprop="author" itemscope itemtype="http://schema.org/Person"><meta itemprop="image" content="/images/avatar.jpg"><meta itemprop="name" content="Lemon Sour"><meta itemprop="description" content=", "></span><span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization"><meta itemprop="name" content=""></span><div class="body md" itemprop="articleBody"><p>总结一下以往看过的文章对于 background，尤其是对于 spatial domain 相关知识的介绍</p><h1 id="benchmarking-spatial-clustering-methods-with-spatially-resolved-transcriptomics-data"><a class="anchor" href="#benchmarking-spatial-clustering-methods-with-spatially-resolved-transcriptomics-data">#</a> Benchmarking spatial clustering methods with spatially resolved transcriptomics data</h1><p>Spatial clustering, which shares an analogy with single-cell clustering, has expanded the scope of tissue physiology from cell-centriod to structure-centroid with spatially resolved transcriptomics(SRT) data.</p><p>advancements in spatially resolved transcriptomics(SRT) enable the multiplexed spatial mapping of gene expression, allowing researchers to move beyond cell clustering to identify higher-order tissue structures, or spatial domains, through the provisdion of additional spatial information.</p><p>Identifying spatial domains by spatial clustering has become a standard initial step in constructing spatial atlas and has proven to be a crucial in visualizing tissue anatomy, inferring tissue spatial signatures of development and disease, and identifying domain-dependent molecular regulatory networks.</p><h2 id="supplymentary-note"><a class="anchor" href="#supplymentary-note">#</a> Supplymentary Note</h2><p>Spatial clustering, also known as spatial domain identification, aims to utilize both gene expression profiles and spatial information to identify tissue regions with spatially coherent gene expression patterns.</p><p>The primary applied spatial data types are spot-like non-single-cell resolution technologies, such as 10X Visium and ST. With the development of advanced technologies that have improved spatial resolution, the applied data type has been extended to single-cell level datasets, such as Slide-seq, Slide-seqV2, Stereo-seq, STARmap, MERFISH, and seqFISH1, among others.</p><p>These methods are useful in the following aspects:</p><ul><li>The identification of tissue structures. Traditionally, histological experts annotate tissue structures, which is a highly labor-demanding and subjective approach. However, spatial clustering provides an unbiased and efficient way to segregate tissue structures without human supervision, saving a lot of time and effort. For example, the classical dorsolateral prefrontal cortex (DLPFC) dataset was originally annotated by experts, but current advances in spatial clustering methods have been able to automatically annotate the tissue with very similar performance to human labels. Other brain regions were also annotated computationally, such as the hypothalamic preoptic region, primary visual area, hippocampus, and cerebellum. In the future, other brain regions, the whole brain, and more complex tissues and organs will hopefully be annotated in an unbiased and effective manner by advancements in computational methods.</li><li>Some tissue structures cannot be trivially identified by referring to H&amp;E images. An important example is the liver. In a liver lobule, the tissue spatially is classically divided into different zones (termed liver zonation), and each zone contains major portions of hepatocytes having different metabolic functions while displaying similar morphological features. In such tissue, manual annotation is especially challenging, given the large amount of liver lobules with complex shapes. Computational methods are very important in such cases.</li><li>The identification of tissue spatial continuity. Some tissues display continuous spatial patterns instead of having sharp tissue interfaces. For example, the brain cortex, although partitioned into several discrete layers, may not have strictly defined boundaries. Many genes exhibited strong continuity along the cortical axis, and some higher-level cellular organizations also showed continuity, such as cell type complexity and spatial heterogeneity, which were reported to have strong correlations with cortical depth. The continuity sometimes has clinical significance in disease and cancer. Investigations have been made in studying the gradual structural changes from adjacent normal tissue to cancerous tissue to understand the molecular and tissue biology of cancer.</li><li>The identification of domain-specific marker genes. Different from cell-type-specific marker genes, domain-specific marker genes show stronger spatial autocorrelations.Their functions are more likely to be spatially related or display specificity to certain proximities to certain microenvironments. Traditionally, gene markers that define specific brain regions are identified by traditional assays (such as bulk or scRNA-seq) from selected brain domains. Such practice is labor-demanding and has limited domain specificity. Some genes within certain domains or tissue microenvironments are strongly related to their localizations. For example, in disease, molecular polarizations have been found in liver fibrosis, where metabolites related to glycolysis within the hepatic domains showed polarization patterns towards the fibrosis domain boundary. In another example in cancer, some immune-related proteins showed polarization patterns within the immune regions towards tumor domain boundaries, which is reported to be associated with tumor-immune interactions.</li><li>Mining spatial signatures of development and disease. In the single-cell era, cell type proportions were typically used to differentiate different patient groups for better diagnosis. However, many diseases and cancers do not show significant differences in cell type proportions but only have differences in terms of the spatial organizations of cells30,31. Such cases have been reported in different cancers (e.g., breast cancer12 and colorectal cancer30) and COVID-19 diseases31. Pioneer computational methods have already illustrated the diagnostic significance of such cases.</li><li>Identifying domain-dependent molecular regulatory networks. Such studies are mainly concentrated on investigating the distribution of disease-related molecules (e.g., proteins) across spatial domains. A recent study developed STARmapPlus to simultaneously spatially profile targeted proteins and gene expressions. They used the new technology to study the different distributions of Alzheimer's disease-related proteins within domains. Another study utilized computational methods to predict the abundance of such proteins by incorporating spatial microenvironment information.</li></ul><h1 id="spagcn"><a class="anchor" href="#spagcn">#</a> SpaGCN</h1><p>Recent technological advances in SRT have enabled gene expression profilingh with spatial information in tissue.</p><p>Knoledge of the relative locations of different cells in a tissue is critical for understanding disease pathology because spatial information helps in understanding how the gene expression of a cell is influenced by its surrounding environment.</p><p>methods for SRT:</p><ul><li>in situ hybridization or sequencing-based technologies with single-cell resolution<ul><li>e.g. seqFISH, seqFISH+, MERFISH, STARmap and FISSEQ that measure the expression level for hundred to thousands of genes in cells within their tissue context.</li><li>因为能达到单细胞分辨率，这些技术适合于高精度分析，用于深入理解细胞在组织结构中的分布以及细胞间的相互作用。</li></ul></li><li>in situ capturing-based technologies with spatial barcoding followed by sequencing<ul><li>e.g. spatial transcriptomics(ST), SLIDE-seq, SLIDE-seqV2, HDST and 10x Visium that measure the expression level for thousands of genes in captured locations, referred to as spots.</li><li>由于通常只能定位到捕获区域的水平，这类技术的空间分辨率低于单细胞分辨率，但它们适合于大规模研究，因为能够同时检测数千个基因。</li></ul></li></ul><p>To link spatial domains with biological functions, it is crucial to identify genes that show enriched expression in the identified domains. Methods such as Trendsceek, SpatialDE and SPARK have been developed to detect spatially variable genes(SVGs).</p><h1 id="bayesspace"><a class="anchor" href="#bayesspace">#</a> BayesSpace</h1><p>Knowledge of the spatial location of transript expression can provide vital insights into biological function and pathology.</p><p>Single-cell RNA sequencing(scRNA-seq) achieves high-throughput and high-resolution profiling of gene expression, but because tissue is dissociated for sample preparation, spatial information is not retained.</p><p>Recent methods for high-throughput profiling of gene expression while retaining spatial information allow analyses to be made within the context of the biological tissue.Studies performed with the Spatial Transcriptomics(ST) platform and the improved Visium platform have already generated insights into diverse areas such as tumor heterogeneity, brain function and the pathophysiology of sepsis. The primary technological limitation of these spatial gene expression platforms is resolution, with the unit of observation being spots that are 100 μm in diameter on the ST platform and 55 μm in diameter on the Visium platform. As such, the number of cells within a spot may range from one to 30 on the Visium platform and up to 200 on the older ST platform, depending on the biological tissue. Alternative approaches include fluorescence in situ hybridization (FISH) technologies, such as seqFISH and multiplexed error-robust FISH, and other recently developed spatial sequencing methods, such as Slide-seq and ZipSeq. While these methods provide increased resolution, most are lower throughput, less sensitive, rely on custom protocols or are not widely available.</p><p>+++ 解释<br>这段话主要讨论了不同 ** 空间转录组技术（spatial transcriptomics, ST）<strong>在</strong>分辨率（resolution）、通量（throughput）、灵敏度（sensitivity）和可用性（availability）** 方面的差异。可以从以下几个方面理解：</p><hr><h3 id="1-visium-与-st-平台的细胞数差异"><a class="anchor" href="#1-visium-与-st-平台的细胞数差异">#</a> <strong>1. Visium 与 ST 平台的细胞数差异</strong></h3><ul><li>在 <strong>10X Genomics Visium</strong> 平台上，每个检测点（spot）可能包含 <strong>1 到 30 个细胞</strong>，具体取决于生物组织的密度。</li><li>在 <strong>较早的 ST（Spatial Transcriptomics）平台</strong> 上，每个检测点可能包含<strong>多达 200 个细胞</strong>。</li><li>这表明 <strong>Visium 提供了更高的分辨率</strong>，因为每个 spot 覆盖的细胞数量更少，可以更精细地解析基因表达的空间分布。</li></ul><hr><h3 id="2-其他高分辨率技术fish-和新型空间测序方法"><a class="anchor" href="#2-其他高分辨率技术fish-和新型空间测序方法">#</a> <strong>2. 其他高分辨率技术：FISH 和新型空间测序方法</strong></h3><ul><li><p><strong>荧光原位杂交（FISH）技术</strong>：</p><ul><li><strong>seqFISH（sequential FISH）</strong></li><li><strong>MERFISH（multiplexed error-robust FISH）</strong><br>这些技术通过荧光探针直接检测 RNA 分子，可以达到单细胞甚至亚细胞（subcellular）级别的分辨率。</li></ul></li><li><p><strong>其他新型空间测序方法</strong>：</p><ul><li><strong>Slide-seq</strong>：将 RNA 分子定位到微珠阵列（bead array）上，实现高分辨率空间测序。</li><li><strong>ZipSeq</strong>：结合光学方法和条形码标记（barcoding）来进行空间解析。</li></ul></li></ul><hr><h3 id="3-这些方法的局限性"><a class="anchor" href="#3-这些方法的局限性">#</a> <strong>3. 这些方法的局限性</strong></h3><p>尽管 <strong>seqFISH、MERFISH、Slide-seq 和 ZipSeq</strong> 在分辨率方面优于 Visium 和 ST 平台，但它们存在以下不足：</p><ol><li><strong>通量较低（lower throughput）</strong>：一次实验可以检测的 RNA 分子或细胞数量有限，难以大规模应用。</li><li><strong>灵敏度较低（less sensitive）</strong>：相比于 Visium 这样的商业化平台，FISH 技术可能在检测低丰度基因表达时存在问题。</li><li><strong>依赖定制实验流程（custom protocols）</strong>：这些方法往往需要特殊的实验设置，不如 Visium 这样的平台易于使用。</li><li><strong>尚未广泛可用（not widely available）</strong>：这些技术可能仍处于实验室研发阶段，尚未大规模商业化，难以在各类实验室推广。</li></ol><hr><h3 id="总结"><a class="anchor" href="#总结">#</a> <strong>总结</strong></h3><table><thead><tr><th>技术</th><th>细胞覆盖范围</th><th>分辨率</th><th>通量</th><th>灵敏度</th><th>可用性</th></tr></thead><tbody><tr><td><strong>Visium</strong></td><td>1-30 细胞 /spot</td><td>中等</td><td>高</td><td>高</td><td>商业化，易获取</td></tr><tr><td><strong>ST（老版）</strong></td><td>高达 200 细胞 /spot</td><td>低</td><td>高</td><td>高</td><td>商业化，易获取</td></tr><tr><td><strong>seqFISH / MERFISH</strong></td><td>单细胞甚至亚细胞级</td><td>高</td><td>低</td><td>低</td><td>需要特殊设备和协议</td></tr><tr><td><strong>Slide-seq / ZipSeq</strong></td><td>取决于条形码技术</td><td>高</td><td>低</td><td>低</td><td>仍在研究中，尚未大规模应用</td></tr></tbody></table><p><strong>核心理解</strong>：Visium 和 ST 平台虽然分辨率不如 seqFISH 或 Slide-seq，但它们的<strong>通量高、灵敏度强、实验流程标准化，适用于大规模应用</strong>。而 FISH 和新型空间测序方法提供了更高分辨率，但<strong>实验复杂度高，难以大规模推广</strong>。<br>+++</p><h1 id="stagate"><a class="anchor" href="#stagate">#</a> STAGATE</h1><p>Deciphering spatial domains (i.e., regions with similar spatial expression patterns) is one of the great challenges from STs.For example, the laminar organization of the human cerebral cortex is especially related to its biological functions, in which cells residing within different cortical layers often differ in expressions, morphology and physiology.</p><p>These non-spatial methods can be roughly divided into two categories. The first category uses traditional clustering methods such as k-means and Louvain algorithm. These methods are limited to the small number of spots or the sparsity according to the different resolutions of ST technologies, and clustering results may be discontinuous in the tissue section. The second category utilizes the cell type signatures defined by single-cell RNA-seq to deconvolute the spots. They are not applicable to ST data at a resolution of cellular or subcellular levels.</p><p>Some recent algorithms adapt the clustering methods by considering the similarity between adjacent spots to better account for the spatial dependency of gene expressions. These methods show significant improvements in identifying spatial domains of sections from brain and cancer tissues. For example, BayesSpace is a Bayesian statistical method that encourages neighboring spots to belong to the same cluster by introducing spatial neighbor structure into the prior. Giotto identifies spatial domains by implementing a hidden Markov random field (HMRF) model with the spatial neighbor prior. stLearn defines the morphological distance based on features extracted from a histology image and utilizes such distances as well as spatial neighbor structure to smooth gene expressions. SEDR employs a deep auto-encoder network for learning gene representations and uses a variational graph auto-encoder to simultaneously embed spatial information. SpaGCN also applies the graph convolutional network to integrate gene expression and spatial location, and further coupled with a self-supervised module to identify domains. Besides, a recent developed method named RESEPT leverages the supervised image segmentation method to perform tissue structure identification. Although these methods consider the spatial structure of STs, the similarity of neighboring spots defined by them is pre-defined before training and cannot be learned adaptively. Moreover, these methods do not consider the spatial similarity of spots at the boundary of spatial domains in more detail and do not well integrate spatial information to impute and denoise gene expressions. More importantly, these approaches cannot be applied to multiple consecutive sections to reconstruct a three-dimensional (3D) ST model and extract 3D expression domains.</p><p>全是英文看的想吐，这里补充一下中文的版本：</p><h1 id="benchmarking"><a class="anchor" href="#benchmarking">#</a> Benchmarking</h1><p>空间转录组学（SRT）的进步实现了基因表达的多重空间定位，使研究人员能够突破传统的细胞聚类，借助额外的空间信息识别更高层级的组织结构，即空间域（spatial domains）[<sup>1,</sup>2,^3]。通过空间聚类识别空间域已成为构建空间图谱（spatial atlas）的标准初始步骤 [<sup>4–</sup>8]，并在可视化组织解剖结构 [<sup>9]、推断组织空间连续性 [</sup>10,<sup>11]、检测空间域特异性标志基因 [</sup>12,<sup>13]、挖掘发育与疾病的空间特征 [</sup>14,<sup>15]、识别空间域依赖的分子调控网络 [</sup>16,^17] 等方面发挥了关键作用（见补充说明 1）。</p><p>尽管近年来已经提出了基于概率图模型和图神经网络（GNNs）等多种识别空间域的计算方法 [<sup>18,</sup>19]，但由于所用数据集和评估指标缺乏一致性与全面性，仍面临较大挑战。这些问题的产生原因包括：空间技术快速迭代、部分应用中评估指标过于有限，以及数据往往来自特定实验室基于特定组织和技术生成的数据集（见补充说明 2）。</p><p>尽管已有一些空间转录组数据的基准评估工作，尤其聚焦于细胞类型识别 [^20]，但针对空间聚类方法用于空间域识别的系统性基准研究仍然缺失（见扩展图 1）。在本项研究中，我们从四个维度 —— 预测准确性、空间域连续性、空间标志基因识别能力以及方法的可扩展性（详见方法部分）—— 设定了十项评估指标。需要说明的是，本文中 “准确性” 一词的含义较传统统计学术语更广，而 “数据”（data）指的是来自同一切片的空间转录组数据，“数据集”（dataset）则表示来源于同一篇文献、基于相同技术生成的一组数据。</p><h1 id="spagcn-2"><a class="anchor" href="#spagcn-2">#</a> SpaGCN</h1><p>近年来空间转录组技术（SRT）的快速发展，使得我们能够在组织中获取带有空间信息的基因表达谱 [^1]。在理解疾病病理过程中，不同细胞在组织中的相对位置尤为关键，因为空间信息有助于揭示细胞基因表达如何受到其周围微环境的影响。</p><p>目前常用的空间转录组实验方法大致可分为两类：<br>第一类是基于原位杂交或测序的技术，具有单细胞分辨率，代表性方法包括 seqFISH [<sup>2,</sup>3]、seqFISH+[<sup>4]、MERFISH[</sup>5,<sup>6]、STARmap[</sup>7] 和 FISSEQ [^8]，它们可在保持组织结构的同时测量数百到上千个基因的表达水平。<br>第二类是基于原位捕获的技术，通过空间条形码标记并后续进行测序，代表性方法包括 Spatial Transcriptomics（ST）[<sup>9]、SLIDE-seq[</sup>10]、SLIDE-seqV2[<sup>11]、HDST[</sup>12] 以及 10x Visium。这类技术可以在被称为 “spot” 的捕获位置上检测数千个基因的表达。</p><p>这些多样化的 SRT 技术使得揭示异质性组织中复杂的转录组结构成为可能，也极大地拓展了我们对疾病相关细胞机制的理解 [<sup>13,</sup>14]。</p><p>在空间转录组研究中，识别 “空间域” 是一个重要步骤 —— 空间域是指在基因表达和组织形态上都具有空间连续性的区域。传统聚类方法如 K-means 和 Louvain 算法 [^15] 仅依赖基因表达数据，忽略了空间和组织结构信息，往往会得到不连续或生物学解释性较差的聚类结果。</p><p>为了解决这一问题，近年来发展了多种考虑空间依赖性的聚类方法。例如，Zhu 等人 [<sup>16] 提出了基于隐马尔可夫随机场（HMRF）的模型来捕捉基因表达的空间依赖关系；stLearn [</sup>17] 在聚类前融合了组织学图像特征以及邻近 spot 的表达信息以归一化数据；BayesSpace [^18] 则通过贝叶斯方法引入物理邻近性作为先验来优化聚类效果。这些方法虽能有效将 spot 或细胞划分为不同组，但由于在处理不同数据类型（模态）上的灵活性不足，限制了其广泛适用性。随着新一代空间转录组技术不断涌现 [<sup>19–</sup>22]，亟需开发更具通用性的方法来适配多种平台。</p><p>为了进一步探究空间域的生物学功能，识别在特定空间区域中高表达的基因是至关重要的一步。为此，研究者已开发出多种方法来检测具有空间差异表达的基因（spatially variable genes, SVGs），如 Trendsceek [<sup>23]、SpatialDE[</sup>24] 和 SPARK [^25]。这些方法通常逐个基因独立评估其空间变异性，并通过 P 值反映其空间表达的显著性。然而，由于这些方法未考虑空间域的整体结构，所筛选出的基因可能缺乏明确的空间表达模式，从而限制了其在后续生物学研究中的应用潜力。</p><h1 id="stagate-2"><a class="anchor" href="#stagate-2">#</a> STAGATE</h1><p>复杂组织的功能在根本上与不同细胞类型的空间环境密切相关 [<sup>1]。组织中转录表达的相对位置对于理解其生物学功能、描述细胞间的交互网络具有重要意义 [</sup>2]。近年来，空间转录组（Spatial Transcriptomics, ST）技术取得了突破性进展，如 10x Visium [<sup>3]、Slide-seq[</sup>4,<sup>5]、Stereo-seq[</sup>6] 和 PIXEL-seq [^7] 等，使得我们能够在被称为 “spot” 的捕获位置上，以数个细胞甚至亚细胞级别的分辨率进行全基因组水平的表达谱分析。</p><p>解析空间域（即表达模式相似的区域）是空间转录组研究面临的重要挑战之一。例如，人类大脑皮层的层状结构与其生物学功能密切相关，其中不同皮层层级中的细胞在基因表达、形态结构和生理功能上往往存在显著差异 [^8]。然而，多数现有聚类方法未能有效利用空间信息。这些非空间方法大致可分为两类：<br>第一类是传统聚类算法，如 K-means 和 Louvain 算法 [^9]，它们在 spot 数量较少或数据稀疏时（因 ST 技术分辨率不同）效果有限，且常导致聚类结果在组织切片中不连续；<br>第二类方法基于单细胞 RNA 测序定义的细胞类型特征对 spot 进行解卷积 [<sup>10,</sup>11]，但该类方法并不适用于细胞或亚细胞分辨率的 ST 数据。</p><p>为更好地建模基因表达的空间依赖性，部分近期算法在聚类时引入了邻近 spot 之间的相似性考虑，从而在脑组织和癌症组织的空间域识别方面取得了显著进展 [<sup>12,</sup>13,^14]。<br>例如，BayesSpace 是一种贝叶斯统计方法，通过在先验中引入空间邻近结构，鼓励相邻 spot 属于同一聚类 [^12]；Giotto 采用隐马尔可夫随机场（HMRF）模型，结合空间邻接信息以识别空间域 [^15]；stLearn 利用组织切片图像中提取的形态特征定义 “形态距离”，并结合空间邻接结构对基因表达进行平滑处理 [^13]；SEDR 则通过深度自动编码器学习基因表达表示，并利用变分图自编码器嵌入空间信息 [^14]；SpaGCN 应用了图卷积网络来整合基因表达与空间位置，并通过自监督模块提升空间域识别效果 [^16]；此外，新近提出的 RESEPT 方法则借助有监督图像分割方法进行组织结构识别 [^17]。</p><p>虽然这些方法均考虑了 ST 的空间结构，但它们所定义的邻近 spot 间相似性在训练前是固定的，无法自适应学习；同时，这些方法在处理空间域边界处 spot 之间的空间相似性上考虑不足，亦未能充分整合空间信息对基因表达进行插补与去噪。更重要的是，这些方法无法应用于多张连续切片，进而无法构建三维（3D）ST 模型，也无法识别三维空间表达域（详见补充表 S1）。</p><h1 id="bayesspace-2"><a class="anchor" href="#bayesspace-2">#</a> BayesSpace</h1><p>了解转录表达的空间位置可以为生物功能和病理过程提供重要的洞见。单细胞 RNA 测序（scRNA-seq）虽然可以实现高通量和高分辨率的基因表达谱分析，但由于组织在样本制备过程中被解离，空间信息无法保留。近年来，一些新方法能够在保留空间信息的同时进行高通量的基因表达分析，使得研究可以在生物组织的上下文中进行 [^1]。</p><p>基于 Spatial Transcriptomics（ST）平台及其改进版 Visium 平台的研究，已经在多个领域产生了新的见解，如肿瘤异质性 [<sup>2,</sup>3]、脑功能 [<sup>4] 以及败血症的病理生理机制 [</sup>5]。这些空间基因表达平台的主要技术限制在于分辨率：在 ST 平台上，观察单位是直径为 100 μm 的 spot，而在 Visium 平台上为 55 μm。因此，在 Visium 平台的一个 spot 中，可能包含 1 至 30 个细胞，而在老一代 ST 平台中，这一数量可达 200 个，具体取决于所研究的组织类型 [^6]。</p><p>其他替代方法包括荧光原位杂交（FISH）技术，如 seqFISH 和多重误差稳健 FISH（MERFISH），以及最近发展的一些空间测序方法，如 Slide-seq 和 ZipSeq [<sup>7,</sup>8,<sup>9,</sup>10]。尽管这些方法在空间分辨率上有所提高，但它们多数通量较低、灵敏度较差、依赖定制实验方案，或尚未广泛普及。</p><h1 id="mender"><a class="anchor" href="#mender">#</a> MENDER</h1><p>空间分辨单细胞（SRSC, Spatially Resolved Single-Cell）技术的最新进展，使得在组织环境中对细胞基因表达进行高精度分析成为可能，从而可以全面地刻画各种生物系统的空间特征 [<sup>1,</sup>2,<sup>3,</sup>4,<sup>5,</sup>6,<sup>7]。在不同基因表达状态的细胞协同作用下，空间域作为更高阶的功能单元，在组织空间中呈现出规律性的分布，并与组织生理功能密切相关 [</sup>8,<sup>9]。在癌症等复杂疾病中，越来越多的证据表明，特定空间域在疾病的诊断与监测中发挥着关键作用 [</sup>10,<sup>11,</sup>12]。</p><p>随着 SRSC 数据的不断积累 [<sup>13,</sup>14]，众多计算方法被提出以识别这些空间域 [<sup>15,</sup>16,^17]。</p><p>在典型的 SRSC 数据集中，每个细胞都被测量了其空间坐标和基因表达谱。这种数据结构天然地构成了一个空间图，其中细胞作为图的节点，基因表达作为节点属性。这一结构促生了该领域的两大主流建模范式：图神经网络（GNN, Graph Neural Network）[<sup>18,</sup>19,^20] 和 贝叶斯网络（BN, Bayesian Network）[<sup>21,</sup>22,^23]。</p><p>这两种建模路线的发展，大多数方法都是通过增加模型复杂度来提升性能。GNN 方法通常引入了专门的神经模块、损失函数和网络结构；而 BN 方法则扩展了隐变量、变量之间的依赖关系及先验知识。虽然模型复杂度的提升往往带来性能的增强，但近年来的一些研究显示，这种提升正呈现出边际收益递减的趋势 [^24]。</p><p>此外，模型的复杂性还可能带来如下问题：参数调整困难、时间效率低、泛化能力下降。因此，这些挑战共同呼唤一种新的建模范式，以突破当前领域的发展瓶颈。</p><p>我们首先解释为何在已有众多方法的前提下，仍然需要一种新的空间域识别方法。我们选取了近两年发表的 8 种现有方法，并从 6 个评估维度进行比较，包括：是否支持多切片分析、稳定性、可解释性、可扩展性、运行速度，以及是否能输出细胞上下文表示（见补充图 1A）。每个评估维度的简要定义见补充图 1A，详细说明见 “方法” 部分中 “从六个方面评估现有方法” 章节。</p><p>这 8 种方法包括 4 种基于图神经网络（GNN）的模型（SpaGCN [<sup>18]、STAGATE[</sup>19]、CCST [^25] 和 SpaceFlow [<sup>20]），以及 4 种基于贝叶斯网络（BN）的模型（BayesSpace [</sup>21]、BASS[<sup>23]、SpatialPCA[</sup>26] 和 SOTIP [^27]）。</p><p>从结果中可以看到，多数评估指标与方法的基本原理密切相关（见补充图 1A）。所有 GNN 类方法在可扩展性和运行速度（在使用 GPU 的前提下）方面表现更好，并且能够输出细胞的上下文表示。但这类方法也继承了深度学习模型普遍存在的缺陷：稳定性差、可解释性不足。相较之下，BN 类方法在输出结果的稳定性和可解释性上更具优势，因为它们通常构建在明确的概率变量依赖关系上。然而，这类方法往往无法很好地适配大规模数据集，运行时间较长，且通常不能输出细胞上下文表示（SpatialPCA [^26] 是个例外）。</p><p>这些评估结果不仅被我们在本文中的基准测试所验证，也得到了近期相关研究 [<sup>20,</sup>23] 的支持。</p><p>特别地，在空间组学迈入 “大数据时代” 的背景下，上述某些指标变得尤为关键。目前，许多大型科研合作项目已生成了包含上百万个细胞、跨多个切片的大规模空间组学数据集 [<sup>28,</sup>29,^30]。在这种情境下，方法的可扩展性、运行效率和对多切片分析的支持变得格外重要。</p><p>因此，尽管现有空间域识别方法种类繁多，仍然需要新的创新方法，以尽可能满足上述关键评估标准。</p><p>组织内多种细胞类型复杂的空间组织结构，对于理解其生物学功能和病理状态至关重要。近年来，空间转录组学（Spatially Resolved Transcriptomics, SRT）技术取得了显著进展，如 10x Visium 等平台，通过原位测序大幅提升了我们在转录组水平上绘制组织结构和细胞间相互作用图谱的能力 [^1–4]。在分析空间转录组数据时，一个关键任务是将测点（spot）准确划分到空间域（spatial domain）中，这些空间域通常在基因表达模式和组织学结构上具有一致性。对复杂组织区域的准确划分对后续分析具有重要意义，例如识别空间变异基因（Spatially Variable Genes, SVGs）[<sup>5]、解析组织异质性 [</sup>6]，以及 ** 探索细胞 - 细胞间相互作用 [^7]** 等。</p><h1 id="ensdd"><a class="anchor" href="#ensdd">#</a> EnSDD</h1><p>当前的空间域检测（Spatial Domain Detection, SDD）方法大致可分为非空间聚类方法与空间聚类方法。传统的非空间聚类方法，如 K-means 和 Louvain 算法 [^8]，仅利用基因表达数据，常常导致聚类结果在空间上不连贯。相比之下，空间聚类方法（如 BayesSpace [<sup>9]、DR-SC[</sup>10]、GraphST [^11] 和 STAGATE [^12]）在分析中引入了 SRT 数据的空间坐标，从而考虑基因表达的空间依赖性。这些方法通常会根据基因表达和空间位置对相邻 spot 进行联合聚类，通过不同的空间邻域建模策略来捕捉空间关系。虽然这些方法在脑组织和癌症组织中表现出显著优势 [<sup>12]，但它们往往忽略了组织学图像中蕴含的形态学信息，这可能导致划分出的空间域与真实组织结构不一致 [</sup>13]。</p><p>为了解决这一问题，出现了一类融合多模态信息的方法，如 SpaGCN [<sup>13]、stLearn[</sup>14]、SiGra [^15] 和 spaVAE [^16]，它们将基因表达、空间坐标和组织学图像等多种信息结合，通过学习相邻 spot 间的空间依赖关系，从而在表达和形态学两方面都实现连续一致的空间域检测。</p><p>由于各类空间聚类方法在假设和数据依赖方面存在差异，它们在划分空间域时产生的结果也存在差异 [<sup>17]。具体方法的效果还会受到组织复杂程度的影响 [</sup>18]，这也增加了在新数据集中选择合适方法的难度。一个应对策略是开发集成方法（ensemble methods），通过整合多个基础方法的结果提高稳健性 [^19]。然而，集成方法也面临两个主要挑战：（i）不同基础方法可能生成不同数量和类型的空间域，使得结果整合变得复杂；（ii）方法性能在不同数据集间表现不一致，难以自动评估并融合各方法的效果。</p><div class="tags"><a href="/tags/%E7%94%9F%E7%89%A9/" rel="tag"><i class="ic i-tag"></i> 生物</a> <a href="/tags/%E5%AD%A6%E4%B9%A0%E7%AC%94%E8%AE%B0%E7%B1%BB/" rel="tag"><i class="ic i-tag"></i> 学习笔记类</a> <a href="/tags/%E7%AE%97%E6%B3%95/" rel="tag"><i class="ic i-tag"></i> 算法</a></div></div><footer><div class="meta"><span class="item"><span class="icon"><i class="ic i-calendar-check"></i> </span><span class="text">更新于</span> <time title="修改时间：2025-04-30 10:51:14" itemprop="dateModified" datetime="2025-04-30T10:51:14+08:00">2025-04-30</time> </span><span id="article/domain内容的总结/" class="item leancloud_visitors" data-flag-title="对spatial domain内容的总结" title="阅读次数"><span class="icon"><i class="ic i-eye"></i> </span><span class="text">阅读次数</span> <span class="leancloud-visitors-count"></span> <span class="text">次</span></span></div><div id="copyright"><ul><li class="author"><strong>本文作者： </strong>Lemon Sour <i class="ic i-at"><em>@</em></i></li><li class="link"><strong>本文链接：</strong> <a href="http://amentiraz.github.io/article/domain%E5%86%85%E5%AE%B9%E7%9A%84%E6%80%BB%E7%BB%93/" title="对spatial domain内容的总结">http://amentiraz.github.io/article/domain内容的总结/</a></li><li class="license"><strong>版权声明： </strong>本站所有文章除特别声明外，均采用 <span class="exturl" data-url="aHR0cHM6Ly9jcmVhdGl2ZWNvbW1vbnMub3JnL2xpY2Vuc2VzL2J5LW5jLXNhLzQuMC96aC1DTg=="><i class="ic i-creative-commons"><em>(CC)</em></i>BY-NC-SA</span> 许可协议。转载请注明出处！</li></ul></div></footer></article></div><div class="post-nav"><div class="item left"><a href="/music/%E5%B8%B8%E8%A7%81%E5%92%8C%E5%BC%A6/" itemprop="url" rel="prev" data-background-image="https:&#x2F;&#x2F;amentirazblogpic.oss-cn-hangzhou.aliyuncs.com&#x2F;blogpic&#x2F;%E5%BE%AE%E4%BF%A1%E5%9B%BE%E7%89%87_20240925124429.jpg" title="常见和弦"><span class="type">上一篇</span> <span class="category"><i class="ic i-flag"></i> 乐理</span><h3>常见和弦</h3></a></div><div class="item right"><a href="/life/%E7%B2%BE%E7%A5%9E%E5%88%86%E6%9E%90%E4%B8%AD%E7%9A%84%E5%80%92%E9%94%99/" itemprop="url" rel="next" data-background-image="https:&#x2F;&#x2F;amentirazblogpic.oss-cn-hangzhou.aliyuncs.com&#x2F;blogpic&#x2F;wallhaven-e7pwdw.jpg" title="精神分析中的倒错"><span class="type">下一篇</span> <span class="category"><i class="ic i-flag"></i> 思考</span><h3>精神分析中的倒错</h3></a></div></div><div class="wrap" id="comments"></div></div><div id="sidebar"><div class="inner"><div class="panels"><div class="inner"><div class="contents panel pjax" data-title="文章目录"><ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#benchmarking-spatial-clustering-methods-with-spatially-resolved-transcriptomics-data"><span class="toc-number">1.</span> <span class="toc-text">Benchmarking spatial clustering methods with spatially resolved transcriptomics data</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#supplymentary-note"><span class="toc-number">1.1.</span> <span class="toc-text">Supplymentary Note</span></a></li></ol></li><li class="toc-item toc-level-1"><a class="toc-link" href="#spagcn"><span class="toc-number">2.</span> <span class="toc-text">SpaGCN</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#bayesspace"><span class="toc-number">3.</span> <span class="toc-text">BayesSpace</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#1-visium-%E4%B8%8E-st-%E5%B9%B3%E5%8F%B0%E7%9A%84%E7%BB%86%E8%83%9E%E6%95%B0%E5%B7%AE%E5%BC%82"><span class="toc-number">3.0.1.</span> <span class="toc-text">1. Visium 与 ST 平台的细胞数差异</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#2-%E5%85%B6%E4%BB%96%E9%AB%98%E5%88%86%E8%BE%A8%E7%8E%87%E6%8A%80%E6%9C%AFfish-%E5%92%8C%E6%96%B0%E5%9E%8B%E7%A9%BA%E9%97%B4%E6%B5%8B%E5%BA%8F%E6%96%B9%E6%B3%95"><span class="toc-number">3.0.2.</span> <span class="toc-text">2. 其他高分辨率技术：FISH 和新型空间测序方法</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#3-%E8%BF%99%E4%BA%9B%E6%96%B9%E6%B3%95%E7%9A%84%E5%B1%80%E9%99%90%E6%80%A7"><span class="toc-number">3.0.3.</span> <span class="toc-text">3. 这些方法的局限性</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#%E6%80%BB%E7%BB%93"><span class="toc-number">3.0.4.</span> <span class="toc-text">总结</span></a></li></ol></li></ol><li class="toc-item toc-level-1"><a class="toc-link" href="#stagate"><span class="toc-number">4.</span> <span class="toc-text">STAGATE</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#benchmarking"><span class="toc-number">5.</span> <span class="toc-text">Benchmarking</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#spagcn-2"><span class="toc-number">6.</span> <span class="toc-text">SpaGCN</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#stagate-2"><span class="toc-number">7.</span> <span class="toc-text">STAGATE</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#bayesspace-2"><span class="toc-number">8.</span> <span class="toc-text">BayesSpace</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#mender"><span class="toc-number">9.</span> <span class="toc-text">MENDER</span></a></li><li class="toc-item toc-level-1"><a class="toc-link" href="#ensdd"><span class="toc-number">10.</span> <span class="toc-text">EnSDD</span></a></li></div><div class="related panel pjax" data-title="系列文章"><ul><li><a href="/article/scRNA%E8%AE%BA%E6%96%87%E7%AC%94%E8%AE%B0/" rel="bookmark" title="scRNA论文笔记">scRNA论文笔记</a></li><li><a href="/article/What-is-A-Cell-Type/" rel="bookmark" title="What_is_A_Cell_Type">What_is_A_Cell_Type</a></li><li><a href="/article/Cell-Review-What-is-a-cell-type-and-how-to-define-it/" rel="bookmark" title="Cell_Review_What_is_a_cell_type_and_how_to_define_it">Cell_Review_What_is_a_cell_type_and_how_to_define_it</a></li><li><a href="/article/Benchmarking-spatial-claustering-methods-with-spatially-resolved-transcriptomic-data/" rel="bookmark" title="Benchmarking-spatial-claustering-methods-with-spatially-resolved-transcriptomic-data">Benchmarking-spatial-claustering-methods-with-spatially-resolved-transcriptomic-data</a></li><li><a href="/article/%E8%AE%BA%E6%96%87ppt1/" rel="bookmark" title="论文ppt1">论文ppt1</a></li><li><a href="/article/Matching-Anything-by-Segmenting-Anything/" rel="bookmark" title="Matching_Anything_by_Segmenting_Anything">Matching_Anything_by_Segmenting_Anything</a></li><li><a href="/article/SpaGCN/" rel="bookmark" title="SpaGCN">SpaGCN</a></li><li><a href="/article/STAGATE/" rel="bookmark" title="STAGATE">STAGATE</a></li><li><a href="/article/MENDER/" rel="bookmark" title="MENDER">MENDER</a></li><li><a href="/article/scanpy%E6%95%B0%E6%8D%AE%E4%BD%BF%E7%94%A8%E8%AE%B0%E5%BD%95/" rel="bookmark" title="scanpy数据使用笔记">scanpy数据使用笔记</a></li><li><a href="/article/BayesSpace/" rel="bookmark" title="BayesSpace">BayesSpace</a></li><li><a href="/article/%E6%95%B0%E6%8D%AE%E9%9B%86%E8%AE%B0%E5%BD%95/" rel="bookmark" title="数据集记录">数据集记录</a></li><li><a href="/article/DeepST/" rel="bookmark" title="DeepST">DeepST</a></li><li><a href="/article/EnSDD/" rel="bookmark" title="EnSDD">EnSDD</a></li><li class="active"><a href="/article/domain%E5%86%85%E5%AE%B9%E7%9A%84%E6%80%BB%E7%BB%93/" rel="bookmark" title="对spatial domain内容的总结">对spatial domain内容的总结</a></li><li><a href="/article/%E6%AF%95%E4%B8%9A%E8%AE%BE%E8%AE%A1%E6%96%87%E6%A1%A3%E6%B1%87%E6%80%BB/" rel="bookmark" title="毕业设计文档汇总">毕业设计文档汇总</a></li><li><a href="/article/scPerturb/" rel="bookmark" title="scPerturb">scPerturb</a></li><li><a href="/GEARS/" rel="bookmark" title="GEARS">GEARS</a></li></ul></div><div class="overview panel" data-title="站点概览"><div class="author" itemprop="author" itemscope itemtype="http://schema.org/Person"><img class="image" itemprop="image" alt="Lemon Sour" data-src="/images/avatar.jpg"><p class="name" itemprop="name">Lemon Sour</p><div class="description" itemprop="description"></div></div><nav class="state"><div class="item posts"><a href="/archives/"><span class="count">85</span> <span class="name">文章</span></a></div><div class="item categories"><a href="/categories/"><span class="count">26</span> <span class="name">分类</span></a></div><div class="item tags"><a href="/tags/"><span class="count">57</span> <span class="name">标签</span></a></div></nav><div class="social"><span class="exturl item github" data-url="aHR0cHM6Ly9naXRodWIuY29tL0FtZW50aXJheg==" title="https:&#x2F;&#x2F;github.com&#x2F;Amentiraz"><i class="ic i-github"></i></span> <span class="exturl item zhihu" data-url="aHR0cHM6Ly93d3cuemhpaHUuY29tL3Blb3BsZS9kc2ZseS04" title="https:&#x2F;&#x2F;www.zhihu.com&#x2F;people&#x2F;dsfly-8"><i class="ic i-zhihu"></i></span> <span class="exturl item music" data-url="aHR0cHM6Ly9tdXNpYy4xNjMuY29tLyMvdXNlci9ob21lP2lkPTE1MTc2ODUzMzM=" title="https:&#x2F;&#x2F;music.163.com&#x2F;#&#x2F;user&#x2F;home?id&#x3D;1517685333"><i class="ic i-cloud-music"></i></span> <span class="exturl item email" data-url="bWFpbHRvOnZpb2xlbW9uQDE2My5jb20=" title="mailto:violemon@163.com"><i class="ic i-envelope"></i></span> <span class="exturl item bangumi" data-url="aHR0cHM6Ly9iYW5ndW1pLnR2L2FuaW1lL2xpc3QvNjY4MDE2" title="https:&#x2F;&#x2F;bangumi.tv&#x2F;anime&#x2F;list&#x2F;668016"><i class="ic i-bilibili"></i></span></div><ul class="menu"><li class="item"><a href="/" rel="section"><i class="ic i-home"></i>首页</a></li><li class="item dropdown"><a href="javascript:void(0);"><i class="ic i-magic"></i>关于</a><ul class="submenu"><li class="item"><a href="/author/" rel="section"><i class="ic i-user"></i>本人</a></li></ul></li><li class="item dropdown"><a href="javascript:void(0);"><i class="ic i-feather"></i>文章</a><ul class="submenu"><li class="item"><a href="/archives/" rel="section"><i class="ic i-list-alt"></i>归档</a></li><li class="item"><a href="/categories/" rel="section"><i class="ic i-th"></i>分类</a></li><li class="item"><a href="/tags/" rel="section"><i class="ic i-tags"></i>标签</a></li></ul></li><li class="item dropdown"><a href="javascript:void(0);"><i class="ic i-cloud"></i>其它</a><ul class="submenu"><li class="item"><a href="/music/" rel="section"><i class="ic i-music"></i>音乐区</a></li><li class="item"><a href="/friends/" rel="section"><i class="ic i-heart"></i>朋友</a></li></ul></li></ul></div></div></div><ul id="quick"><li class="prev pjax"><a href="/music/%E5%B8%B8%E8%A7%81%E5%92%8C%E5%BC%A6/" rel="prev" title="上一篇"><i class="ic i-chevron-left"></i></a></li><li class="up"><i class="ic i-arrow-up"></i></li><li class="down"><i class="ic i-arrow-down"></i></li><li class="next pjax"><a href="/life/%E7%B2%BE%E7%A5%9E%E5%88%86%E6%9E%90%E4%B8%AD%E7%9A%84%E5%80%92%E9%94%99/" rel="next" title="下一篇"><i class="ic i-chevron-right"></i></a></li><li class="percent"></li></ul></div></div><div class="dimmer"></div></div></main><footer id="footer"><div class="inner"><div class="widgets"><div class="rpost pjax"><h2>随机文章</h2><ul><li class="item"><div class="breadcrumb"><a href="/categories/note/" title="分类于 影视书籍">影视书籍</a> <i class="ic i-angle-right"></i> <a href="/categories/note/%E4%B9%A6/" title="分类于 书">书</a></div><span><a href="/%E8%BF%98%E5%8E%9F%E4%B8%8E%E7%BB%99%E4%BA%88%E8%AF%BB%E4%B9%A6%E7%AC%94%E8%AE%B0/" title="还原与给予读书笔记">还原与给予读书笔记</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/code/" title="分类于 代码">代码</a> <i class="ic i-angle-right"></i> <a href="/categories/code/python/" title="分类于 python">python</a></div><span><a href="/code/%E7%B3%BB%E7%BB%9F%E5%AD%A6%E4%B9%A0python3/" title="系统学习python（三）对于Anndata的整理">系统学习python（三）对于Anndata的整理</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/code/" title="分类于 代码">代码</a> <i class="ic i-angle-right"></i> <a href="/categories/code/%E6%95%B0%E5%AD%A6%E5%BB%BA%E6%A8%A1/" title="分类于 数学建模">数学建模</a></div><span><a href="/code/%E5%B1%82%E6%AC%A1%E5%88%86%E6%9E%90%E6%B3%95/" title="层次分析法">层次分析法</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/article/" title="分类于 论文">论文</a></div><span><a href="/article/domain%E5%86%85%E5%AE%B9%E7%9A%84%E6%80%BB%E7%BB%93/" title="对spatial domain内容的总结">对spatial domain内容的总结</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/code/" title="分类于 代码">代码</a> <i class="ic i-angle-right"></i> <a href="/categories/code/%E7%BC%96%E7%A8%8B%E8%AF%AD%E8%A8%80/" title="分类于 编程语言">编程语言</a></div><span><a href="/code/c%E5%A4%8D%E4%B9%A0/" title="c复习">c复习</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/code/" title="分类于 代码">代码</a> <i class="ic i-angle-right"></i> <a href="/categories/code/OI/" title="分类于 OI">OI</a></div><span><a href="/code/Dijkstra/" title="Dijkstra">Dijkstra</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/music/" title="分类于 音乐">音乐</a> <i class="ic i-angle-right"></i> <a href="/categories/music/%E4%B9%90%E8%B0%B1/" title="分类于 乐谱">乐谱</a></div><span><a href="/music/%E6%8C%87%E5%BC%B9%E8%B0%B1%E5%90%88%E9%9B%86/" title="指弹谱合集">指弹谱合集</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/note/" title="分类于 影视书籍">影视书籍</a> <i class="ic i-angle-right"></i> <a href="/categories/note/%E4%B9%A6/" title="分类于 书">书</a></div><span><a href="/note/%E3%80%8A%E7%B4%A0%E9%A3%9F%E8%80%85%E3%80%8B%EF%BC%9A%E5%B7%A7%E5%85%8B%E5%8A%9B%E5%91%B3%E7%9A%84%E5%B1%8E/" title="《素食者》：巧克力味的屎">《素食者》：巧克力味的屎</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/code/" title="分类于 代码">代码</a> <i class="ic i-angle-right"></i> <a href="/categories/code/OI/" title="分类于 OI">OI</a></div><span><a href="/code/LCA/" title="LCA">LCA</a></span></li><li class="item"><div class="breadcrumb"><a href="/categories/code/" title="分类于 代码">代码</a> <i class="ic i-angle-right"></i> <a href="/categories/code/%E6%95%B0%E5%AD%A6%E5%BB%BA%E6%A8%A1/" title="分类于 数学建模">数学建模</a></div><span><a href="/code/TOPSIS-MATLAB/" title="TOPSIS-MATLAB">TOPSIS-MATLAB</a></span></li></ul></div><div><h2>最新评论</h2><ul class="leancloud-recent-comment"></ul></div></div><div class="status"><div class="copyright">&copy; 2010 – <span itemprop="copyrightYear">2025</span> <span class="with-love"><i class="ic i-sakura rotate"></i> </span><span class="author" itemprop="copyrightHolder">Lemon Sour @ Amentiraz</span></div><div class="powered-by">基于 <span class="exturl" data-url="aHR0cHM6Ly9oZXhvLmlv">Hexo</span> & Theme.<span class="exturl" data-url="aHR0cHM6Ly9naXRodWIuY29tL2FtZWhpbWUvaGV4by10aGVtZS1zaG9rYQ==">Shoka</span></div></div></div></footer></div><script data-config type="text/javascript">var LOCAL={path:"article/domain内容的总结/",favicon:{show:"⁽⁽ଘ( ˊᵕˋ )ଓ⁾⁾",hide:"(つд⊂)"},search:{placeholder:"文章搜索",empty:"关于 「 ${query} 」，什么也没搜到",stats:"${time} ms 内找到 ${hits} 条结果"},valine:!0,copy_tex:!0,katex:!0,fancybox:!0,copyright:'复制成功，转载请遵守 <i class="ic i-creative-commons"></i>BY-NC-SA 协议。',ignores:[function(e){return e.includes("#")},function(e){return new RegExp(LOCAL.path+"$").test(e)}]}</script><script src="https://cdn.polyfill.io/v2/polyfill.js"></script><script src="//cdn.jsdelivr.net/combine/npm/pace-js@1.0.2/pace.min.js,npm/pjax@0.2.8/pjax.min.js,npm/whatwg-fetch@3.4.0/dist/fetch.umd.min.js,npm/animejs@3.2.0/lib/anime.min.js,npm/algoliasearch@4/dist/algoliasearch-lite.umd.js,npm/instantsearch.js@4/dist/instantsearch.production.min.js,npm/lozad@1/dist/lozad.min.js,npm/quicklink@2/dist/quicklink.umd.js"></script><script src="/js/app.js?v=0.2.5"></script></body></html><!-- rebuild by hrmmi -->